---
output: 
    html_document: 
        code_folding: hide
---
<style type="text/css">
.main-container {
  max-width: 1800px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<pre>
</pre>
# Neuronet{.tabset}
<br>
</br>
<font size = "5">For all experimental conditions, the bulk RNA was isolated from 
the mouse brain
    
+ of **treated** (4Rx) or **untreated** mice
+ from different regions (**penumbra** or **core**) 
+ for different timepoints (**3 days** or **7 days**) 

after **middle cerebral artery occlusion**, then sequenced via Illumina next-generation sequencing. 
The **raw reads** were used for pseudomapping with **salmon**, then subsequently
**DESeq2** was used to infer differential gene expression, adn gene set enrichment 
analysis was performed via **clusteRprofiler**.
</font>
<br>
</br>
<font size = "5">Experimental groups:

 + ctrl - 3 days - penumbra
 + sham - 3 days - penumbra
 + trt(4Rx) - 3 days - penumbra 
 + ctrl - 7 days - penumbra
 + trt(4Rx) - 7 days - penumbra
 + ctrl - 7 days - core
 + trt(4Rx) - 7 days - core 
  
The model for differentially expression is based on the 
**contrast of control to treatment**, ctrl is set as the reference level.</font>
<pre>
</pre>
## Models
### DESeq2 model 
```{r RcodeDESeq2, message = FALSE, warning = FALSE, results = FALSE}
# library import 
library("DESeq2")
library("readr")
library("tximport")
library("edgeR")
library("httpgd")
library("tximport")
library("knitr")
library("ggplot2")
library("ggrepel")
library("tidyverse")
library("RColorBrewer")
library("pheatmap")
library("reshape2")
library("vsn")
library("ggvenn")
library("ggpubr")
library("ggplotify")
library("biomaRt")
library("clusterProfiler")
library("org.Mm.eg.db")
library("msigdbr")
library("vissE")
library("msigdb")
library("GSEABase")
library("igraph")
library("patchwork")
library("ggpattern")
library("tm")
library("lessR")
library("gridGraphics")
library("grid")
require("devtools")
library("cowplot")
library("ggwordcloud")
library("stringr")
library("Glimma")
library("scales")
library("visNetwork")
library("gridExtra")
library("GOplot")
library("ggthemes")
# size of plots
knitr::opts_chunk$set(fig.width = 18, fig.height = 12)
# loading mapping statistics
stats_star <- read.csv(
    "/projects/neuronet/seq_april/processing/mapping/ensembl/raw/statistics.tsv",
    sep = "\t", header = TRUE, row.names = "ID"
)
stats_hisat2 <- read.csv(
    "/projects/neuronet/neuronet/processing/mapping/statistics.tsv",
    sep = "\t", header = TRUE, row.names = "ID"
)
# SAMPLES TABLE, txi imports etc ---------------------------------------------
samples <- read.table("/projects/neuronet/neuronet/processing/samples_neuronet.csv", sep = ",", header = TRUE)
samples$condition <- as.factor(samples$condition)
samples$timepoint <- as.factor(samples$timepoint)
samples$brainRegion <- as.factor(samples$brainRegion)
samples$fileName <- do.call(paste, c(samples, sep = "-"))
samples$fileName <- str_replace(samples$fileName, "4Rx", "trt")
rownames(samples) <- samples$fileName
samples$group <- factor(paste0(samples$condition, samples$timepoint, samples$brainRegion))
samples$timeAfterStroke <- factor(samples$timepoint, levels = c("day3", "day7", "day14", "sham"))
samples$timeAfterStroke[samples$condition == "sham"] <- factor("sham")

tx2gene <- read_csv("/projects/neuronet/neuronet/processing/genAnno/Mus_musculus.GRCm39.110.tx2gene", col_names = FALSE)
quantFiles <- file.path("/projects/neuronet/neuronet/processing/mapping/salmon", samples$fileName, "quant.sf")
all(file.exists(quantFiles))
names(quantFiles) <- samples$fileName
txi <- tximport(quantFiles, type = "salmon", tx2gene = tx2gene)

txiTime <- lapply(txi, function(x) if(is.matrix(x)) return(x[,samples$fileName[which(samples$group %in% c("ctrlday3ipsiPenumbra", "4Rxday3ipsiPenumbra", "ctrlday7ipsiPenumbra", "4Rxday7ipsiPenumbra", "ctrlday14ipsiPenumbra", "4Rxday14ipsiPenumbra"))]]) else return(x))

samplesTime <- samples[colnames(txiTime$counts),]
samplesTime$condition <- droplevels(samplesTime$condition)
# r models, chi2 significance cutoff for wald test

mm <- model.matrix(~ group, data= samples)
mm2 <- model.matrix(~ condition + timepoint + condition:timepoint, data= samplesTime)
mm3 <- model.matrix(~ timepoint + condition + timepoint:condition, data= samplesTime)
qchisq(p = 0.1, df = 43, lower.tail = TRUE)

# ------------------- DESeq2 MODELS BLOCK-WISE COMPARISON------------

dds <- DESeqDataSetFromTximport(txi, colData = samples, design = ~group)
dds$condition <- relevel(dds$group, ref = "ctrlday3ipsiPenumbra")
dds_keep <- rowSums(counts(dds) >= 5) >= 5
dds <- dds[dds_keep, ]
dds <- DESeq(dds)

#same days, same regions
trt3penVSctrl3pen <- results(dds, alpha = 0.1, contrast = c("group", "4Rxday3ipsiPenumbra", "ctrlday3ipsiPenumbra"))
SIGtrt3penVSctrl3pen <- subset(trt3penVSctrl3pen, padj < 0.1)
trt7penVSctrl7pen <- results(dds, alpha = 0.1, contrast = c("group", "4Rxday7ipsiPenumbra", "ctrlday7ipsiPenumbra"))
SIGtrt7penVSctrl7pen <- subset(trt7penVSctrl7pen, padj < 0.1)
trt7coreVSctrl7core <- results(dds, alpha = 0.1, contrast = c("group", "4Rxday7ipsiCore", "ctrlday7ipsiCore"))
SIGtrt7coreVSctrl7core <- subset(trt7coreVSctrl7core, padj < 0.1)
trt14penVSctrl14pen <- results(dds, alpha = 0.1, contrast = c("group", "4Rxday14ipsiPenumbra", "ctrlday14ipsiPenumbra"))
SIGtrt14penVSctrl14pen <- subset(trt14penVSctrl14pen, padj < 0.1)
SIG05trt14penVSctrl14pen <- subset(trt14penVSctrl14pen, padj < 0.05)


#different days, same regions
ctrl7penVSctrl3pen <- results(dds, alpha = 0.1, contrast = c("group", "ctrlday7ipsiPenumbra", "ctrlday3ipsiPenumbra"))
SIGctrl7penVSctrl3pen <- subset(ctrl7penVSctrl3pen, padj < 0.1)
ctrl14penVSctrl3pen <- results(dds, alpha = 0.1, contrast = c("group", "ctrlday14ipsiPenumbra", "ctrlday3ipsiPenumbra"))
SIGctrl14penVSctrl3pen <- subset(ctrl14penVSctrl3pen, padj < 0.1)
ctrl14penVSctrl7pen <- results(dds, alpha = 0.1, contrast = c("group", "ctrlday14ipsiPenumbra", "ctrlday7ipsiPenumbra"))
SIGctrl14penVSctrl7pen <- subset(ctrl14penVSctrl7pen, padj < 0.1)

trt7penVStrt3pen <- results(dds, alpha = 0.1, contrast = c("group", "4Rxday7ipsiPenumbra", "4Rxday3ipsiPenumbra"))
SIGtrt7penVStrt3pen <- subset(trt7penVStrt3pen, padj < 0.1)
trt14penVStrt3pen <- results(dds, alpha = 0.1, contrast = c("group", "4Rxday14ipsiPenumbra", "4Rxday3ipsiPenumbra"))
SIGtrt14penVStrt3pen <- subset(trt14penVStrt3pen, padj < 0.1)
trt14penVStrt7pen <- results(dds, alpha = 0.1, contrast = c("group", "4Rxday14ipsiPenumbra", "4Rxday7ipsiPenumbra"))
SIGtrt14penVStrt7pen <- subset(trt14penVStrt7pen, padj < 0.1)


#sham
sham3penVSctrl3pen <- results(dds, alpha = 0.1, contrast = c("group", "shamday3ipsiPenumbra", "ctrlday3ipsiPenumbra"))
SIGsham3penVSctrl3pen <- subset(sham3penVSctrl3pen, padj < 0.1)
sham3penVStrt3pen <- results(dds, alpha = 0.1, contrast = c("group", "shamday3ipsiPenumbra", "4Rxday3ipsiPenumbra"))
SIGsham3penVStrt3pen <- subset(sham3penVStrt3pen, padj < 0.1)
sham3penVSctrl7pen <- results(dds, alpha = 0.1, contrast = c("group", "shamday3ipsiPenumbra", "ctrlday7ipsiPenumbra"))
SIGsham3penVSctrl7pen <- subset(sham3penVSctrl7pen, padj < 0.1)
sham3penVStrt7pen <- results(dds, alpha = 0.1, contrast = c("group", "shamday3ipsiPenumbra", "4Rxday7ipsiPenumbra"))
SIGsham3penVStrt7pen <- subset(sham3penVStrt7pen, padj < 0.1)
sham3penVSctrl7core <- results(dds, alpha = 0.1, contrast = c("group", "shamday3ipsiPenumbra", "ctrlday7ipsiCore"))
SIGsham3penVSctrl7core <- subset(sham3penVSctrl7core, padj < 0.1)
sham3penVStrt7core <- results(dds, alpha = 0.1, contrast = c("group", "shamday3ipsiPenumbra", "4Rxday7ipsiCore"))
SIGsham3penVStrt7core <- subset(sham3penVStrt7core, padj < 0.1)
sham3penVSctrl14pen <- results(dds, alpha = 0.1, contrast = c("group", "shamday3ipsiPenumbra", "ctrlday14ipsiPenumbra"))
SIGsham3penVSctrl14pen <- subset(sham3penVSctrl14pen, padj < 0.1)
sham3penVStrt14pen <- results(dds, alpha = 0.1, contrast = c("group", "shamday3ipsiPenumbra", "4Rxday14ipsiPenumbra"))
SIGsham3penVStrt14pen <- subset(sham3penVStrt14pen, padj < 0.1)



#different regions
ctrl7coreVSctrl7pen <- results(dds, alpha = 0.1, contrast = c("group", "ctrlday7ipsiCore", "ctrlday7ipsiPenumbra"))
SIGctrl7coreVSctrl7pen <- subset(ctrl7coreVSctrl7pen, padj < 0.1)
trt7coreVStrt7pen <- results(dds, alpha = 0.1, contrast = c("group", "4Rxday7ipsiCore", "4Rxday7ipsiPenumbra"))
SIGtrt7coreVStrt7pen <- subset(trt7coreVStrt7pen, padj < 0.1)


#--------------------------DESeq2 MODEL MODELING TIME AND INTERACTION----------------

# TIME FULL MODEL, with or without interaction
# now this model asks if the predictor time, acts different on my dependent variables in trt and ctrl mice

## for the explanation of the coefficients 
### see [the youtube video ](https://www.youtube.com/watch?v=X6p3E-QTcUc), pretty well explained
ddsTime <- DESeqDataSetFromTximport(txiTime, colData = samples[which(samples$group %in% c("ctrlday3ipsiPenumbra", "4Rxday3ipsiPenumbra", "ctrlday7ipsiPenumbra", "4Rxday7ipsiPenumbra", "ctrlday14ipsiPenumbra", "4Rxday14ipsiPenumbra")), ], design = ~ timepoint + condition + timepoint:condition )
ddsTime$condition <- relevel(ddsTime$condition, ref = "ctrl")
ddsTime_keep <- rowSums(counts(ddsTime) >= 5) >= 5
ddsTime <- ddsTime[ddsTime_keep, ]
ddsTime <- DESeq(ddsTime, test = "LRT", reduced = ~ timepoint + condition)

#resultsNames(ddsTime)
#head(coef(ddsTime))

timeInteraction <- results(ddsTime) # this is the condition:timepoint interaction term
SIGtimeInteraction <- subset(timeInteraction, padj < 0.1)
timeCtrlTrtDay3 <- results(ddsTime, name = "condition_4Rx_vs_ctrl") # this is the difference between ctrl and trt on day 3 (condition effect), should be the same as in blockwise comparison
SIGtimeCtrlTrtDay3 <- subset(timeCtrlTrtDay3, padj < 0.1)

## REDUCED TIME MODEL, without interaction, and with or without condition
## does time alone explain the change in gene expression???
ddsTimeReduced <- DESeqDataSetFromTximport(txiTime, colData = samplesTime, design = ~ timepoint + condition )
ddsTimeReduced$condition <- relevel(ddsTimeReduced$condition, ref = "ctrl")
ddsTimeReduced_keep <- rowSums(counts(ddsTimeReduced) >= 5) >= 5
ddsTimeReduced <- ddsTimeReduced[ddsTimeReduced_keep, ]
ddsTimeReduced <- DESeq(ddsTimeReduced, test = "LRT", reduced = ~ timepoint)

timeReduced <- results(ddsTimeReduced)
SIGtimeReduced <- subset(timeReduced, padj < 0.1)

#---------------------------------VARIOUS-SELECTIONS------------------------------------

counts <- as.data.frame(txi$counts)
sampleCounts <- as.data.frame(colSums(txi$counts))
geneCounts <- as.data.frame(rowSums(txi$counts))
names(geneCounts) <- "counts"
geneCounts <- geneCounts[order(geneCounts$counts, decreasing = TRUE), , drop = FALSE]

#---------------data transformations

vst <- vst(dds, blind = TRUE) # blind = TRUE for comparing samples in QA
rlog <- rlog(dds, blind = TRUE)
normTransf <- normTransform(dds)

vstTime <- vst(ddsTime, blind = TRUE)
rlogTime <- rlog(ddsTime, blind = TRUE)
## not sure when to use those two with "blind = FALSE"
vstEye <- vst(dds, blind = FALSE)
rlogEye <- rlog(dds, blind = FALSE)


#lfcShrink <- lfcShrink(dds, coef = , type = "apeglm")
#resultsNames(dds)


## markus genes of interest
head(trt3penVSctrl3pen)
trt3penVSctrl3pen[c("Hexb", "Cst3", "Cx3cr1", "Ctsd", "Csf1r", "Ctss", "Sparc", "Tmsb4x", "P2ry12", "C1qa", "C1qb"),]


## some results as csv files 
write.table(SIGtimeInteraction, file = "/projects/neuronet/neuronet/results/SIGtimeInteraction.csv", row.names = TRUE, sep = ",")

write.table(SIGtrt3penVSctrl3pen, file = "/projects/neuronet/neuronet/results/SIGtrt3penVSctrl3pen.csv", row.names = TRUE, sep = ",")
write.table(SIGtrt7penVSctrl7pen, file = "/projects/neuronet/neuronet/results/SIGtrt7penVSctrl7pen.csv", row.names = TRUE, sep = ",")
write.table(SIGtrt14penVSctrl14pen, file = "/projects/neuronet/neuronet/results/SIGtrt14penVSctrl14pen.csv", row.names = TRUE, sep = ",")

write.table(SIGtrt7penVStrt3pen, file = "/projects/neuronet/neuronet/results/SIGtrt7penVStrt3pen.csv", row.names = TRUE, sep = ",")
write.table(SIGtrt14penVStrt3pen, file = "/projects/neuronet/neuronet/results/SIGtrt14penVStrt3pen.csv", row.names = TRUE, sep = ",")
write.table(SIGtrt14penVStrt7pen, file = "/projects/neuronet/neuronet/results/SIGtrt14penVStrt7pen.csv", row.names = TRUE, sep = ",")


write.table(SIGctrl7penVSctrl3pen, file = "/projects/neuronet/neuronet/results/SIGctrl7penVSctrl3pen.csv", row.names = TRUE, sep = ",")
write.table(SIGctrl14penVSctrl3pen, file = "/projects/neuronet/neuronet/results/SIGctrl14penVSctrl3pen.csv", row.names = TRUE, sep = ",")
write.table(SIGctrl14penVSctrl7pen, file = "/projects/neuronet/neuronet/results/SIGctrl14penVSctrl7pen.csv", row.names = TRUE, sep = ",")

write.table(SIGtrt7coreVSctrl7core, file = "/projects/neuronet/neuronet/results/SIGtrt7coreVSctrl7core.csv", row.names = TRUE, sep = ",")
write.table(SIGtrt7coreVStrt7pen, file = "/projects/neuronet/neuronet/results/SIGtrt7coreVStrt7pen.csv", row.names = TRUE, sep = ",")


```
<pre>



</pre>
### Gene Ontology 
```{r goModel, message = FALSE, warning = FALSE, results = FALSE}

####---------trt3penVSctrl3pen-------------------------------------
# Gene set enrichment analysis GO terms
trt3penVSctrl3penWald <- trt3penVSctrl3pen$stat
names(trt3penVSctrl3penWald) <- rownames(trt3penVSctrl3pen)
trt3penVSctrl3penWald <- trt3penVSctrl3penWald[order(trt3penVSctrl3penWald, decreasing = TRUE)]
trt3penVSctrl3penGSEA <- gseGO(
    geneList = trt3penVSctrl3penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
trt3penVSctrl3penGSEA <- trt3penVSctrl3penGSEA[order(trt3penVSctrl3penGSEA$NES), asis = TRUE]
trt3penVSctrl3penGSEA_simplify <- clusterProfiler::simplify(trt3penVSctrl3penGSEA)
## GSEA results as dataframe
trt3penVSctrl3penGSEADf <- as.data.frame(trt3penVSctrl3penGSEA)
trt3penVSctrl3penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt3penVSctrl3penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
trt3penVSctrl3penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt3penVSctrl3penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
trt3penVSctrl3penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt3penVSctrl3penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100

trt3penVSctrl3penGSEADf$leading_edge <- NULL
trt3penVSctrl3penGSEADf <- trt3penVSctrl3penGSEADf[order(trt3penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
trt3penVSctrl3penGSEADf$Direction <- "DownRegulated"
trt3penVSctrl3penGSEADf[which(trt3penVSctrl3penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"


### GSEA selection
trt3penVSctrl3penGSEADf$Description[grepl("regulation of tumor necrosis", trt3penVSctrl3penGSEADf$Description)]
trt3penVSctrl3penChord <- c("regulation of synaptic transmission, glutamatergic", "regulation of synaptic plasticity", "angiogenesis", "positive regulation of vasculature development", "leukocyte migration", "regulation of tumor necrosis factor superfamily cytokine production")

neuroplasticity <- c("regulation of synaptic plasticity", "regulation of synaptic transmission, glutamatergic", "central nervous system neuron development", "modulation of chemical synaptic transmission")
inflammation <- c("positive regulation of tumor necrosis factor superfamily cytokine production", "regulation of tumor necrosis factor superfamily cytokine production", "leukocyte migration", "production of molecular mediator involved in inflammatory response")
angiogenesis <- c("angiogenesis", "positive regulation of vasculature development", "positive regulation of angiogenesis", "apical junction assembly")

trt3penVSctrl3penGSEA_neuroplasticity <- trt3penVSctrl3penGSEA[which(trt3penVSctrl3penGSEA$Description %in% neuroplasticity),, asis = TRUE]
trt3penVSctrl3penGSEA_inflammation <- trt3penVSctrl3penGSEA[which(trt3penVSctrl3penGSEA$Description %in% inflammation),, asis = TRUE]
trt3penVSctrl3penGSEA_angiogenesis <- trt3penVSctrl3penGSEA[which(trt3penVSctrl3penGSEA$Description %in% angiogenesis),, asis = TRUE]

trt3penVSctrl3penGSEA_selection <- trt3penVSctrl3penGSEA[which(trt3penVSctrl3penGSEA$Description %in% c(neuroplasticity, inflammation, angiogenesis)),, asis = TRUE]


###----------------trt7penVSctrl7pen---------------
# Gene set enrichment analysis GO terms
trt7penVSctrl7penWald <- trt7penVSctrl7pen$stat
names(trt7penVSctrl7penWald) <- rownames(trt7penVSctrl7pen)
trt7penVSctrl7penWald <- trt7penVSctrl7penWald[order(trt7penVSctrl7penWald, decreasing = TRUE)]
trt7penVSctrl7penGSEA <- gseGO(
    geneList = trt7penVSctrl7penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
trt7penVSctrl7penGSEA <- trt7penVSctrl7penGSEA[order(trt7penVSctrl7penGSEA$NES), asis = TRUE]
trt7penVSctrl7penGSEA_simplify <- clusterProfiler::simplify(trt7penVSctrl7penGSEA)
## GSEA results as dataframe
trt7penVSctrl7penGSEADf <- as.data.frame(trt7penVSctrl7penGSEA)
trt7penVSctrl7penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7penVSctrl7penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
trt7penVSctrl7penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7penVSctrl7penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
trt7penVSctrl7penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7penVSctrl7penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100


trt7penVSctrl7penGSEADf$leading_edge <- NULL
trt7penVSctrl7penGSEADf <- trt7penVSctrl7penGSEADf[order(trt7penVSctrl7penGSEADf$leadTag, decreasing = TRUE), ]
trt7penVSctrl7penGSEADf$Direction <- "DownRegulated"
trt7penVSctrl7penGSEADf[which(trt7penVSctrl7penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"


### GSEA selection
trt7penVSctrl7penGSEADf$Description[grepl("autophagy of mito", trt7penVSctrl7penGSEADf$Description)]
trt7penVSctrl7penSelection <- c("cell cycle checkpoint signaling", "positive regulation of proteolysis", "extrinsic apoptotic signaling pathway", "autophagy of mitochondrion")
trt7penVSctrl7penGSEA_selection <- trt7penVSctrl7penGSEA[which(trt7penVSctrl7penGSEA$Description %in% trt7penVSctrl7penSelection),, asis = TRUE]



#------------------------------all the other results as gsea----------------------------------------


# trt7coreVSctrl7core
trt7coreVSctrl7coreWald <- trt7coreVSctrl7core$stat
names(trt7coreVSctrl7coreWald) <- rownames(trt7coreVSctrl7core)
trt7coreVSctrl7coreWald <- trt7coreVSctrl7coreWald[order(trt7coreVSctrl7coreWald, decreasing = TRUE)]
trt7coreVSctrl7coreGSEA <- gseGO(
    geneList = trt7coreVSctrl7coreWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
trt7coreVSctrl7coreGSEADf <- as.data.frame(trt7coreVSctrl7coreGSEA)
trt7coreVSctrl7coreGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7coreVSctrl7coreGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
trt7coreVSctrl7coreGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7coreVSctrl7coreGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
trt7coreVSctrl7coreGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7coreVSctrl7coreGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
trt7coreVSctrl7coreGSEADf$leading_edge <- NULL
trt7coreVSctrl7coreGSEADf <- trt7coreVSctrl7coreGSEADf[order(trt7coreVSctrl7coreGSEADf$leadTag, decreasing = TRUE), ]
trt7coreVSctrl7coreGSEADf$Direction <- "DownRegulated"
trt7coreVSctrl7coreGSEADf[which(trt7coreVSctrl7coreGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

## selection

trt7coreVSctrl7coreGSEADf$Description[grepl("oxidative phosphorylation", trt7coreVSctrl7coreGSEADf$Description)]

trt7coreVSctrl7coreChord <- c(
    "regulation of cytokine production involved in immune response", 
    "regulation of production of molecular mediator of immune response", 
    "alpha-beta T cell differentiation", 
    "bone morphogenesis", 
    "regulation of leukocyte migration", 
    "CD4-positive, alpha-beta T cell activation" 
    )

BPselection <- c(
    "regulation of cytokine production involved in immune response", 
    "regulation of production of molecular mediator of immune response", 
    "alpha-beta T cell differentiation", 
    "bone morphogenesis", 
    "regulation of leukocyte migration", 
    "CD4-positive, alpha-beta T cell activation", 
    "CD4-positive, alpha-beta T cell differentiation", 
    "oxidative phosphorylation"
    )

trt7coreVSctrl7coreGSEA_selection <- trt7coreVSctrl7coreGSEA[which(trt7coreVSctrl7coreGSEA$Description %in% BPselection),, asis = TRUE]


trt7coreVSctrl7coreGSEA_selection$NES

# ctrl7penVSctrl3pen-------------------------------
ctrl7penVSctrl3penWald <- ctrl7penVSctrl3pen$stat
names(ctrl7penVSctrl3penWald) <- rownames(ctrl7penVSctrl3pen)
ctrl7penVSctrl3penWald <- ctrl7penVSctrl3penWald[order(ctrl7penVSctrl3penWald, decreasing = TRUE)]
ctrl7penVSctrl3penGSEA <- gseGO(
    geneList = ctrl7penVSctrl3penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
ctrl7penVSctrl3penGSEADf <- as.data.frame(ctrl7penVSctrl3penGSEA)
ctrl7penVSctrl3penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl7penVSctrl3penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
ctrl7penVSctrl3penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl7penVSctrl3penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
ctrl7penVSctrl3penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl7penVSctrl3penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
ctrl7penVSctrl3penGSEADf$leading_edge <- NULL
ctrl7penVSctrl3penGSEADf <- ctrl7penVSctrl3penGSEADf[order(ctrl7penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
ctrl7penVSctrl3penGSEADf$Direction <- "DownRegulated"
ctrl7penVSctrl3penGSEADf[which(ctrl7penVSctrl3penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"


# trt7penVStrt3pen--------------------------------------------------------------------------------
trt7penVStrt3penWald <- trt7penVStrt3pen$stat
names(trt7penVStrt3penWald) <- rownames(trt7penVStrt3pen)
trt7penVStrt3penWald <- trt7penVStrt3penWald[order(trt7penVStrt3penWald, decreasing = TRUE)]
trt7penVStrt3penGSEA <- gseGO(
    geneList = trt7penVStrt3penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
trt7penVStrt3penGSEADf <- as.data.frame(trt7penVStrt3penGSEA)
trt7penVStrt3penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7penVStrt3penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
trt7penVStrt3penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7penVStrt3penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
trt7penVStrt3penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7penVStrt3penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
trt7penVStrt3penGSEADf$leading_edge <- NULL
trt7penVStrt3penGSEADf <- trt7penVStrt3penGSEADf[order(trt7penVStrt3penGSEADf$leadTag, decreasing = TRUE), ]
trt7penVStrt3penGSEADf$Direction <- "DownRegulated"
trt7penVStrt3penGSEADf[which(trt7penVStrt3penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

# sham3penVSctrl3pen------------------------------------------------------------------------------
sham3penVSctrl3penWald <- sham3penVSctrl3pen$stat
names(sham3penVSctrl3penWald) <- rownames(sham3penVSctrl3pen)
sham3penVSctrl3penWald <- sham3penVSctrl3penWald[order(sham3penVSctrl3penWald, decreasing = TRUE)]
sham3penVSctrl3penGSEA <- gseGO(
    geneList = sham3penVSctrl3penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
sham3penVSctrl3penGSEADf <- as.data.frame(sham3penVSctrl3penGSEA)
sham3penVSctrl3penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl3penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl3penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl3penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl3penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl3penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl3penGSEADf$leading_edge <- NULL
sham3penVSctrl3penGSEADf <- sham3penVSctrl3penGSEADf[order(sham3penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
sham3penVSctrl3penGSEADf$Direction <- "DownRegulated"
sham3penVSctrl3penGSEADf[which(sham3penVSctrl3penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

# sham3penVStrt3pen------------------------------------------------------------------------------
sham3penVStrt3penWald <- sham3penVStrt3pen$stat
names(sham3penVStrt3penWald) <- rownames(sham3penVStrt3pen)
sham3penVStrt3penWald <- sham3penVStrt3penWald[order(sham3penVStrt3penWald, decreasing = TRUE)]
sham3penVStrt3penGSEA <- gseGO(
    geneList = sham3penVStrt3penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
sham3penVStrt3penGSEADf <- as.data.frame(sham3penVStrt3penGSEA)
sham3penVStrt3penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt3penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt3penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt3penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt3penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt3penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt3penGSEADf$leading_edge <- NULL
sham3penVStrt3penGSEADf <- sham3penVStrt3penGSEADf[order(sham3penVStrt3penGSEADf$leadTag, decreasing = TRUE), ]
sham3penVStrt3penGSEADf$Direction <- "DownRegulated"
sham3penVStrt3penGSEADf[which(sham3penVStrt3penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

# sham3penVSctrl7pen----------------------------------------------------------------------
sham3penVSctrl7penWald <- sham3penVSctrl7pen$stat
names(sham3penVSctrl7penWald) <- rownames(sham3penVSctrl7pen)
sham3penVSctrl7penWald <- sham3penVSctrl7penWald[order(sham3penVSctrl7penWald, decreasing = TRUE)]
sham3penVSctrl7penGSEA <- gseGO(
    geneList = sham3penVSctrl7penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
sham3penVSctrl7penGSEADf <- as.data.frame(sham3penVSctrl7penGSEA)
sham3penVSctrl7penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl7penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl7penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl7penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl7penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl7penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl7penGSEADf$leading_edge <- NULL
sham3penVSctrl7penGSEADf <- sham3penVSctrl7penGSEADf[order(sham3penVSctrl7penGSEADf$leadTag, decreasing = TRUE), ]
sham3penVSctrl7penGSEADf$Direction <- "DownRegulated"
sham3penVSctrl7penGSEADf[which(sham3penVSctrl7penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

# sham3penVStrt7pen --------------------------------------------------------
sham3penVStrt7penWald <- sham3penVStrt7pen$stat
names(sham3penVStrt7penWald) <- rownames(sham3penVStrt7pen)
sham3penVStrt7penWald <- sham3penVStrt7penWald[order(sham3penVStrt7penWald, decreasing = TRUE)]
sham3penVStrt7penGSEA <- gseGO(
    geneList = sham3penVStrt7penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
sham3penVStrt7penGSEADf <- as.data.frame(sham3penVStrt7penGSEA)
sham3penVStrt7penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt7penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt7penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt7penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt7penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt7penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt7penGSEADf$leading_edge <- NULL
sham3penVStrt7penGSEADf <- sham3penVStrt7penGSEADf[order(sham3penVStrt7penGSEADf$leadTag, decreasing = TRUE), ]
sham3penVStrt7penGSEADf$Direction <- "DownRegulated"
sham3penVStrt7penGSEADf[which(sham3penVStrt7penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

# sham3penVSctrl7core ------------------------------------------------------------------
sham3penVSctrl7coreWald <- sham3penVSctrl7core$stat
names(sham3penVSctrl7coreWald) <- rownames(sham3penVSctrl7core)
sham3penVSctrl7coreWald <- sham3penVSctrl7coreWald[order(sham3penVSctrl7coreWald, decreasing = TRUE)]
sham3penVSctrl7coreGSEA <- gseGO(
    geneList = sham3penVSctrl7coreWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
sham3penVSctrl7coreGSEADf <- as.data.frame(sham3penVSctrl7coreGSEA)
sham3penVSctrl7coreGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl7coreGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl7coreGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl7coreGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl7coreGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVSctrl7coreGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
sham3penVSctrl7coreGSEADf$leading_edge <- NULL
sham3penVSctrl7coreGSEADf <- sham3penVSctrl7coreGSEADf[order(sham3penVSctrl7coreGSEADf$leadTag, decreasing = TRUE), ]
sham3penVSctrl7coreGSEADf$Direction <- "DownRegulated"
sham3penVSctrl7coreGSEADf[which(sham3penVSctrl7coreGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"


# sham3penVStrt7core----------------------------------------------------------------------
sham3penVStrt7coreWald <- sham3penVStrt7core$stat
names(sham3penVStrt7coreWald) <- rownames(sham3penVStrt7core)
sham3penVStrt7coreWald <- sham3penVStrt7coreWald[order(sham3penVStrt7coreWald, decreasing = TRUE)]
sham3penVStrt7coreGSEA <- gseGO(
    geneList = sham3penVStrt7coreWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
sham3penVStrt7coreGSEADf <- as.data.frame(sham3penVStrt7coreGSEA)
sham3penVStrt7coreGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt7coreGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt7coreGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt7coreGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt7coreGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(sham3penVStrt7coreGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
sham3penVStrt7coreGSEADf$leading_edge <- NULL
sham3penVStrt7coreGSEADf <- sham3penVStrt7coreGSEADf[order(sham3penVStrt7coreGSEADf$leadTag, decreasing = TRUE), ]
sham3penVStrt7coreGSEADf$Direction <- "DownRegulated"
sham3penVStrt7coreGSEADf[which(sham3penVStrt7coreGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

# ctrl7coreVSctrl7pen -------------------------------------------------------------------
ctrl7coreVSctrl7penWald <- ctrl7coreVSctrl7pen$stat
names(ctrl7coreVSctrl7penWald) <- rownames(ctrl7coreVSctrl7pen)
ctrl7coreVSctrl7penWald <- ctrl7coreVSctrl7penWald[order(ctrl7coreVSctrl7penWald, decreasing = TRUE)]
ctrl7coreVSctrl7penGSEA <- gseGO(
    geneList = ctrl7coreVSctrl7penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
ctrl7coreVSctrl7penGSEADf <- as.data.frame(ctrl7coreVSctrl7penGSEA)
ctrl7coreVSctrl7penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl7coreVSctrl7penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
ctrl7coreVSctrl7penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl7coreVSctrl7penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
ctrl7coreVSctrl7penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl7coreVSctrl7penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
ctrl7coreVSctrl7penGSEADf$leading_edge <- NULL
ctrl7coreVSctrl7penGSEADf <- ctrl7coreVSctrl7penGSEADf[order(ctrl7coreVSctrl7penGSEADf$leadTag, decreasing = TRUE), ]
ctrl7coreVSctrl7penGSEADf$Direction <- "DownRegulated"
ctrl7coreVSctrl7penGSEADf[which(ctrl7coreVSctrl7penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

# trt7coreVStrt7pen -------------------------------------------------------------------------
trt7coreVStrt7penWald <- trt7coreVStrt7pen$stat
names(trt7coreVStrt7penWald) <- rownames(trt7coreVStrt7pen)
trt7coreVStrt7penWald <- trt7coreVStrt7penWald[order(trt7coreVStrt7penWald, decreasing = TRUE)]
trt7coreVStrt7penGSEA <- gseGO(
    geneList = trt7coreVStrt7penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
trt7coreVStrt7penGSEADf <- as.data.frame(trt7coreVStrt7penGSEA)
trt7coreVStrt7penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7coreVStrt7penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
trt7coreVStrt7penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7coreVStrt7penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
trt7coreVStrt7penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt7coreVStrt7penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
trt7coreVStrt7penGSEADf$leading_edge <- NULL
trt7coreVStrt7penGSEADf <- trt7coreVStrt7penGSEADf[order(trt7coreVStrt7penGSEADf$leadTag, decreasing = TRUE), ]
trt7coreVStrt7penGSEADf$Direction <- "DownRegulated"
trt7coreVStrt7penGSEADf[which(trt7coreVStrt7penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"



# selection
trt7coreVStrt7penGSEADf$Description[grepl("oxidative phosphorylation", trt7coreVStrt7penGSEADf$Description)]

trt7coreVStrt7penChord <- c(
    "regulation of cytokine production involved in immune response", 
    "regulation of T cell mediated immunity", 
    "regulation of glial cell differentiation", 
    "tissue remodeling", 
    "synapse organization", 
    "axo-dendritic transport")

BPselection <- c(
    "regulation of cytokine production involved in immune response", 
    "regulation of T cell mediated immunity", 
    "regulation of glial cell differentiation", 
    "tissue remodeling", 
    "synapse organization", 
    "axo-dendritic transport", 
    "positive regulation of lymphocyte mediated immunity", 
    "regulation of interleukin-6 production", 
    "excitatory postsynaptic potential", 
    "axoneme assembly", 
    "neurotransmitter secretion", 
    "synapse assembly", 
    "oxidative phosphorylation")

trt7coreVStrt7penGSEA_selection <- trt7coreVStrt7penGSEA[which(trt7coreVStrt7penGSEA$Description %in% BPselection),, asis = TRUE]

trt7coreVStrt7penGSEA_selection$NES

# trt14penVSctrl14pen --------------------------------------------

trt14penVSctrl14penWald <- trt14penVSctrl14pen$stat
names(trt14penVSctrl14penWald) <- rownames(trt14penVSctrl14pen)
trt14penVSctrl14penWald <- trt14penVSctrl14penWald[order(trt14penVSctrl14penWald, decreasing = TRUE)]
trt14penVSctrl14penGSEA <- gseGO(
    geneList = trt14penVSctrl14penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
trt14penVSctrl14penGSEADf <- as.data.frame(trt14penVSctrl14penGSEA)
trt14penVSctrl14penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVSctrl14penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
trt14penVSctrl14penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVSctrl14penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
trt14penVSctrl14penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVSctrl14penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
trt14penVSctrl14penGSEADf$leading_edge <- NULL
trt14penVSctrl14penGSEADf <- trt14penVSctrl14penGSEADf[order(trt14penVSctrl14penGSEADf$leadTag, decreasing = TRUE), ]
trt14penVSctrl14penGSEADf$Direction <- "DownRegulated"
trt14penVSctrl14penGSEADf[which(trt14penVSctrl14penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"


### GSEA selection
trt14penVSctrl14penGSEADf$Description[grepl("ositive regulation of leukocyte cell-cell adhesion", trt14penVSctrl14penGSEADf$Description)]

trt14penVSctrl14penChord <- c(
    "regulation of synaptic transmission, glutamatergic", 
    "learning", 
    "neurotransmitter secretion", 
    "positive regulation of T cell activation", 
    "myeloid leukocyte migration", 
    "regulation of tumor necrosis factor superfamily cytokine production"
    )

BPselection <- c(
    "regulation of synaptic transmission, glutamatergic", 
    "learning", 
    "neurotransmitter secretion", 
    "positive regulation of T cell activation", 
    "myeloid leukocyte migration", 
    "regulation of tumor necrosis factor superfamily cytokine production", 
    "associative learning", 
    "synaptic vesicle cycle", 
    "excitatory postsynaptic potential", 
    "neutrophil migration", 
    "T cell activation involved in immune response", 
    "lymphocyte activation involved in immune response", 
    "positive regulation of leukocyte cell-cell adhesion")


trt14penVSctrl14penGSEA_selection <- trt14penVSctrl14penGSEA[which(trt14penVSctrl14penGSEA$Description %in% BPselection),, asis = TRUE]

# trt14penVStrt3pen--------------------------------------------------------------------------------
trt14penVStrt3penWald <- trt14penVStrt3pen$stat
names(trt14penVStrt3penWald) <- rownames(trt14penVStrt3pen)
trt14penVStrt3penWald <- trt14penVStrt3penWald[order(trt14penVStrt3penWald, decreasing = TRUE)]
trt14penVStrt3penGSEA <- gseGO(
    geneList = trt14penVStrt3penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
trt14penVStrt3penGSEADf <- as.data.frame(trt14penVStrt3penGSEA)
trt14penVStrt3penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVStrt3penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
trt14penVStrt3penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVStrt3penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
trt14penVStrt3penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVStrt3penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
trt14penVStrt3penGSEADf$leading_edge <- NULL
trt14penVStrt3penGSEADf <- trt14penVStrt3penGSEADf[order(trt14penVStrt3penGSEADf$leadTag, decreasing = TRUE), ]
trt14penVStrt3penGSEADf$Direction <- "DownRegulated"
trt14penVStrt3penGSEADf[which(trt14penVStrt3penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"


# ctrl14penVSctrl3pen--------------------------------------------------------------------------------
ctrl14penVSctrl3penWald <- ctrl14penVSctrl3pen$stat
names(ctrl14penVSctrl3penWald) <- rownames(ctrl14penVSctrl3pen)
ctrl14penVSctrl3penWald <- ctrl14penVSctrl3penWald[order(ctrl14penVSctrl3penWald, decreasing = TRUE)]
ctrl14penVSctrl3penGSEA <- gseGO(
    geneList = ctrl14penVSctrl3penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
ctrl14penVSctrl3penGSEADf <- as.data.frame(ctrl14penVSctrl3penGSEA)
ctrl14penVSctrl3penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl14penVSctrl3penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
ctrl14penVSctrl3penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl14penVSctrl3penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
ctrl14penVSctrl3penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl14penVSctrl3penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
ctrl14penVSctrl3penGSEADf$leading_edge <- NULL
ctrl14penVSctrl3penGSEADf <- ctrl14penVSctrl3penGSEADf[order(ctrl14penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
ctrl14penVSctrl3penGSEADf$Direction <- "DownRegulated"
ctrl14penVSctrl3penGSEADf[which(ctrl14penVSctrl3penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"


# trt14penVStrt7pen--------------------------------------------------------------------------------
trt14penVStrt7penWald <- trt14penVStrt7pen$stat
names(trt14penVStrt7penWald) <- rownames(trt14penVStrt7pen)
trt14penVStrt7penWald <- trt14penVStrt7penWald[order(trt14penVStrt7penWald, decreasing = TRUE)]
trt14penVStrt7penGSEA <- gseGO(
    geneList = trt14penVStrt7penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
trt14penVStrt7penGSEADf <- as.data.frame(trt14penVStrt7penGSEA)
trt14penVStrt7penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVStrt7penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
trt14penVStrt7penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVStrt7penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
trt14penVStrt7penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(trt14penVStrt7penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
trt14penVStrt7penGSEADf$leading_edge <- NULL
trt14penVStrt7penGSEADf <- trt14penVStrt7penGSEADf[order(trt14penVStrt7penGSEADf$leadTag, decreasing = TRUE), ]
trt14penVStrt7penGSEADf$Direction <- "DownRegulated"
trt14penVStrt7penGSEADf[which(trt14penVStrt7penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"

# ctrl14penVSctrl7pen--------------------------------------------------------------------------------
ctrl14penVSctrl7penWald <- ctrl14penVSctrl7pen$stat
names(ctrl14penVSctrl7penWald) <- rownames(ctrl14penVSctrl7pen)
ctrl14penVSctrl7penWald <- ctrl14penVSctrl7penWald[order(ctrl14penVSctrl7penWald, decreasing = TRUE)]
ctrl14penVSctrl7penGSEA <- gseGO(
    geneList = ctrl14penVSctrl7penWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
ctrl14penVSctrl7penGSEADf <- as.data.frame(ctrl14penVSctrl7penGSEA)
ctrl14penVSctrl7penGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl14penVSctrl7penGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
ctrl14penVSctrl7penGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl14penVSctrl7penGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
ctrl14penVSctrl7penGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ctrl14penVSctrl7penGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
ctrl14penVSctrl7penGSEADf$leading_edge <- NULL
ctrl14penVSctrl7penGSEADf <- ctrl14penVSctrl7penGSEADf[order(ctrl14penVSctrl7penGSEADf$leadTag, decreasing = TRUE), ]
ctrl14penVSctrl7penGSEADf$Direction <- "DownRegulated"
ctrl14penVSctrl7penGSEADf[which(ctrl14penVSctrl7penGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"


# timeInteraction--------------------------------------------------------------------------------
timeInteractionWald <- timeInteraction$stat
names(timeInteractionWald) <- rownames(timeInteraction)
timeInteractionWald <- timeInteractionWald[order(timeInteractionWald, decreasing = TRUE)]
timeInteractionGSEA <- gseGO(
    geneList = timeInteractionWald,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
timeInteractionGSEADf <- as.data.frame(timeInteractionGSEA)
timeInteractionGSEADf$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(timeInteractionGSEADf$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
timeInteractionGSEADf$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(timeInteractionGSEADf$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
timeInteractionGSEADf$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(timeInteractionGSEADf$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
timeInteractionGSEADf$leading_edge <- NULL
timeInteractionGSEADf <- timeInteractionGSEADf[order(timeInteractionGSEADf$leadTag, decreasing = TRUE), ]
timeInteractionGSEADf$Direction <- "DownRegulated"
timeInteractionGSEADf[which(timeInteractionGSEADf$enrichmentScore > 0), ]$Direction <- "UpRegulated"



## some results as csv files 

write.table(trt3penVSctrl3penGSEADf, file = "/projects/neuronet/neuronet/results/trt3penVSctrl3penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(trt7penVSctrl7penGSEADf, file = "/projects/neuronet/neuronet/results/trt7penVSctrl7penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(trt14penVSctrl14penGSEADf, file = "/projects/neuronet/neuronet/results/trt14penVSctrl14penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(trt7coreVSctrl7coreGSEADf, file = "/projects/neuronet/neuronet/results/trt7coreVSctrl7coreGSEADf.csv", row.names = TRUE, sep = ",")
write.table(trt7coreVStrt7penGSEADf, file = "/projects/neuronet/neuronet/results/trt7coreVStrt7penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(trt7penVStrt3penGSEADf, file = "/projects/neuronet/neuronet/results/trt7penVStrt3penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(trt14penVStrt3penGSEADf, file = "/projects/neuronet/neuronet/results/trt14penVStrt3penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(trt14penVStrt7penGSEADf, file = "/projects/neuronet/neuronet/results/trt14penVStrt7penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(timeInteractionGSEADf, file = "/projects/neuronet/neuronet/results/timeInteractionGSEADf.csv", row.names = TRUE, sep = ",")


write.table(ctrl7penVSctrl3penGSEADf, file = "/projects/neuronet/neuronet/results/ctrl7penVSctrl3penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(ctrl14penVSctrl3penGSEADf, file = "/projects/neuronet/neuronet/results/ctrl14penVSctrl3penGSEADf.csv", row.names = TRUE, sep = ",")
write.table(ctrl14penVSctrl7penGSEADf, file = "/projects/neuronet/neuronet/results/ctrl14penVSctrl7penGSEADf.csv", row.names = TRUE, sep = ",")

```
<pre> 



</pre>
## Data Quality Assurance
### Sparsity plot
A simple plot of the concentration of counts in a single sample 
over the sum of counts per gene. Not technically the same as "sparsity", 
but this plot is useful diagnostic for datasets which might not fit a 
negative binomial assumption: genes with many zeros and individual very 
large counts are difficult to model with the negative binomial 
distribution.
```{r plotSparsity, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
plotSparsity(
    dds,
    col = dds$group,
    ann = TRUE,
    cex = 0.4,
    pch = 4
)
legend(
    "topright",
    legend = levels(dds$group),
    pch = 4,
    col = unique(dds$group)
)
```
<pre>



</pre>
### Plot dispersion estimates
```{r plotDispEst, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
DESeq2::plotDispEsts(dds)
```
<pre>



</pre>
### Bland-Altman plots
An MA-plot is a plot of log-intensity ratios (M-values) versus intensity averages (A-values). 
```{r MAplot, message = FALSE, warning = FALSE, results = FALSE, fig.height = 6, fig.width = 12, fig.align = "center"}
counter <- 1
resultsList <- list(
    "trt3penVSctrl3pen" = trt3penVSctrl3pen,
    "trt7penVSctrl7pen" = trt7penVSctrl7pen,
    "trt7coreVSctrl7core" = trt7coreVSctrl7core,
    "trt14penVSctrl14pen" = trt14penVSctrl14pen,
    "ctrl7penVSctrl3pen" = ctrl7penVSctrl3pen,
    "ctrl14penVSctrl3pen" = ctrl14penVSctrl3pen,
    "ctrl14penVSctrl7pen" = ctrl14penVSctrl7pen,
    "trt7penVStrt3pen" = trt7penVStrt3pen,
    "trt14penVStrt3pen" = trt14penVStrt3pen,
    "trt14penVStrt7pen" = trt14penVStrt7pen,
    "sham3penVSctrl3pen" = sham3penVSctrl3pen,
    "sham3penVStrt3pen" = sham3penVStrt3pen,
    "sham3penVSctrl7pen" = sham3penVSctrl7pen,
    "sham3penVStrt7pen" = sham3penVStrt7pen,
    "sham3penVSctrl7core" = sham3penVSctrl7core,
    "sham3penVStrt7core" = sham3penVStrt7core,
    "sham3penVSctrl14pen" = sham3penVSctrl14pen,
    "sham3penVStrt14pen" = sham3penVStrt14pen,
    "ctrl7coreVSctrl7pen" = ctrl7coreVSctrl7pen,
    "trt7coreVStrt7pen" = trt7coreVStrt7pen,
    "timeInteraction" = timeInteraction,
    "timeReduced" = timeReduced)

for(result in resultsList){
    p1 <- DESeq2::plotMA(result,
        returnData = TRUE)
    plot1 <- p1 %>% dplyr::mutate(
        outOfBounds = abs(lfc) > quantile(p1$lfc, probs = 0.9992), 
        lfc = ifelse(outOfBounds, quantile(p1$lfc, probs = 0.9992) * sign(lfc), lfc)
        ) %>%
        ggplot(aes(x = mean, y = lfc, color = isDE, shape = outOfBounds)) +
        geom_point(size = 0.5) +
        scale_color_manual(values = c("black", "red")) +
        scale_x_log10() + 
        ylim(c(-quantile(p1$lfc, probs = 0.9992),quantile(p1$lfc, probs = 0.9992))) + 
        guides(shape = FALSE) +
        ggtitle(names(resultsList)[counter]) +
        theme_minimal()
    counter <- counter + 1
    print(plot1)
}
```
<pre>



</pre>
### mean vs sd on different transformations
```{r meanVSsdRaw, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
plt1 <- vsn::meanSdPlot(assay(dds), plot = FALSE)
plt1 <- plt1$gg + ggtitle("NON-transformed dds object")+ scale_fill_gradient(low = "yellow", high = "darkred") + scale_y_log10(labels = label_log())

plt2 <- vsn::meanSdPlot(assay(vst), plot = FALSE)
plt2 <- plt2$gg + ggtitle("variance stabilizing transformation blind") + scale_fill_gradient(low = "yellow", high = "darkred")

plt3 <- vsn::meanSdPlot(assay(vstEye), plot = FALSE)
plt3 <- plt3$gg + ggtitle("variance stabilizing transformation non-blind")+ scale_fill_gradient(low = "yellow", high = "darkred")

plt4 <- vsn::meanSdPlot(assay(rlog), plot = FALSE)
plt4 <- plt4$gg + ggtitle("regularized log transformation blind")+ scale_fill_gradient(low = "yellow", high = "darkred")

plt5 <- vsn::meanSdPlot(assay(rlogEye), plot = FALSE)
plt5 <- plt5$gg + ggtitle("regularized log transformation non-blind")+ scale_fill_gradient(low = "yellow", high = "darkred")

plt6 <- vsn::meanSdPlot(assay(normTransf), plot = FALSE)
plt6 <- plt6$gg + ggtitle("normalized counts transformation")+ scale_fill_gradient(low = "yellow", high = "darkred")


## ?? which coefficient should I shrink?? 
#plt7 <- vsn::meanSdPlot(assay(lfcShrink))

ggarrange(plt1, plt6, plt2, plt3, plt4, plt5, 
    ncol = 2, 
    nrow = 3)
```
<pre> 



</pre> 
### PCA
```{r pca, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## why to use log transformed data for PCA: 
## https://stats.stackexchange.com/questions/164381/why-log-transforming-the-data-before-performing-principal-component-analysis
plt <- prcomp(t(assay(rlog)), scale = TRUE)
var_explained <- data.frame("var_explained" = plt$sdev**2/sum(plt$sdev**2))
var_explained$PC <- factor(sprintf("PC%s", seq(1:nrow(var_explained))), levels = sprintf("PC%s", seq(1:nrow(var_explained))))
## setting color and shape palette
colorPalette <- c("#000000", "#E69F00", "#56B4E9", "springgreen2", "#F0E442", "#D55E00", "#CC79A7", "purple4", "aquamarine4" )
shapePalette <- c(15, 16, 17, 18, 19, 8, 4, 9, 7)
# variation explained by the respectice PC
ellbowplot <- ggplot(var_explained, aes(y = var_explained, x = PC)) + 
geom_bar(stat = "identity") +
geom_text(aes(label = round(var_explained, digits = 2)), vjust = -0.5) + 
ggtitle("Variance explained by the respective PC's")
# PCA clustering to group
p1 <- plt$x %>%
    as.data.frame %>%
        ggplot(aes(x = PC1, y = PC2, color = samples$group, shape = samples$group)) + 
        geom_point(size = 4) + 
        guides(color = guide_legend("group"), shape = guide_legend("group")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = shapePalette) +
        labs (
            x = paste ("PC1: ", round(var_explained[1,1]*100, 0), "%"), 
            y = paste("PC2: ", round(var_explained[2,1]*100, 0), "%")) +
        ggtitle("PCA clustered to group") + 
        theme_minimal()

# pca clustering to condition
p2 <- plt$x %>%
    as.data.frame %>%
        ggplot(aes(x = PC1, y = PC2, color = samples$condition, shape = samples$condition)) + 
        geom_point(size = 4) + guides(color = guide_legend("condition"), shape = guide_legend("condition")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = shapePalette) +
        labs (
            x = paste ("PC1: ", round(var_explained[1,1]*100, 0), "%"), 
            y = paste("PC2: ", round(var_explained[2,1]*100, 0), "%")) +
        ggtitle("PCA clustered to condition") + 
        theme_minimal()
# pca clustering to day
p3 <- plt$x %>%
    as.data.frame %>%
        ggplot(aes(x = PC1, y = PC2, color = samples$timeAfterStroke, shape = samples$timeAfterStroke)) + 
        geom_point(size = 4) + guides(color = guide_legend("timepoint"), shape = guide_legend("timepoint")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = shapePalette) +
        labs (
            x = paste ("PC1: ", round(var_explained[1,1]*100, 0), "%"), 
            y = paste("PC2: ", round(var_explained[2,1]*100, 0), "%")) +
        ggtitle("PCA clustered to timepoint") + 
        theme_minimal()
# pca clustering to brain region
p4 <- plt$x %>%
    as.data.frame %>%
        ggplot(aes(x = PC1, y = PC2, color = samples$brainRegion, shape = samples$brainRegion)) + 
        geom_point(size = 4) + guides(color = guide_legend("brain region"), shape = guide_legend("brain region")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = shapePalette) +
        labs (
            x = paste ("PC1: ", round(var_explained[1,1]*100, 0), "%"), 
            y = paste("PC2: ", round(var_explained[2,1]*100, 0), "%")) +
        ggtitle("PCA clustered to brain region") + 
        theme_minimal()
# pca clustering to the deterministic factors
p5 <- plt$x %>%
    as.data.frame %>%
        ggplot(aes(x = PC1, y = PC2, color = samples$timeAfterStroke, shape = samples$brainRegion)) + 
        geom_point(size = 4) + 
        guides(color = guide_legend("timepoint"), shape = guide_legend("brain region")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = c(17, 16)) +
        labs (
            x = paste ("PC1: ", round(var_explained[1,1]*100, 0), "%"), 
            y = paste("PC2: ", round(var_explained[2,1]*100, 0), "%")) +
        ggtitle("PCA clustered to timepoint and brain region") + 
        theme_minimal()


# PCA with k-means clustering
set.seed(123)
plt_kmeans <- kmeans(as.data.frame(plt$x[,1:2]), centers = 4, iter.max = 100, nstart = 50)
p6 <- plt$x %>%
    as.data.frame %>%
        ggplot(aes(x = PC1, y = PC2, color = samples$timeAfterStroke, shape = factor(plt_kmeans$cluster))) + 
        geom_point(size = 4) + 
        guides(color = guide_legend("timepoint"), shape = guide_legend("k-means cluster")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = c(15, 16, 17, 18, 19)) +
        ggtitle("PCA with kmeans clustering and timepoint") + 
        labs (x = paste ("PC1: ", round(var_explained[1,1]*100, 0), "%"), 
        y = paste("PC2: ", round(var_explained[2,1]*100, 0), "%")) + 
        theme_minimal()

        
p1 + p2 + p3 + p4 + plot_layout(2,2)

p5 + p6 + plot_layout(1,2)
```
<pre> 



</pre>
### PCA DESeq2
```{r pcaDESeq2, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
plt <- DESeq2::plotPCA(rlog, intgroup = "condition", returnData = TRUE)
colorPalette <- c("#000000", "#E69F00", "#56B4E9", "springgreen2", "#F0E442", "#D55E00", "#CC79A7", "purple4", "aquamarine4" )
shapePalette <- c(15, 16, 17, 18, 19, 8, 4, 9, 7)

# clustering to group
p1 <- ggplot(plt, aes(x = PC1, y = PC2, color = group, shape = group)) +
geom_point(size = 4) +
scale_color_manual(values = colorPalette) +
scale_shape_manual(values = shapePalette) +
guides(color = guide_legend("group"), shape = guide_legend("group")) +
xlab(paste("PC1: ", percent(round(attr(plt, "percentVar")[1], digits = 2)))) + 
ylab(paste("PC2: ", percent(round(attr(plt, "percentVar")[2], digits = 2)))) + 
theme_minimal() +
ggtitle("PCA clustered to group")

# pca clustering to condition
p2 <- ggplot(plt, aes(x = PC1, y = PC2, color = samples$condition, shape = samples$condition)) + 
        geom_point(size = 4) + 
        guides(color = guide_legend("condition"), shape = guide_legend("condition")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = shapePalette) +
        xlab(paste("PC1: ", percent(round(attr(plt, "percentVar")[1], digits = 2)))) + 
        ylab(paste("PC2: ", percent(round(attr(plt, "percentVar")[2], digits = 2)))) +
        ggtitle("PCA clustered to condition") + 
        theme_minimal()
# pca clustering to day
p3 <- ggplot(plt, aes(x = PC1, y = PC2, color = samples$timeAfterStroke, shape = samples$timeAfterStroke)) + 
        geom_point(size = 4) + guides(color = guide_legend("timepoint"), shape = guide_legend("timepoint")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = shapePalette) +
        xlab(paste("PC1: ", percent(round(attr(plt, "percentVar")[1], digits = 2)))) + 
        ylab(paste("PC2: ", percent(round(attr(plt, "percentVar")[2], digits = 2)))) + 
        ggtitle("PCA clustered to timepoint") + 
        theme_minimal()
# pca clustering to brain region
p4 <- ggplot(plt, aes(x = PC1, y = PC2, color = samples$brainRegion, shape = samples$brainRegion)) + 
        geom_point(size = 4) + guides(color = guide_legend("brain region"), shape = guide_legend("brain region")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = shapePalette) +
        xlab(paste("PC1: ", percent(round(attr(plt, "percentVar")[1], digits = 2)))) + 
        ylab(paste("PC2: ", percent(round(attr(plt, "percentVar")[2], digits = 2)))) + 
        ggtitle("PCA clustered to brain region") + 
        theme_minimal()
# pca clustering to the deterministic factors
p5 <- ggplot(plt, aes(x = PC1, y = PC2, color = samples$timeAfterStroke, shape = samples$brainRegion)) + 
        geom_point(size = 4) + 
        guides(color = guide_legend("timepoint"), shape = guide_legend("brain region")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = c(17, 16)) +    
        xlab(paste("PC1: ", percent(round(attr(plt, "percentVar")[1], digits = 2)))) + 
        ylab(paste("PC2: ", percent(round(attr(plt, "percentVar")[2], digits = 2)))) + 
        ggtitle("PCA clustered to timepoint and brain region") + 
        theme_minimal()


# PCA with k-means clustering
set.seed(123)
plt_kmeans <- kmeans(plt[,c("PC1", "PC2")], centers = 4, iter.max = 100, nstart = 50)
p6 <- ggplot(plt, aes(x = PC1, y = PC2, color = samples$timeAfterStroke, shape = factor(plt_kmeans$cluster))) + 
        geom_point(size = 4) + 
        guides(color = guide_legend("timepoint"), shape = guide_legend("k-means cluster")) +
        scale_color_manual(values = colorPalette) +
        scale_shape_manual(values = c(16, 17, 18, 19)) +
        ggtitle("PCA with kmeans clustering and timepoint") + 
        xlab(paste("PC1: ", percent(round(attr(plt, "percentVar")[1], digits = 2)))) + 
        ylab(paste("PC2: ", percent(round(attr(plt, "percentVar")[2], digits = 2)))) + 
        theme_minimal()

p1 + p2 + p3 + p4 + plot_layout(2,2)

p5 + p6 + plot_layout(1,2)

```
<pre>



</pre>
### Heatmap of sample distances
```{r heatmapSamples, message = FALSE, warning = FALSE, results = FALSE, fig.height = 20, fig.width = 20, fig.align = "center"}
head(vst)

colData(vst)$group
ensembl_raw_salmon_dist <- dist(t(assay(vst)), method = "euclidean")
ensembl_raw_salmon_distMatrix <- as.matrix(ensembl_raw_salmon_dist)
rownames(ensembl_raw_salmon_distMatrix) <- colData(vst)$group
colnames(ensembl_raw_salmon_distMatrix) <- colData(vst)$group

paletteLength <- 255
myColor <- colorRampPalette(c("#000099", "white"))(paletteLength)
plt <- pheatmap(ensembl_raw_salmon_distMatrix,
    clustering_distance_rows = ensembl_raw_salmon_dist,
    clustering_distance_cols = ensembl_raw_salmon_dist,
    col = myColor,
    cellheight = 25, 
    cellwidth = 25 
)
plt
dev.off()
```
<pre>



</pre>
### MDS 
```{r mds}
#glimmaMDS(dds)
```
<pre>



</pre>
### Reads assigned to genes
```{r readsAssigned, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
pltCounts <- as.data.frame(colSums(txi$counts))
pltCounts$condition <- samples$condition
pltCounts$sample <- rownames(pltCounts)
pltCounts$sample <- factor(pltCounts$sample, levels = pltCounts$sample)
names(pltCounts)[names(pltCounts) == "colSums(txi$counts)"] <- "counts"
pltCounts_quantiles <- quantile(pltCounts$counts, probs = c(0.05, 0.25, 0.75, 0.95))

ggplot(
    pltCounts,
    aes(x = sample, y = counts),
    col = condition
) +
    geom_bar(
        aes(fill = condition),
        stat = "identity"
    ) +
    geom_hline(
        yintercept = as.numeric(pltCounts_quantiles),
        color = "red"
    ) +
    geom_label_repel(
        data = data.frame(x = 0, y = as.numeric(pltCounts_quantiles)),
        aes(x = x, y = y, label = names(pltCounts_quantiles))
    ) +
    scale_y_continuous(labels = scales::comma) +
    coord_flip() +
    xlab("sample")
```
<pre>



</pre>
### Top mapped features AVERAGE
Piechart showing the number of unmapped reads, ambiguos reads and the proportion of the 5 most highly expressed features across all samples.
```{r MappingInfoTotal, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# ISSUE
total_reads <- sum(as.numeric(gsub(",", "", stats_hisat2[, "reads"])))
total_mapped <- sum(geneCounts)
top_ten <- geneCounts[1:6, , drop = FALSE]
unmapped_reads <- total_reads - total_mapped
remaining_mapped <- total_mapped - sum(top_ten)
pltData <- data.frame(
    row.names = c("unmapped reads", "various", row.names(top_ten)),
    values = mapply(c, unmapped_reads, remaining_mapped, top_ten)
)
pltData <- pltData[order(pltData$values, decreasing = TRUE), , drop = FALSE]
total_reads == sum(pltData)

pltMillion <- ggplot(
    pltData,
    aes(x = "", y = values, fill = row.names(pltData))
) +
    geom_bar(stat = "identity", width = 1, color = "white") +
    coord_polar(theta = "y") +
    geom_label_repel(aes(label = paste(format(round(values / 1e6, 1), trim = TRUE), "M")), position = position_stack(vjust = 0.5)) +
    theme_void() +
    guides(fill = guide_legend(title = element_blank())) +
    scale_fill_discrete(labels = pltData)
pltPercentage <- ggplot(
    pltData,
    aes(x = "", y = values, fill = row.names(pltData))
) +
    geom_bar(stat = "identity", width = 1, color = "white") +
    coord_polar(theta = "y") +
    geom_label_repel(aes(label = paste(format(round((values / sum(pltData$values)) * 100), trim = TRUE), "%")), position = position_stack(vjust = 0.5)) +
    theme_void() +
    guides(fill = guide_legend(title = element_blank()))

print(ggarrange(
    pltMillion,
    pltPercentage,
    labels = "SALMON: count fractions of all samples combined",
    common.legend = TRUE,
    legend = "bottom",
    legend.grob = get_legend(pltMillion),
    ncol = 2,
    nrow = 1
))
```
<pre>



</pre>
### Top mapped features INDIVIDUALLY
Piechart showing the number of unmapped reads, various mapped reads and the fraction of the 5 most highly expressed genes for each sample. 
```{r MappingInfoSample, warning = FALSE, results = FALSE, message = FALSE, fig.height = 5, fig.width = 10, fig.align = "center"}
## ISSUE
counts$gene <- rownames(counts)
for (i in samples$fileName) {
    total_reads <- (as.numeric(gsub(",", "", stats_hisat2[paste("hisat2/", i, sep = ""), "reads"])))
    total_mapped <- as.numeric(sampleCounts[i, ])
    genes_top_count <- as.data.frame(counts[order(counts[, i], decreasing = TRUE), c(i, "gene")][1:5,], header = TRUE)
    genes_top_count_sum <- sum(genes_top_count[,i])
    unmapped_reads <- total_reads - total_mapped
    remaining_mapped <- total_mapped - sum(genes_top_count[, i])

    pltData <- data.frame(
        row.names = c("unmapped reads", "various", row.names(genes_top_count)),
        values = c(unmapped_reads, remaining_mapped, genes_top_count[,i])
    )

    total_reads == sum(pltData)
    
    pltMillion <- ggplot(
        pltData,
        aes(x = "", y = values, fill = row.names(pltData))
    ) +
        geom_bar(stat = "identity", width = 1, color = "white") +
        coord_polar(theta = "y") +
        geom_label_repel(aes(label = paste(format(round(values / 1e6, 1), trim = TRUE), "M")), position = position_stack(vjust = 0.5)) +
        theme_void() +
        guides(fill = guide_legend(title = element_blank()))
    pltPercentage <- ggplot(
        pltData,
        aes(x = "", y = values, fill = row.names(pltData))
    ) +
        geom_bar(stat = "identity", width = 1, color = "white") +
        coord_polar(theta = "y") +
        geom_label_repel(aes(label = paste(format(round((values / colSums(pltData)) * 100), trim = TRUE), "%")), position = position_stack(vjust = 0.5)) +
        theme_void() +
        guides(fill = guide_legend(title = element_blank()))

    print(ggarrange(
        pltMillion,
        pltPercentage,
        labels = paste("Sample: ", i, sep = ""),
        common.legend = TRUE,
        legend = "bottom",
        ncol = 2,
        nrow = 1
    ))
}
counts$gene <- NULL
```
<pre>



</pre>
## Differential Gene Expression
### The Amount of significant Features across results
```{r, overviewSigResults, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
pltSigFeatures <- data.frame(
    results = c(
    "SIGtrt3penVSctrl3pen",
    "SIGtrt7penVSctrl7pen",
    "SIGtrt7coreVSctrl7core",
    "SIGtrt14penVSctrl14pen",
    "SIGctrl7penVSctrl3pen",
    "SIGctrl14penVSctrl3pen",
    "SIGctrl14penVSctrl7pen",
    "SIGtrt7penVStrt3pen",
    "SIGtrt14penVStrt3pen",
    "SIGtrt14penVStrt7pen",
    "SIGsham3penVSctrl3pen",
    "SIGsham3penVStrt3pen",
    "SIGsham3penVSctrl7pen",
    "SIGsham3penVStrt7pen",
    "SIGsham3penVSctrl7core",
    "SIGsham3penVStrt7core",
    "SIGsham3penVSctrl14pen",
    "SIGsham3penVStrt14pen",
    "SIGctrl7coreVSctrl7pen",
    "SIGtrt7coreVStrt7pen",
    "SIGtimeInteraction",
    "SIGtimeReduced"  
    ),
    sigFeatures = c(
    nrow(SIGtrt3penVSctrl3pen),
    nrow(SIGtrt7penVSctrl7pen),
    nrow(SIGtrt7coreVSctrl7core),
    nrow(SIGtrt14penVSctrl14pen),
    nrow(SIGctrl7penVSctrl3pen),
    nrow(SIGctrl14penVSctrl3pen),
    nrow(SIGctrl14penVSctrl7pen),
    nrow(SIGtrt7penVStrt3pen),
    nrow(SIGtrt14penVStrt3pen),
    nrow(SIGtrt14penVStrt7pen),
    nrow(SIGsham3penVSctrl3pen),
    nrow(SIGsham3penVStrt3pen),
    nrow(SIGsham3penVSctrl7pen),
    nrow(SIGsham3penVStrt7pen),
    nrow(SIGsham3penVSctrl7core),
    nrow(SIGsham3penVStrt7core),
    nrow(SIGsham3penVSctrl14pen),
    nrow(SIGsham3penVStrt14pen),
    nrow(SIGctrl7coreVSctrl7pen),
    nrow(SIGtrt7coreVStrt7pen),
    nrow(SIGtimeInteraction),
    nrow(SIGtimeReduced)
    ))



pltSigFeatures$results <- factor(pltSigFeatures$results, levels = pltSigFeatures$results)


ggplot(pltSigFeatures, aes(x = results, y = sigFeatures)) + 
geom_bar(stat = "identity") + 
geom_text(aes(label = sigFeatures), vjust = -0.5) + 
theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
```
<pre>



</pre>
### Volcano plots 
Scatterplot showing statistical significance [-log10(padj)] versus magnitude of change [log2FoldChange].
```{r volcano, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
counter <- 1

resultsList <- list(
    "trt3penVSctrl3pen" = trt3penVSctrl3pen,
    "trt7penVSctrl7pen" = trt7penVSctrl7pen,
    "trt7coreVSctrl7core" = trt7coreVSctrl7core,
    "trt14penVSctrl14pen" = trt14penVSctrl14pen,
    "ctrl7penVSctrl3pen" = ctrl7penVSctrl3pen,
    "ctrl14penVSctrl3pen" = ctrl14penVSctrl3pen,
    "ctrl14penVSctrl7pen" = ctrl14penVSctrl7pen,
    "trt7penVStrt3pen" = trt7penVStrt3pen,
    "trt14penVStrt3pen" = trt14penVStrt3pen,
    "trt14penVStrt7pen" = trt14penVStrt7pen,
    "sham3penVSctrl3pen" = sham3penVSctrl3pen,
    "sham3penVStrt3pen" = sham3penVStrt3pen,
    "sham3penVSctrl7pen" = sham3penVSctrl7pen,
    "sham3penVStrt7pen" = sham3penVStrt7pen,
    "sham3penVSctrl7core" = sham3penVSctrl7core,
    "sham3penVStrt7core" = sham3penVStrt7core,
    "sham3penVSctrl14pen" = sham3penVSctrl14pen,
    "sham3penVStrt14pen" = sham3penVStrt14pen,
    "ctrl7coreVSctrl7pen" = ctrl7coreVSctrl7pen,
    "trt7coreVStrt7pen" = trt7coreVStrt7pen,
    "timeInteraction" = timeInteraction,
    "timeReduced" = timeReduced)

for(result in resultsList){
    plot <- as.data.frame(na.omit(result)) %>% dplyr::mutate(
        significant = padj < 0.1, 
        significant = ifelse(significant & log2FoldChange < 0, "Up", significant), 
        significant = ifelse(significant == "TRUE" & log2FoldChange > 0, "Down", significant)
        ) %>%
        ggplot(aes(x = log2FoldChange, y = -log10(padj), color = significant, label = "row.names")) +
        geom_point(size = 2, na.rm = TRUE) +
        theme_minimal() +
        theme(legend.title = element_blank()) +
        geom_vline(xintercept = c(1, -1), col = "gray", linetype = "dashed") +
        geom_hline(yintercept = -log10(.1), col = "gray", linetype = "dashed") +
        scale_color_manual(
            values = c("#990000", "grey80", "#000099"),
            labels = c("sig Up", "no sig", "sig Down")) +
        geom_text_repel(
            aes(x = log2FoldChange,
                y = -log10(padj),
                label = ifelse(-log10(padj) > quantile(-log10(padj), probs = 0.998) & abs(log2FoldChange) > 1, as.character(rownames(as.data.frame(result))), "")),
            hjust = 0,
            vjust = -1.2,
            segment.color = NA) + 
        xlim((0-max(abs(result$log2FoldChange))),(0+max(abs(result$log2FoldChange)))) +
        ggtitle(names(resultsList)[counter]) 
    
    print(plot)
    counter <- counter + 1
}

```
<pre> 




</pre> 
### Heatmap padj<0.1 clustered
The normalized expression of our candidate genes across all samples plotted in a heatmap.
```{r heatmapClustered, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#plotData1 <- assay(vst)[rownames(SIGtrt3penVSctrl3pen), ]
#plotData1 <- plotData1 - rowMeans(plotData1)
#paletteLength <- 50
#myColor <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
#myBreaks <- c(
#    seq(max(abs(plotData1))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
#    seq(max(abs(plotData1)) / paletteLength, max(abs(plotData1)), length.out = floor(paletteLength / 2)))
#annotation_col <- data.frame(
#    "condition" = samples[, "condition", drop = FALSE], 
#    "timepoint" = samples[, "timepoint"], 
#    "brainRegion" = samples[, "brainRegion"]
#    ) 
#
#pltNoSham <- function(){
#    pltNoSham <- pheatmap(
#    plotData1[, !grepl("sham", samples$condition)],
#    color = myColor,
#    breaks = myBreaks,
#    annotation_col = annotation_col[!grepl("sham", annotation_col$condition),],
#    annotation_colors = list(
#        condition = c(ctrl = "black", "4Rx" = "white"), 
#        timepoint = c(day3 = "red", day7 = "green", day14 = "blue")),
#    show_colnames = FALSE,
#    scale = "row", 
#    silent = TRUE, 
#    cellheight = 50, 
#    cellwidth = 25 )
#    rowOrder <<- pltNoSham$tree_row$order # rather than returning we <<- add the variable to the workspace
#    return(pltNoSham) }
#pltSham <- function(){
#    pltSham <- pheatmap(
#    plotData1[, grepl("sham", samples$condition)][rowOrder, ],
#    color = myColor,
#    breaks = myBreaks,
#    annotation_col = samples[, "condition", drop = FALSE],
#    annotation_colors = list(condition = c(sham = "grey60")),
#    show_colnames = FALSE,
#    cluster_rows = FALSE, 
#    silent = TRUE, 
#    cellheight = 50, 
#    cellwidth = 25)
#    return(pltSham) }
#plot <- ggarrange(
#    as.ggplot(pltNoSham()),
#    as.ggplot(pltSham()),
#    common.legend = TRUE,
#    legend = "bottom",
#    ncol = 2,
#    nrow = 1,
#    widths = c(2, 1)) 
#annotate_figure(plot, top = text_grob("Expression of the set SIGtrt3penVSctrl3pen", 
#            color = "black", face = "bold", size = 14))
```
<pre>



</pre>
### Barplot up/down
This plot summarizes the number of significantly up and down-regulated genes based on different padj and log2FoldChange thresholds (comparison control vs treatment, control is always the base level of the comparison).
```{r barplotUpDown, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
counter <- 1

resultsList <- list(
    "trt3penVSctrl3pen" = trt3penVSctrl3pen,
    "trt7penVSctrl7pen" = trt7penVSctrl7pen,
    "trt7coreVSctrl7core" = trt7coreVSctrl7core,
    "trt14penVSctrl14pen" = trt14penVSctrl14pen,
    "ctrl7penVSctrl3pen" = ctrl7penVSctrl3pen,
    "ctrl14penVSctrl3pen" = ctrl14penVSctrl3pen,
    "ctrl14penVSctrl7pen" = ctrl14penVSctrl7pen,
    "trt7penVStrt3pen" = trt7penVStrt3pen,
    "trt14penVStrt3pen" = trt14penVStrt3pen,
    "trt14penVStrt7pen" = trt14penVStrt7pen,
    "sham3penVSctrl3pen" = sham3penVSctrl3pen,
    "sham3penVStrt3pen" = sham3penVStrt3pen,
    "sham3penVSctrl7pen" = sham3penVSctrl7pen,
    "sham3penVStrt7pen" = sham3penVStrt7pen,
    "sham3penVSctrl7core" = sham3penVSctrl7core,
    "sham3penVStrt7core" = sham3penVStrt7core,
    "sham3penVSctrl14pen" = sham3penVSctrl14pen,
    "sham3penVStrt14pen" = sham3penVStrt14pen,
    "ctrl7coreVSctrl7pen" = ctrl7coreVSctrl7pen,
    "trt7coreVStrt7pen" = trt7coreVStrt7pen,
    "timeInteraction" = timeInteraction,
    "timeReduced" = timeReduced
    )



for(result in resultsList){
    filterUp <- function(result, log2fc = 1, p = 0.1) {
        nrow(result[result$log2FoldChange >= log2fc & !is.na(result$padj) & result$padj <= p, ])}
    filterDown <- function(result, log2fc = 1, p = 0.1) {
        nrow(result[result$log2FoldChange <= -log2fc & !is.na(result$padj) & result$padj <= p, ])}
    pVals <- c(0.001, 0.01, 0.05, 0.1)
    fcVals <- c(0:(max(result$log2FoldChange) + 1))
    summaryUpDown <- do.call(rbind, lapply(pVals, function(p) {
        do.call(rbind, lapply(fcVals, function(f) {
            up <- filterUp(result, f, p)
            down <- filterDown(result, f, p)
            return(data.frame(
                "log2FoldChange" = f, "padj" = p,
                "upRegulated" = up, "downRegulated" = down
            ))
        }))
    }))
    dataUpDown <- melt(summaryUpDown, id.vars = c("log2FoldChange", "padj"))

    print(ggplot(
        dataUpDown,
        aes(x = log2FoldChange, y = value)) +
    geom_bar(
        aes(fill = variable),
        stat = "identity",
        position = "dodge") +
    facet_grid(~padj) +
    scale_fill_manual(values = c("upRegulated" = "#990000", "downRegulated" = "#000099")) +
    theme(legend.position = "bottom", legend.title = element_blank()) +
    labs(
        title = names(resultsList)[counter],
        subtitle = "Number of differentially up/down regulated genes based on different padj values and log2FoldChange cut-off values") +
    ylab("gene count"))
    counter <- counter + 1
}

```
<pre>



<pre>
## Interaction time:condition
<font size = "4">**The normalized counts of genes that significantly differently affected by time between ctrl and trt are plotted.**</font>
### Features individually
```{r timeconditionInteractionSeparated, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
pltData <- data.frame(t(counts(ddsTime, normalized = TRUE)[rownames(SIGtimeInteraction),]))
pltData$timepoint <- samplesTime$timepoint
pltData$condition <- samplesTime$condition
pltData$group <- samplesTime$group

pltData <- melt(pltData, id.vars = c("timepoint", "condition", "group"))
pltMeans <- aggregate(pltData$value, list(pltData$group, pltData$variable), FUN = mean)
pltData <- merge(pltData, pltMeans, by.x = c("group", "variable"), by.y = c("Group.1", "Group.2"))
names(pltData) <- c("group", "variable", "timepoint", "condition", "value", "means")
pltData$timepoint <- factor(pltData$timepoint, levels = c("day3", "day7", "day14"))
pltData <- arrange(pltData, variable, timepoint)


ggplot(pltData, aes(x = timepoint, y = value, color = condition)) + 
geom_point() + 
geom_line(aes(x = timepoint, y = means, group = condition, color = condition)) +
scale_y_log10() +
facet_wrap(~variable) 
```
<pre>




</pre>
### Features together
<font size = "4">**The normalized counts of genes that significantly differently affected by time between ctrl and trt are plotted.**</font>
```{r timeconditionInteractionTogether, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
ggplot(pltData, aes(x = timepoint, y = means, group = interaction(variable, condition), color = condition)) + 
geom_line() +
scale_y_log10()
```
<pre>



</pre>
### The Trajectories of the significant genes clustered elbowplot
```{r timeconditionInteractionClusteringElbow, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 8, fig.align = "center"}
## tscR, mfuzz, TMixClust DID NOT WORK FOR ME
## how about classical k-means clustering 
# https://www.r-bloggers.com/2019/10/cluster-multiple-time-series-using-k-means-2/ 
## ON MY TIME INTERACTION DATA significant results rlog transformed
pltData <- data.frame(t(assay(rlogTime)[rownames(SIGtimeInteraction),]))
pltData$timepoint <- samplesTime$timepoint
pltData$condition <- samplesTime$condition
pltData$group <- samplesTime$group

pltData <- melt(pltData, id.vars = c("timepoint", "condition", "group"))
pltMeans <- aggregate(pltData$value, list(pltData$group, pltData$variable), FUN = mean)
pltData <- merge(pltData, pltMeans, by.x = c("group", "variable"), by.y = c("Group.1", "Group.2"))
names(pltData) <- c("group", "variable", "timepoint", "condition", "value", "means")
pltData$timepoint <- factor(pltData$timepoint, levels = c("day3", "day7", "day14"))
pltData <- arrange(pltData, variable, timepoint)

# spreading my data
clstrData <- unique(pltData[,c("variable", "condition", "timepoint", "means")])
clstrData$key <- paste(clstrData$variable, clstrData$condition, sep = "-")
clstrData$variable <- NULL; clstrData$condition <- NULL
clstrData <- data.frame(t(spread(clstrData, key, means)))
colnames(clstrData) <- clstrData[1,]
clstrData <- clstrData[-c(1),]
clstrData <- mutate_all(clstrData, function(x) as.numeric(as.character(x)))
clstrData <- clstrData - rowMeans(clstrData)

# how many clusters would make sense
wss <- map_dbl(1:20, ~{kmeans(clstrData, ., nstart = 50, iter.max = 50)$tot.withinss})
nClusters <- 1:20
pltElbow <- as.data.frame(cbind("nClusters" = nClusters, "wss" = wss))

ggplot(pltElbow) + 
geom_line(aes(y = wss, x = nClusters), colors = '#82518c') 

```
<pre>



</pre>
### The Trajectories of the significant genes clustered
```{r timeconditionInteractionClustering, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 16, fig.align = "center"}
## clustering my data with k-means
clusters <- kmeans(clstrData, centers = 6, nstart = 100, iter.max = 50)
centers <- rownames_to_column(as.data.frame(clusters$centers), "cluster")
clusters$centers <- clusters$centers[,1:3]  # if clusters$centers is broken again 
clstrData <- clstrData %>% mutate(cluster = clusters$cluster)


# making my table long again for plotting
clustrDataLong <- clstrData %>% 
  mutate(geneCond = rownames(clstrData)) %>% 
  pivot_longer(cols = c(-cluster, -geneCond), names_to = "timepoint", values_to = "mean") %>%
  mutate(mean = as.numeric(mean))
clustrDataLong$timepoint <- factor(clustrDataLong$timepoint, levels = c("day3", "day7", "day14"))
clustrDataLong$condition <- ifelse(grepl("4Rx", clustrDataLong$geneCond), "4Rx", "ctrl")


centersLong <- centers %>% 
  pivot_longer(cols = -cluster, names_to = "timepoint", values_to = "mean")

# plotting my data
ggplot() +
  geom_line(data = clustrDataLong, aes(y = mean, x = timepoint, group = geneCond, colour = condition), size = 2) +
  facet_wrap(~cluster, nrow = 3) + 
  scale_color_manual(values = c("#82518c", "#10a7a7")) +
  geom_line(data = centersLong, aes(y = mean, x = timepoint, group = cluster), col = "#b58900", size = 13, alpha = 0.4) +
  theme_dark() + 
  ggtitle("K-means clustering of rlog-transformed trajectories scaled on a gene-condition level")

```
<pre>




</pre>
### The trajectories of all genes clustered elbowplot
```{r timeconditionInteractionClusteringALLElbow, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 8, fig.align = "center"}

#### WITH THE ALL THE GENES FROM TIME INTERACTION MODEL ----------------------------

pltData <- data.frame(t(assay(rlogTime)))
pltData$timepoint <- samplesTime$timepoint
pltData$condition <- samplesTime$condition
pltData$group <- samplesTime$group

pltData <- melt(pltData, id.vars = c("timepoint", "condition", "group"))
pltMeans <- aggregate(pltData$value, list(pltData$group, pltData$variable), FUN = mean)
pltData <- merge(pltData, pltMeans, by.x = c("group", "variable"), by.y = c("Group.1", "Group.2"))
names(pltData) <- c("group", "variable", "timepoint", "condition", "value", "means")
pltData$timepoint <- factor(pltData$timepoint, levels = c("day3", "day7", "day14"))
pltData <- arrange(pltData, variable, timepoint)

# spreading my data
clstrData <- unique(pltData[,c("variable", "condition", "timepoint", "means")])
clstrData$key <- paste(clstrData$variable, clstrData$condition, sep = "-")
clstrData$variable <- NULL; clstrData$condition <- NULL; clstrData$brainRegion <- NULL
clstrData <- data.frame(t(spread(clstrData, key, means)))
colnames(clstrData) <- clstrData[1,]
clstrData <- clstrData[-c(1),]
clstrData <- mutate_all(clstrData, function(x) as.numeric(as.character(x)))
clstrData <- clstrData - rowMeans(clstrData)


# how many clusters would make sense
wss <- map_dbl(1:25, ~{kmeans(clstrData, ., nstart = 50, iter.max = 15)$tot.withinss})
nClusters <- 1:25
pltElbow <- as.data.frame(cbind("nClusters" = nClusters, "wss" = wss))

ggplot(pltElbow) + 
geom_line(aes(y = wss, x = nClusters), colors = '#82518c') + 
theme_dark()
```
<pre>




</pre>
### The trajectories of all genes clustered
```{r timeconditionInteractionClusteringALL, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 16, fig.align = "center"}

## clustering my data with k-means
clusters <- kmeans(clstrData, centers = 16, nstart = 100, iter.max = 50)
centers <- rownames_to_column(as.data.frame(clusters$centers), "cluster")

clstrData <- clstrData %>% mutate(cluster = clusters$cluster)

# making my table long again for plotting

clustrDataLong <- clstrData %>% 
  mutate(geneCond = rownames(clstrData)) %>% 
  pivot_longer(cols = c(-cluster, -geneCond), names_to = "timepoint", values_to = "mean") %>%
  mutate(mean = as.numeric(mean))
clustrDataLong$timepoint <- factor(clustrDataLong$timepoint, levels = c("day3", "day7", "day14"))

centersLong <- centers %>% 
  pivot_longer(cols = -cluster, names_to = "timepoint", values_to = "mean")

# plotting my data
ggplot() +
  geom_line(data = clustrDataLong, aes(y = mean, x = timepoint, group = geneCond), colour = "#82518c") +
  facet_wrap(~cluster, nrow = 4) + 
  geom_line(data = centersLong, aes(y = mean, x = timepoint, group = cluster), col = "#b58900", size = 10, alpha = 0.4) + 
  theme_dark() +
  ggtitle("K-means clustering of all the rlog-transformed scaled (gene and condition) counts of all genes")
```
<pre>



</pre>
## GSEA timeInteraction
### Top Biological Processes GO
```{r top40NES_timeInteraction, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------timeInteraction------------------------------------
timeInteractionGSEADf <- timeInteractionGSEADf[order(timeInteractionGSEADf$NES, decreasing = FALSE),]
timeInteractionGSEADf50 <- timeInteractionGSEADf
timeInteractionGSEADf50$Description <- factor(timeInteractionGSEADf50$Description, levels = timeInteractionGSEADf50$Description)

#### bubble plot for GOs
ggplot(timeInteractionGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

# xpos <- c(1:nrow(timeInteractionGSEADf50))
# ypos <- ifelse(timeInteractionGSEADf50$NES>0,-1.1,1.1)
#
# ggplot(data = timeInteractionGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(timeInteractionGSEADf50$NES)), max(abs(timeInteractionGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_timeInteraction, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
timeInteractionGSEADf <- timeInteractionGSEADf[order(timeInteractionGSEADf$NES, decreasing = TRUE), ]
timeInteraction <- timeInteraction[order(timeInteraction$stat, decreasing = TRUE), ]
tail(timeInteraction)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(timeInteractionGSEADf)), ID = timeInteractionGSEADf$ID, Term = timeInteractionGSEADf$Description, Genes = toupper(timeInteractionGSEADf$core_enrichment), adj_pval = timeInteractionGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(timeInteraction)), logFC = timeInteraction$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- pltPathway
pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(timeInteraction, n = 500)[1:500,], tail(timeInteraction, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.02, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_timeInteraction, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    timeInteractionGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_timeInteraction, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}

GO1 <- strsplit(tail(timeInteractionGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(timeInteractionGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO3 <- strsplit(tail(timeInteractionGSEADf, n = 6)$core_enrichment[3], "/")[[1]][10:14]
GO4 <- strsplit(tail(timeInteractionGSEADf, n = 6)$core_enrichment[4], "/")[[1]][26:30]


GOI <- c(GO1, GO2, GO3, GO4)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


#pltAnno <- data.frame (row.names = GOI, 
#    "GO" = c(
#    rep(paste(tail(timeInteractionGSEADf, n = 6)$Description[1], tail(timeInteractionGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(timeInteractionGSEADf, n = 6)$Description[2], tail(timeInteractionGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(timeInteractionGSEADf, n = 6)$Description[3], tail(timeInteractionGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(timeInteractionGSEADf, n = 6)$Description[4], tail(timeInteractionGSEADf, n = 6)$ID[4], sep = ":"), 5)
#    ))
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_timeInteraction, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(timeInteractionGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(timeInteractionGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(timeInteractionGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(timeInteractionGSEADf, n = 6)$core_enrichment[4], "/")[[1]]




GOI_objects <- list(GO1, GO2, GO3, GO4)

GOI_names <- c("GO1", "GO2", "GO3", "GO4")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(timeInteractionGSEADf, n = 6)$Description[1], tail(timeInteractionGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(timeInteractionGSEADf, n = 6)$Description[2], tail(timeInteractionGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(timeInteractionGSEADf, n = 6)$Description[3], tail(timeInteractionGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(timeInteractionGSEADf, n = 6)$Description[4], tail(timeInteractionGSEADf, n = 6)$ID[4], sep = ":")
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 50,
         cellwidth = 10,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_timeInteraction, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
timeInteractionGSEADf <- timeInteractionGSEADf[order(timeInteractionGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(timeInteractionGSEA, geneSetID = rownames(head(timeInteractionGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
timeInteractionGSEADf <- timeInteractionGSEADf[order(timeInteractionGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(timeInteractionGSEA, geneSetID = rownames(head(timeInteractionGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
timeInteractionGSEADf <- timeInteractionGSEADf[order(timeInteractionGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(timeInteractionGSEA, geneSetID = rownames(head(timeInteractionGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
timeInteractionGSEADf <- timeInteractionGSEADf[order(timeInteractionGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(timeInteractionGSEA, geneSetID = rownames(head(timeInteractionGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA trt3penVSctrl3pen
### Top Biological Processes GO
```{r top40NES, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------trt3penVSctrl3pen------------------------------------
trt3penVSctrl3penGSEADf <- trt3penVSctrl3penGSEADf[order(trt3penVSctrl3penGSEADf$NES, decreasing = FALSE),]
trt3penVSctrl3penGSEADf50 <- rbind(head(trt3penVSctrl3penGSEADf, n = 25), tail(trt3penVSctrl3penGSEADf, n = 25))
trt3penVSctrl3penGSEADf50$Description <- factor(trt3penVSctrl3penGSEADf50$Description, levels = trt3penVSctrl3penGSEADf50$Description)

#### bubble plot for GOs
ggplot(trt3penVSctrl3penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(trt3penVSctrl3penGSEADf50))
#ypos <- ifelse(trt3penVSctrl3penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = trt3penVSctrl3penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(trt3penVSctrl3penGSEADf50$NES)), max(abs(trt3penVSctrl3penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_trt3penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
trt3penVSctrl3penGSEADf <- trt3penVSctrl3penGSEADf[order(trt3penVSctrl3penGSEADf$NES, decreasing = TRUE), ]
trt3penVSctrl3pen <- trt3penVSctrl3pen[order(trt3penVSctrl3pen$stat, decreasing = TRUE), ]

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(trt3penVSctrl3penGSEADf)), ID = trt3penVSctrl3penGSEADf$ID, Term = trt3penVSctrl3penGSEADf$Description, Genes = toupper(trt3penVSctrl3penGSEADf$core_enrichment), adj_pval = trt3penVSctrl3penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(trt3penVSctrl3pen)), logFC = trt3penVSctrl3pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(trt3penVSctrl3penGSEADf)[1:3,], tail(trt3penVSctrl3penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)


pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(trt3penVSctrl3pen, n = 500)[1:500,], tail(trt3penVSctrl3pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.03, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt3penVSctrl3penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt3penVSctrl3penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt3penVSctrl3penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt3penVSctrl3penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4 , fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, trt3penVSctrl3penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

nrow(grps)

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately
set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)
```
<pre>



</pre>
### GO terms within network
<font size = "4">**On pause - will be continued if necessary**</font>
```{r, termsNetwork, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt3penVSctrl3penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt3penVSctrl3penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)


# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt3penVSctrl3penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt3penVSctrl3penGSEA@result)

## visNetwork to pinpoint nodes
ntwk <- toVisNetworkData(gs_ovnet)
#visIgraph(gs_ovnet, idToLabel = TRUE)
colnames(ntwk$nodes)[3] <- "group"

ntwk$nodes[trt3penVSctrl3penGSEA_neuroplasticity$ID,"group"] <- "neuroplasticity"
ntwk$nodes[trt3penVSctrl3penGSEA_neuroplasticity$ID,"SubCategory"] <- neuroplasticity

ntwk$nodes[trt3penVSctrl3penGSEA_inflammation$ID,"group"] <- "inflammation"
ntwk$nodes[trt3penVSctrl3penGSEA_inflammation$ID,"SubCategory"] <- inflammation

ntwk$nodes[trt3penVSctrl3penGSEA_angiogenesis$ID,"group"] <- "angiogenesis"
ntwk$nodes[trt3penVSctrl3penGSEA_angiogenesis$ID,"SubCategory"] <- angiogenesis

visNetwork(nodes = ntwk$nodes, edges = ntwk$edges) %>%
visGroups(groupname = "inflammation", color = "red")

```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    trt3penVSctrl3penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_trt3penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GOIgo0021954 <- strsplit(trt3penVSctrl3penGSEA_neuroplasticity["GO:0021954"]$core_enrichment, "/")[[1]][1:5]
GOIgo0051965 <- strsplit(trt3penVSctrl3penGSEA_neuroplasticity["GO:0051965"]$core_enrichment, "/")[[1]][1:5]
GOIgo0048167 <- strsplit(trt3penVSctrl3penGSEA_neuroplasticity["GO:0048167"]$core_enrichment, "/")[[1]][c(6:9, 14)]
GOIgo0050804 <- strsplit(trt3penVSctrl3penGSEA_neuroplasticity["GO:0050804"]$core_enrichment, "/")[[1]][31:35]

GOIgo1903557 <- strsplit(trt3penVSctrl3penGSEA_inflammation["GO:1903557"]$core_enrichment, "/")[[1]][31:35]
GOIgo0050900 <- strsplit(trt3penVSctrl3penGSEA_inflammation["GO:0050900"]$core_enrichment, "/")[[1]][51:55]
GOIgo0002532 <- strsplit(trt3penVSctrl3penGSEA_inflammation["GO:0002532"]$core_enrichment, "/")[[1]][6:10]
GOIgo0034341 <- strsplit(trt3penVSctrl3penGSEA_inflammation["GO:0034341"]$core_enrichment, "/")[[1]][26:30]

GOIgo0045766 <- strsplit(trt3penVSctrl3penGSEA_angiogenesis["GO:0045766"]$core_enrichment, "/")[[1]][c(1:4, 27)]
GOIgo1904018 <- strsplit(trt3penVSctrl3penGSEA_angiogenesis["GO:1904018"]$core_enrichment, "/")[[1]][51:55]
GOIgo0043297 <- strsplit(trt3penVSctrl3penGSEA_angiogenesis["GO:0043297"]$core_enrichment, "/")[[1]][6:10]
GOIgo0001525 <- strsplit(trt3penVSctrl3penGSEA_angiogenesis["GO:0001525"]$core_enrichment, "/")[[1]][26:30]


GOI <- c(GOIgo0021954, GOIgo0051965, GOIgo0048167, 
    GOIgo0050804, GOIgo1903557, GOIgo0050900, GOIgo0002532, GOIgo0034341, 
    GOIgo0045766, GOIgo1904018, GOIgo0043297, GOIgo0001525)

duplicated(GOI)


pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)



pltAnno <- data.frame (row.names = c(GOI), 
    "GO" = c(
    rep(paste(trt3penVSctrl3penGSEA_neuroplasticity["GO:0021954"]$Description, "GO0021954", sep = ":"), 5), 
    rep(paste(trt3penVSctrl3penGSEA_neuroplasticity["GO:0051965"]$Description, "GO0051965", sep = ":"), 5),
    rep(paste(trt3penVSctrl3penGSEA_neuroplasticity["GO:0048167"]$Description, "GO0048167", sep = ":"), 5),
    rep(paste(trt3penVSctrl3penGSEA_neuroplasticity["GO:0050804"]$Description, "GO0050804", sep = ":"), 5),
    rep(paste(trt3penVSctrl3penGSEA_inflammation["GO:1903557"]$Description, "GO1903557", sep = ":"), 5), 
    rep(paste(trt3penVSctrl3penGSEA_inflammation["GO:0050900"]$Description, "GO0050900", sep = ":"), 5), 
    rep(paste(trt3penVSctrl3penGSEA_inflammation["GO:0002532"]$Description, "GO0002532", sep = ":"), 5), 
    rep(paste(trt3penVSctrl3penGSEA_inflammation["GO:0034341"]$Description, "GO0034341", sep = ":"), 5), 
    rep(paste(trt3penVSctrl3penGSEA_angiogenesis["GO:0045766"]$Description, "GO0045766", sep = ":"), 5), 
    rep(paste(trt3penVSctrl3penGSEA_angiogenesis["GO:1904018"]$Description, "GO1904018", sep = ":"), 5), 
    rep(paste(trt3penVSctrl3penGSEA_angiogenesis["GO:0043297"]$Description, "GO0043297", sep = ":"), 5), 
    rep(paste(trt3penVSctrl3penGSEA_angiogenesis["GO:0001525"]$Description, "GO0001525", sep = ":"), 5) 
    ))



newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GOIgo0021954 <- strsplit(trt3penVSctrl3penGSEA_neuroplasticity["GO:0021954"]$core_enrichment, "/")[[1]]
GOIgo0051965 <- strsplit(trt3penVSctrl3penGSEA_neuroplasticity["GO:0051965"]$core_enrichment, "/")[[1]]
GOIgo0048167 <- strsplit(trt3penVSctrl3penGSEA_neuroplasticity["GO:0048167"]$core_enrichment, "/")[[1]]
GOIgo0050804 <- strsplit(trt3penVSctrl3penGSEA_neuroplasticity["GO:0050804"]$core_enrichment, "/")[[1]]
GOIgo1903557 <- strsplit(trt3penVSctrl3penGSEA_inflammation["GO:1903557"]$core_enrichment, "/")[[1]]
GOIgo0050900 <- strsplit(trt3penVSctrl3penGSEA_inflammation["GO:0050900"]$core_enrichment, "/")[[1]]
GOIgo0002532 <- strsplit(trt3penVSctrl3penGSEA_inflammation["GO:0002532"]$core_enrichment, "/")[[1]]
GOIgo0034341 <- strsplit(trt3penVSctrl3penGSEA_inflammation["GO:0034341"]$core_enrichment, "/")[[1]]
GOIgo0045766 <- strsplit(trt3penVSctrl3penGSEA_angiogenesis["GO:0045766"]$core_enrichment, "/")[[1]]
GOIgo1904018 <- strsplit(trt3penVSctrl3penGSEA_angiogenesis["GO:1904018"]$core_enrichment, "/")[[1]]
GOIgo0043297 <- strsplit(trt3penVSctrl3penGSEA_angiogenesis["GO:0043297"]$core_enrichment, "/")[[1]]
GOIgo0001525 <- strsplit(trt3penVSctrl3penGSEA_angiogenesis["GO:0001525"]$core_enrichment, "/")[[1]]

GOI_names <- c("GOIgo0021954", "GOIgo0051965", "GOIgo0048167", 
    "GOIgo0050804", "GOIgo1903557", "GOIgo0050900", "GOIgo0002532", "GOIgo0034341", 
    "GOIgo0045766", "GOIgo1904018", "GOIgo0043297", "GOIgo0001525")

GOI_objects <- list(GOIgo0021954, GOIgo0051965, GOIgo0048167, 
    GOIgo0050804, GOIgo1903557, GOIgo0050900, GOIgo0002532, GOIgo0034341, 
    GOIgo0045766, GOIgo1904018, GOIgo0043297, GOIgo0001525)

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(trt3penVSctrl3penGSEA_neuroplasticity["GO:0021954"]$Description, "GO0021954", sep = ":"), 
    paste(trt3penVSctrl3penGSEA_neuroplasticity["GO:0051965"]$Description, "GO0051965", sep = ":"),
    paste(trt3penVSctrl3penGSEA_neuroplasticity["GO:0048167"]$Description, "GO0048167", sep = ":"),
    paste(trt3penVSctrl3penGSEA_neuroplasticity["GO:0050804"]$Description, "GO0050804", sep = ":"),
    paste(trt3penVSctrl3penGSEA_inflammation["GO:1903557"]$Description, "GO1903557", sep = ":"), 
    paste(trt3penVSctrl3penGSEA_inflammation["GO:0050900"]$Description, "GO0050900", sep = ":"), 
    paste(trt3penVSctrl3penGSEA_inflammation["GO:0002532"]$Description, "GO0002532", sep = ":"), 
    paste(trt3penVSctrl3penGSEA_inflammation["GO:0034341"]$Description, "GO0034341", sep = ":"), 
    paste(trt3penVSctrl3penGSEA_angiogenesis["GO:0045766"]$Description, "GO0045766", sep = ":"), 
    paste(trt3penVSctrl3penGSEA_angiogenesis["GO:1904018"]$Description, "GO1904018", sep = ":"), 
    paste(trt3penVSctrl3penGSEA_angiogenesis["GO:0043297"]$Description, "GO0043297", sep = ":"), 
    paste(trt3penVSctrl3penGSEA_angiogenesis["GO:0001525"]$Description, "GO0001525", sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 50,
         cellwidth = 10,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
trt3penVSctrl3penGSEADf <- trt3penVSctrl3penGSEADf[order(trt3penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt3penVSctrl3penGSEA, geneSetID = rownames(head(trt3penVSctrl3penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
trt3penVSctrl3penGSEADf <- trt3penVSctrl3penGSEADf[order(trt3penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt3penVSctrl3penGSEA, geneSetID = rownames(head(trt3penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
trt3penVSctrl3penGSEADf <- trt3penVSctrl3penGSEADf[order(trt3penVSctrl3penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(trt3penVSctrl3penGSEA, geneSetID = rownames(head(trt3penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
trt3penVSctrl3penGSEADf <- trt3penVSctrl3penGSEADf[order(trt3penVSctrl3penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(trt3penVSctrl3penGSEA, geneSetID = rownames(head(trt3penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )

pdf("rafadontcheat.pdf")
enrichplot::gseaplot2(trt3penVSctrl3penGSEA, geneSetID = rownames(head(trt3penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
dev.off()
```
<pre> 



</pre>
### Enrichment Score Selection
```{r gseaplotSelection, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
enrichplot::gseaplot2(trt3penVSctrl3penGSEA_neuroplasticity, geneSetID = trt3penVSctrl3penGSEA_neuroplasticity$ID, title = "Pathways Neuroplasticity", pvalue_table = FALSE)
enrichplot::gseaplot2(trt3penVSctrl3penGSEA_inflammation, geneSetID = trt3penVSctrl3penGSEA_inflammation$ID, title = "Pathways Inflammation", pvalue_table = FALSE)
enrichplot::gseaplot2(trt3penVSctrl3penGSEA_angiogenesis, geneSetID = trt3penVSctrl3penGSEA_angiogenesis$ID, title = "Pathways Angiogenesis ", pvalue_table = FALSE)
```
<pre> 



</pre>
## GSEA trt7penVSctrl7pen
### Top Biological Processes GO
```{r top40NES_trt7penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------trt7penVSctrl7pen------------------------------------
trt7penVSctrl7penGSEADf <- trt7penVSctrl7penGSEADf[order(trt7penVSctrl7penGSEADf$NES, decreasing = FALSE),]
trt7penVSctrl7penGSEADf$Description <- factor(trt7penVSctrl7penGSEADf$Description, levels = trt7penVSctrl7penGSEADf$Description)

trt7penVSctrl7penGSEADf["GO:0007631",]


#### bubble plot for GOs
ggplot(trt7penVSctrl7penGSEADf, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(trt7penVSctrl7penGSEADf))
#ypos <- ifelse(trt7penVSctrl7penGSEADf$NES>0,-1.1,1.1)
#
#ggplot(data = trt7penVSctrl7penGSEADf,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(trt7penVSctrl7penGSEADf$NES)), max(abs(trt7penVSctrl7penGSEADf$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)

```
<pre>



<pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_trt7penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
trt7penVSctrl7penGSEADf <- trt7penVSctrl7penGSEADf[order(trt7penVSctrl7penGSEADf$NES, decreasing = TRUE), ]
trt7penVSctrl7pen <- trt7penVSctrl7pen[order(trt7penVSctrl7pen$stat, decreasing = TRUE), ]
tail(trt7penVSctrl7pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(trt7penVSctrl7penGSEADf)), ID = trt7penVSctrl7penGSEADf$ID, Term = trt7penVSctrl7penGSEADf$Description, Genes = toupper(trt7penVSctrl7penGSEADf$core_enrichment), adj_pval = trt7penVSctrl7penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(trt7penVSctrl7pen)), logFC = trt7penVSctrl7pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(trt7penVSctrl7penGSEADf)[1:3,], tail(trt7penVSctrl7penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)

pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(trt7penVSctrl7pen, n = 500)[1:500,], tail(trt7penVSctrl7pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.03, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_trt7penVSctrl7pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt7penVSctrl7penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt7penVSctrl7penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt7penVSctrl7penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt7penVSctrl7penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_trt7penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 4)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_trt7penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, trt7penVSctrl7penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

length(grps)

length(trt7penVSctrl7penGSEADf)
nrow(trt7penVSctrl7penGSEADf)

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = length(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)

```
<pre>



</pre>
### GO hierarchy
```{r GoHierarchySignificance_trt7penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    trt7penVSctrl7penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_trt7penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GOIgo0019882 <- strsplit(trt7penVSctrl7penGSEA["GO:0019882"]$core_enrichment, "/")[[1]][1:5]
GOIgo0007631 <- strsplit(trt7penVSctrl7penGSEA["GO:0007631"]$core_enrichment, "/")[[1]][1:5]
GOIgo1903052 <- strsplit(trt7penVSctrl7penGSEA["GO:1903052"]$core_enrichment, "/")[[1]][11:15]
GOIgo0051348 <- strsplit(trt7penVSctrl7penGSEA["GO:0051348"]$core_enrichment, "/")[[1]][26:30]

GOIgo0140053 <- strsplit(trt7penVSctrl7penGSEA["GO:0140053"]$core_enrichment, "/")[[1]][31:35]
GOIgo0032543 <- strsplit(trt7penVSctrl7penGSEA["GO:0032543"]$core_enrichment, "/")[[1]][16:20]


GOI <- c(GOIgo0019882, GOIgo0007631, GOIgo1903052, 
    GOIgo0051348, GOIgo0140053, GOIgo0032543)


duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


pltAnno <- data.frame (row.names = c(GOI), 
    "GO" = c(
    rep(paste(trt7penVSctrl7penGSEA["GO:0019882"]$Description, "GO0019882", sep = ":"), 5), 
    rep(paste(trt7penVSctrl7penGSEA["GO:0007631"]$Description, "GO0007631", sep = ":"), 5),
    rep(paste(trt7penVSctrl7penGSEA["GO:1903052"]$Description, "GO1903052", sep = ":"), 5),
    rep(paste(trt7penVSctrl7penGSEA["GO:0051348"]$Description, "GO0051348", sep = ":"), 5),
    rep(paste(trt7penVSctrl7penGSEA["GO:0140053"]$Description, "GO0140053", sep = ":"), 5), 
    rep(paste(trt7penVSctrl7penGSEA["GO:0032543"]$Description, "GO0032543", sep = ":"), 5)
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))

p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)
p1

dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_trt7penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GOIgo0019882 <- strsplit(trt7penVSctrl7penGSEA["GO:0019882"]$core_enrichment, "/")[[1]]
GOIgo0007631 <- strsplit(trt7penVSctrl7penGSEA["GO:0007631"]$core_enrichment, "/")[[1]]
GOIgo1903052 <- strsplit(trt7penVSctrl7penGSEA["GO:1903052"]$core_enrichment, "/")[[1]]
GOIgo0051348 <- strsplit(trt7penVSctrl7penGSEA["GO:0051348"]$core_enrichment, "/")[[1]]
GOIgo0140053 <- strsplit(trt7penVSctrl7penGSEA["GO:0140053"]$core_enrichment, "/")[[1]]
GOIgo0032543 <- strsplit(trt7penVSctrl7penGSEA["GO:0032543"]$core_enrichment, "/")[[1]]


GOI_objects <- list(GOIgo0019882, GOIgo0007631, GOIgo1903052, GOIgo0051348, GOIgo0140053, GOIgo0032543)

GOI_names <-  c("GOIgo0019882", "GOIgo0007631", "GOIgo1903052", "GOIgo0051348", "GOIgo0140053", "GOIgo0032543")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(trt7penVSctrl7penGSEA["GO:0019882"]$Description, "GO0019882", sep = ":"), 
    paste(trt7penVSctrl7penGSEA["GO:0007631"]$Description, "GO0007631", sep = ":"),
    paste(trt7penVSctrl7penGSEA["GO:1903052"]$Description, "GO1903052", sep = ":"),
    paste(trt7penVSctrl7penGSEA["GO:0051348"]$Description, "GO0051348", sep = ":"),
    paste(trt7penVSctrl7penGSEA["GO:0140053"]$Description, "GO0140053", sep = ":"), 
    paste(trt7penVSctrl7penGSEA["GO:0032543"]$Description, "GO0032543", sep = ":")
    ))



newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)


p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 50,
         cellwidth = 10,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_trt7penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
trt7penVSctrl7penGSEADf <- trt7penVSctrl7penGSEADf[order(trt7penVSctrl7penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7penVSctrl7penGSEA, geneSetID = rownames(head(trt7penVSctrl7penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
trt7penVSctrl7penGSEADf <- trt7penVSctrl7penGSEADf[order(trt7penVSctrl7penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7penVSctrl7penGSEA, geneSetID = rownames(head(trt7penVSctrl7penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
trt7penVSctrl7penGSEADf <- trt7penVSctrl7penGSEADf[order(trt7penVSctrl7penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(trt7penVSctrl7penGSEA, geneSetID = rownames(head(trt7penVSctrl7penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
trt7penVSctrl7penGSEADf <- trt7penVSctrl7penGSEADf[order(trt7penVSctrl7penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7penVSctrl7penGSEA, geneSetID = rownames(head(trt7penVSctrl7penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA trt14penVSctrl14pen
### Top Biological Processes GO
```{r top40NES_trt14penVSctrl14pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------trt14penVSctrl14pen------------------------------------
trt14penVSctrl14penGSEADf <- trt14penVSctrl14penGSEADf[order(trt14penVSctrl14penGSEADf$NES, decreasing = FALSE),]
trt14penVSctrl14penGSEADf50 <- rbind(head(trt14penVSctrl14penGSEADf, n = 25), tail(trt14penVSctrl14penGSEADf, n = 25))
trt14penVSctrl14penGSEADf50$Description <- factor(trt14penVSctrl14penGSEADf50$Description, levels = trt14penVSctrl14penGSEADf50$Description)

#### bubble plot for GOs
ggplot(trt14penVSctrl14penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(trt14penVSctrl14penGSEADf50))
#ypos <- ifelse(trt14penVSctrl14penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = trt14penVSctrl14penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(trt14penVSctrl14penGSEADf50$NES)), max(abs(trt14penVSctrl14penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_trt14penVSctrl14pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
trt14penVSctrl14penGSEADf <- trt14penVSctrl14penGSEADf[order(trt14penVSctrl14penGSEADf$NES, decreasing = TRUE), ]
trt14penVSctrl14pen <- trt14penVSctrl14pen[order(trt14penVSctrl14pen$stat, decreasing = TRUE), ]
tail(trt14penVSctrl14pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(trt14penVSctrl14penGSEADf)), ID = trt14penVSctrl14penGSEADf$ID, Term = trt14penVSctrl14penGSEADf$Description, Genes = toupper(trt14penVSctrl14penGSEADf$core_enrichment), adj_pval = trt14penVSctrl14penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(trt14penVSctrl14pen)), logFC = trt14penVSctrl14pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(trt14penVSctrl14penGSEADf)[1:3,], tail(trt14penVSctrl14penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)


pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(trt14penVSctrl14pen, n = 500)[1:500,], tail(trt14penVSctrl14pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.03, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_trt14penVSctrl14pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt14penVSctrl14penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt14penVSctrl14penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt14penVSctrl14penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt14penVSctrl14penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_trt14penVSctrl14pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_trt14penVSctrl14pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, trt14penVSctrl14penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately
set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)

```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_trt14penVSctrl14pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    trt14penVSctrl14penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_trt14penVSctrl14pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO3 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][29:33]
GO4 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][10:14]

GO7 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][26:30]
GO9 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][19:23]
GO10 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][20:24]

GOI <- c(GO1, GO2, GO3, GO4, GO7, GO8, GO9, GO10)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


#pltAnno <- data.frame (row.names = GOI, 
#    "GO" = c(
#    rep(paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[1], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[2], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[3], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[4], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[1], head(trt14penVSctrl14penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[2], head(trt14penVSctrl14penGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[3], head(trt14penVSctrl14penGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[4], head(trt14penVSctrl14penGSEADf, n = 6)$ID[4], sep = ":"), 5)
#    ))
#
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_trt14penVSctrl14pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(trt14penVSctrl14penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]



GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[1], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[2], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[3], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[4], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[5], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(trt14penVSctrl14penGSEADf, n = 6)$Description[6], tail(trt14penVSctrl14penGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[1], head(trt14penVSctrl14penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[2], head(trt14penVSctrl14penGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[3], head(trt14penVSctrl14penGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[4], head(trt14penVSctrl14penGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[5], head(trt14penVSctrl14penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(trt14penVSctrl14penGSEADf, n = 6)$Description[6], head(trt14penVSctrl14penGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 50,
         cellwidth = 10,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_trt14penVSctrl14pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
trt14penVSctrl14penGSEADf <- trt14penVSctrl14penGSEADf[order(trt14penVSctrl14penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVSctrl14penGSEA, geneSetID = rownames(head(trt14penVSctrl14penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
trt14penVSctrl14penGSEADf <- trt14penVSctrl14penGSEADf[order(trt14penVSctrl14penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVSctrl14penGSEA, geneSetID = rownames(head(trt14penVSctrl14penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
trt14penVSctrl14penGSEADf <- trt14penVSctrl14penGSEADf[order(trt14penVSctrl14penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(trt14penVSctrl14penGSEA, geneSetID = rownames(head(trt14penVSctrl14penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
trt14penVSctrl14penGSEADf <- trt14penVSctrl14penGSEADf[order(trt14penVSctrl14penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVSctrl14penGSEA, geneSetID = rownames(head(trt14penVSctrl14penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA trt7penVStrt3pen
### Top Biological Processes GO
```{r top40NES_trt7penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------trt7penVStrt3pen------------------------------------
trt7penVStrt3penGSEADf <- trt7penVStrt3penGSEADf[order(trt7penVStrt3penGSEADf$NES, decreasing = FALSE),]
trt7penVStrt3penGSEADf50 <- rbind(head(trt7penVStrt3penGSEADf, n = 25), tail(trt7penVStrt3penGSEADf, n = 25))
trt7penVStrt3penGSEADf50$Description <- factor(trt7penVStrt3penGSEADf50$Description, levels = trt7penVStrt3penGSEADf50$Description)

#### bubble plot for GOs
ggplot(trt7penVStrt3penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(trt7penVStrt3penGSEADf50))
#ypos <- ifelse(trt7penVStrt3penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = trt7penVStrt3penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(trt7penVStrt3penGSEADf50$NES)), max(abs(trt7penVStrt3penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_trt7penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
trt7penVStrt3penGSEADf <- trt7penVStrt3penGSEADf[order(trt7penVStrt3penGSEADf$NES, decreasing = TRUE), ]
trt7penVStrt3pen <- trt7penVStrt3pen[order(trt7penVStrt3pen$stat, decreasing = TRUE), ]
tail(trt7penVStrt3pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(trt7penVStrt3penGSEADf)), ID = trt7penVStrt3penGSEADf$ID, Term = trt7penVStrt3penGSEADf$Description, Genes = toupper(trt7penVStrt3penGSEADf$core_enrichment), adj_pval = trt7penVStrt3penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(trt7penVStrt3pen)), logFC = trt7penVStrt3pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(trt7penVStrt3penGSEADf)[1:3,], tail(trt7penVStrt3penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)


pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(trt7penVStrt3pen, n = 500)[1:500,], tail(trt7penVStrt3pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.02, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_trt7penVStrt3pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt7penVStrt3penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt7penVStrt3penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt7penVStrt3penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt7penVStrt3penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_trt7penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("Significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_trt7penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, trt7penVStrt3penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))


set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)
```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_trt7penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    trt7penVStrt3penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_trt7penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO3 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][40:44]
GO4 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][26:30]
GO5 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]][31:35]
GO6 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]][15:19]

GO7 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO9 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][c(1:4, 27)]
GO10 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][20:24]
GO11 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]][1:5]
GO12 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]][10:14]

GOI <- c(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)

#pltAnno <- data.frame (row.names = c(GOI), 
#    "GO" = c(
#    rep(paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[1], tail(trt7penVStrt3penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[2], tail(trt7penVStrt3penGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[3], tail(trt7penVStrt3penGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[4], tail(trt7penVStrt3penGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[5], tail(trt7penVStrt3penGSEADf, n = 6)$ID[5], sep = ":"), 5), 
#    rep(paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[6], tail(trt7penVStrt3penGSEADf, n = 6)$ID[6], sep = ":"), 5), 
#    rep(paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[1], head(trt7penVStrt3penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[2], head(trt7penVStrt3penGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[3], head(trt7penVStrt3penGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[4], head(trt7penVStrt3penGSEADf, n = 6)$ID[4], sep = ":"), 5), 
#    rep(paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[5], head(trt7penVStrt3penGSEADf, n = 6)$ID[5], sep = ":"), 5), 
#    rep(paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[6], head(trt7penVStrt3penGSEADf, n = 6)$ID[6], sep = ":"), 5) 
#    ))
#
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))

p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)
p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_trt7penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(trt7penVStrt3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]

GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)
GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[1], tail(trt7penVStrt3penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[2], tail(trt7penVStrt3penGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[3], tail(trt7penVStrt3penGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[4], tail(trt7penVStrt3penGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[5], tail(trt7penVStrt3penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(trt7penVStrt3penGSEADf, n = 6)$Description[6], tail(trt7penVStrt3penGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[1], head(trt7penVStrt3penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[2], head(trt7penVStrt3penGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[3], head(trt7penVStrt3penGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[4], head(trt7penVStrt3penGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[5], head(trt7penVStrt3penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(trt7penVStrt3penGSEADf, n = 6)$Description[6], head(trt7penVStrt3penGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 50,
         cellwidth = 10,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_trt7penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
trt7penVStrt3penGSEADf <- trt7penVStrt3penGSEADf[order(trt7penVStrt3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7penVStrt3penGSEA, geneSetID = rownames(head(trt7penVStrt3penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
trt7penVStrt3penGSEADf <- trt7penVStrt3penGSEADf[order(trt7penVStrt3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7penVStrt3penGSEA, geneSetID = rownames(head(trt7penVStrt3penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
trt7penVStrt3penGSEADf <- trt7penVStrt3penGSEADf[order(trt7penVStrt3penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(trt7penVStrt3penGSEA, geneSetID = rownames(head(trt7penVStrt3penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
trt7penVStrt3penGSEADf <- trt7penVStrt3penGSEADf[order(trt7penVStrt3penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7penVStrt3penGSEA, geneSetID = rownames(head(trt7penVStrt3penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA trt14penVStrt3pen
### Top Biological Processes GO
```{r top40NES_trt14penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------trt14penVStrt3pen------------------------------------
trt14penVStrt3penGSEADf <- trt14penVStrt3penGSEADf[order(trt14penVStrt3penGSEADf$NES, decreasing = FALSE),]
trt14penVStrt3penGSEADf50 <- rbind(head(trt14penVStrt3penGSEADf, n = 25), tail(trt14penVStrt3penGSEADf, n = 25))
trt14penVStrt3penGSEADf50$Description <- factor(trt14penVStrt3penGSEADf50$Description, levels = trt14penVStrt3penGSEADf50$Description)

#### bubble plot for GOs
ggplot(trt14penVStrt3penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(trt14penVStrt3penGSEADf50))
#ypos <- ifelse(trt14penVStrt3penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = trt14penVStrt3penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(trt14penVStrt3penGSEADf50$NES)), max(abs(trt14penVStrt3penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_trt14penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
trt14penVStrt3penGSEADf <- trt14penVStrt3penGSEADf[order(trt14penVStrt3penGSEADf$NES, decreasing = TRUE), ]
trt14penVStrt3pen <- trt14penVStrt3pen[order(trt14penVStrt3pen$stat, decreasing = TRUE), ]
tail(trt14penVStrt3pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(trt14penVStrt3penGSEADf)), ID = trt14penVStrt3penGSEADf$ID, Term = trt14penVStrt3penGSEADf$Description, Genes = toupper(trt14penVStrt3penGSEADf$core_enrichment), adj_pval = trt14penVStrt3penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(trt14penVStrt3pen)), logFC = trt14penVStrt3pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(trt14penVStrt3penGSEADf)[1:3,], tail(trt14penVStrt3penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)

pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(trt14penVStrt3pen, n = 500)[1:500,], tail(trt14penVStrt3pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.03, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_trt14penVStrt3pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt14penVStrt3penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt14penVStrt3penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt14penVStrt3penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt14penVStrt3penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_trt14penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_trt14penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, trt14penVStrt3penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately
set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)

```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_trt14penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    trt14penVStrt3penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_trt14penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO3 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][29:33]
GO4 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][20:24]

GO7 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][26:30]
GO9 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][20:24]
GO10 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][20:24]

GOI <- c(GO1, GO2, GO3, GO4, GO7, GO8, GO9, GO10)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


#pltAnno <- data.frame (row.names = GOI, 
#    "GO" = c(
#    rep(paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[1], tail(trt14penVStrt3penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[2], tail(trt14penVStrt3penGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[3], tail(trt14penVStrt3penGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[4], tail(trt14penVStrt3penGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[1], head(trt14penVStrt3penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[2], head(trt14penVStrt3penGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[3], head(trt14penVStrt3penGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[4], head(trt14penVStrt3penGSEADf, n = 6)$ID[4], sep = ":"), 5)
#    ))
#
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)
#
paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_trt14penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(trt14penVStrt3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]



GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[1], tail(trt14penVStrt3penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[2], tail(trt14penVStrt3penGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[3], tail(trt14penVStrt3penGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[4], tail(trt14penVStrt3penGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[5], tail(trt14penVStrt3penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(trt14penVStrt3penGSEADf, n = 6)$Description[6], tail(trt14penVStrt3penGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[1], head(trt14penVStrt3penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[2], head(trt14penVStrt3penGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[3], head(trt14penVStrt3penGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[4], head(trt14penVStrt3penGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[5], head(trt14penVStrt3penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(trt14penVStrt3penGSEADf, n = 6)$Description[6], head(trt14penVStrt3penGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 60,
         cellwidth = 12,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_trt14penVStrt3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
trt14penVStrt3penGSEADf <- trt14penVStrt3penGSEADf[order(trt14penVStrt3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVStrt3penGSEA, geneSetID = rownames(head(trt14penVStrt3penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
trt14penVStrt3penGSEADf <- trt14penVStrt3penGSEADf[order(trt14penVStrt3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVStrt3penGSEA, geneSetID = rownames(head(trt14penVStrt3penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
trt14penVStrt3penGSEADf <- trt14penVStrt3penGSEADf[order(trt14penVStrt3penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(trt14penVStrt3penGSEA, geneSetID = rownames(head(trt14penVStrt3penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
trt14penVStrt3penGSEADf <- trt14penVStrt3penGSEADf[order(trt14penVStrt3penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVStrt3penGSEA, geneSetID = rownames(head(trt14penVStrt3penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA trt14penVStrt7pen
### Top Biological Processes GO
```{r top40NES_trt14penVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------trt14penVStrt7pen------------------------------------
trt14penVStrt7penGSEADf <- trt14penVStrt7penGSEADf[order(trt14penVStrt7penGSEADf$NES, decreasing = FALSE),]
trt14penVStrt7penGSEADf50 <- rbind(head(trt14penVStrt7penGSEADf, n = 25), tail(trt14penVStrt7penGSEADf, n = 25))
trt14penVStrt7penGSEADf50$Description <- factor(trt14penVStrt7penGSEADf50$Description, levels = trt14penVStrt7penGSEADf50$Description)

#### bubble plot for GOs
ggplot(trt14penVStrt7penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(trt14penVStrt7penGSEADf50))
#ypos <- ifelse(trt14penVStrt7penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = trt14penVStrt7penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(trt14penVStrt7penGSEADf50$NES)), max(abs(trt14penVStrt7penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_trt14penVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
trt14penVStrt7penGSEADf <- trt14penVStrt7penGSEADf[order(trt14penVStrt7penGSEADf$NES, decreasing = TRUE), ]
trt14penVStrt7pen <- trt14penVStrt7pen[order(trt14penVStrt7pen$stat, decreasing = TRUE), ]
tail(trt14penVStrt7pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(trt14penVStrt7penGSEADf)), ID = trt14penVStrt7penGSEADf$ID, Term = trt14penVStrt7penGSEADf$Description, Genes = toupper(trt14penVStrt7penGSEADf$core_enrichment), adj_pval = trt14penVStrt7penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(trt14penVStrt7pen)), logFC = trt14penVStrt7pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(trt14penVStrt7penGSEADf)[1:3,], tail(trt14penVStrt7penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)

pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(trt14penVStrt7pen, n = 500)[1:500,], tail(trt14penVStrt7pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.02, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_trt14penVStrt7pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt14penVStrt7penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt14penVStrt7penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt14penVStrt7penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt14penVStrt7penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_trt14penVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_trt14penVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, trt14penVStrt7penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately
set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)

```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_trt14penVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    trt14penVStrt7penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_trt14penVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}

GO1 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO3 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][10:14]
GO4 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][26:30]

GO7 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][11:15]
GO9 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][20:24]
GO10 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][30:34]

GOI <- c(GO1, GO2, GO3, GO4, GO7, GO8, GO9, GO10)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


#pltAnno <- data.frame (row.names = GOI, 
#    "GO" = c(
#    rep(paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[1], tail(trt14penVStrt7penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[2], tail(trt14penVStrt7penGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[3], tail(trt14penVStrt7penGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[4], tail(trt14penVStrt7penGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[1], head(trt14penVStrt7penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[2], head(trt14penVStrt7penGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[3], head(trt14penVStrt7penGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[4], head(trt14penVStrt7penGSEADf, n = 6)$ID[4], sep = ":"), 5)
#    ))
#
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_trt14penVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(trt14penVStrt7penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]



GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[1], tail(trt14penVStrt7penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[2], tail(trt14penVStrt7penGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[3], tail(trt14penVStrt7penGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[4], tail(trt14penVStrt7penGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[5], tail(trt14penVStrt7penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(trt14penVStrt7penGSEADf, n = 6)$Description[6], tail(trt14penVStrt7penGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[1], head(trt14penVStrt7penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[2], head(trt14penVStrt7penGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[3], head(trt14penVStrt7penGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[4], head(trt14penVStrt7penGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[5], head(trt14penVStrt7penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(trt14penVStrt7penGSEADf, n = 6)$Description[6], head(trt14penVStrt7penGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 60,
         cellwidth = 12,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_trt14penVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
trt14penVStrt7penGSEADf <- trt14penVStrt7penGSEADf[order(trt14penVStrt7penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVStrt7penGSEA, geneSetID = rownames(head(trt14penVStrt7penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
trt14penVStrt7penGSEADf <- trt14penVStrt7penGSEADf[order(trt14penVStrt7penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVStrt7penGSEA, geneSetID = rownames(head(trt14penVStrt7penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
trt14penVStrt7penGSEADf <- trt14penVStrt7penGSEADf[order(trt14penVStrt7penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(trt14penVStrt7penGSEA, geneSetID = rownames(head(trt14penVStrt7penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
trt14penVStrt7penGSEADf <- trt14penVStrt7penGSEADf[order(trt14penVStrt7penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(trt14penVStrt7penGSEA, geneSetID = rownames(head(trt14penVStrt7penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre> 



</pre>
## GSEA ctrl7penVSctrl3pen
### Top Biological Processes GO
```{r top40NES_ctrl7penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------ctrl7penVSctrl3pen------------------------------------
ctrl7penVSctrl3penGSEADf <- ctrl7penVSctrl3penGSEADf[order(ctrl7penVSctrl3penGSEADf$NES, decreasing = FALSE),]
ctrl7penVSctrl3penGSEADf50 <- rbind(head(ctrl7penVSctrl3penGSEADf, n = 25), tail(ctrl7penVSctrl3penGSEADf, n = 25))
ctrl7penVSctrl3penGSEADf50$Description <- factor(ctrl7penVSctrl3penGSEADf50$Description, levels = ctrl7penVSctrl3penGSEADf50$Description)

#### bubble plot for GOs
ggplot(ctrl7penVSctrl3penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(ctrl7penVSctrl3penGSEADf50))
#ypos <- ifelse(ctrl7penVSctrl3penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = ctrl7penVSctrl3penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(ctrl7penVSctrl3penGSEADf50$NES)), max(abs(ctrl7penVSctrl3penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_ctrl7penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
ctrl7penVSctrl3penGSEADf <- ctrl7penVSctrl3penGSEADf[order(ctrl7penVSctrl3penGSEADf$NES, decreasing = TRUE), ]
ctrl7penVSctrl3pen <- ctrl7penVSctrl3pen[order(ctrl7penVSctrl3pen$stat, decreasing = TRUE), ]
tail(ctrl7penVSctrl3pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(ctrl7penVSctrl3penGSEADf)), ID = ctrl7penVSctrl3penGSEADf$ID, Term = ctrl7penVSctrl3penGSEADf$Description, Genes = toupper(ctrl7penVSctrl3penGSEADf$core_enrichment), adj_pval = ctrl7penVSctrl3penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(ctrl7penVSctrl3pen)), logFC = ctrl7penVSctrl3pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(ctrl7penVSctrl3penGSEADf)[1:3,], tail(ctrl7penVSctrl3penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)

pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(ctrl7penVSctrl3pen, n = 500)[1:500,], tail(ctrl7penVSctrl3pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.02, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_ctrl7penVSctrl3pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(ctrl7penVSctrl3penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(ctrl7penVSctrl3penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- ctrl7penVSctrl3penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(ctrl7penVSctrl3penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_ctrl7penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_ctrl7penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, ctrl7penVSctrl3penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately
set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)

```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_ctrl7penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    ctrl7penVSctrl3penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_ctrl7penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}

GO1 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO3 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][10:14]
GO4 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][1:5]

GO7 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][30:34]
GO9 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][1:5]
GO10 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][40:44]

GOI <- c(GO1, GO2, GO3, GO4, GO7, GO8, GO9, GO10)
duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


#pltAnno <- data.frame (row.names = GOI, 
#    "GO" = c(
#    rep(paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[1], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[2], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[3], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[4], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[1], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[2], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[3], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[4], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[4], sep = ":"), 5)
#    ))
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_ctrl7penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(ctrl7penVSctrl3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]



GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[1], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[2], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[3], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[4], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[5], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(ctrl7penVSctrl3penGSEADf, n = 6)$Description[6], tail(ctrl7penVSctrl3penGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[1], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[2], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[3], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[4], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[5], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(ctrl7penVSctrl3penGSEADf, n = 6)$Description[6], head(ctrl7penVSctrl3penGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 60,
         cellwidth = 12,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_ctrl7penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
ctrl7penVSctrl3penGSEADf <- ctrl7penVSctrl3penGSEADf[order(ctrl7penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl7penVSctrl3penGSEA, geneSetID = rownames(head(ctrl7penVSctrl3penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
ctrl7penVSctrl3penGSEADf <- ctrl7penVSctrl3penGSEADf[order(ctrl7penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl7penVSctrl3penGSEA, geneSetID = rownames(head(ctrl7penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
ctrl7penVSctrl3penGSEADf <- ctrl7penVSctrl3penGSEADf[order(ctrl7penVSctrl3penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(ctrl7penVSctrl3penGSEA, geneSetID = rownames(head(ctrl7penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
ctrl7penVSctrl3penGSEADf <- ctrl7penVSctrl3penGSEADf[order(ctrl7penVSctrl3penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl7penVSctrl3penGSEA, geneSetID = rownames(head(ctrl7penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA ctrl14penVSctrl3pen
### Top Biological Processes GO
```{r top40NES_ctrl14penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------ctrl14penVSctrl3pen------------------------------------
ctrl14penVSctrl3penGSEADf <- ctrl14penVSctrl3penGSEADf[order(ctrl14penVSctrl3penGSEADf$NES, decreasing = FALSE),]
ctrl14penVSctrl3penGSEADf50 <- rbind(head(ctrl14penVSctrl3penGSEADf, n = 25), tail(ctrl14penVSctrl3penGSEADf, n = 25))
ctrl14penVSctrl3penGSEADf50$Description <- factor(ctrl14penVSctrl3penGSEADf50$Description, levels = ctrl14penVSctrl3penGSEADf50$Description)

#### bubble plot for GOs
ggplot(ctrl14penVSctrl3penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(ctrl14penVSctrl3penGSEADf50))
#ypos <- ifelse(ctrl14penVSctrl3penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = ctrl14penVSctrl3penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(ctrl14penVSctrl3penGSEADf50$NES)), max(abs(ctrl14penVSctrl3penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_ctrl14penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
ctrl14penVSctrl3penGSEADf <- ctrl14penVSctrl3penGSEADf[order(ctrl14penVSctrl3penGSEADf$NES, decreasing = TRUE), ]
ctrl14penVSctrl3pen <- ctrl14penVSctrl3pen[order(ctrl14penVSctrl3pen$stat, decreasing = TRUE), ]
tail(ctrl14penVSctrl3pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(ctrl14penVSctrl3penGSEADf)), ID = ctrl14penVSctrl3penGSEADf$ID, Term = ctrl14penVSctrl3penGSEADf$Description, Genes = toupper(ctrl14penVSctrl3penGSEADf$core_enrichment), adj_pval = ctrl14penVSctrl3penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(ctrl14penVSctrl3pen)), logFC = ctrl14penVSctrl3pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(ctrl14penVSctrl3penGSEADf)[1:3,], tail(ctrl14penVSctrl3penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)

pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(ctrl14penVSctrl3pen, n = 500)[1:500,], tail(ctrl14penVSctrl3pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.02, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_ctrl14penVSctrl3pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(ctrl14penVSctrl3penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(ctrl14penVSctrl3penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- ctrl14penVSctrl3penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(ctrl14penVSctrl3penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_ctrl14penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_ctrl14penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, ctrl14penVSctrl3penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately
set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)

```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_ctrl14penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    ctrl14penVSctrl3penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_ctrl14penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}

GO1 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][10:14]
GO3 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][10:14]
GO4 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][24:28]

GO7 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][11:15]
GO9 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][20:24]
GO10 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][11:15]

GOI <- c(GO1, GO2, GO3, GO4, GO7, GO8, GO9, GO10)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


#pltAnno <- data.frame (row.names = GOI, 
#    "GO" = c(
#    rep(paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[1], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[2], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[3], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[4], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[1], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[2], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[3], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[4], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[4], sep = ":"), 5)
#    ))
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_ctrl14penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(ctrl14penVSctrl3penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]



GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[1], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[2], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[3], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[4], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[5], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(ctrl14penVSctrl3penGSEADf, n = 6)$Description[6], tail(ctrl14penVSctrl3penGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[1], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[2], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[3], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[4], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[5], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(ctrl14penVSctrl3penGSEADf, n = 6)$Description[6], head(ctrl14penVSctrl3penGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 60,
         cellwidth = 12,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_ctrl14penVSctrl3pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
ctrl14penVSctrl3penGSEADf <- ctrl14penVSctrl3penGSEADf[order(ctrl14penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl14penVSctrl3penGSEA, geneSetID = rownames(head(ctrl14penVSctrl3penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
ctrl14penVSctrl3penGSEADf <- ctrl14penVSctrl3penGSEADf[order(ctrl14penVSctrl3penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl14penVSctrl3penGSEA, geneSetID = rownames(head(ctrl14penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
ctrl14penVSctrl3penGSEADf <- ctrl14penVSctrl3penGSEADf[order(ctrl14penVSctrl3penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(ctrl14penVSctrl3penGSEA, geneSetID = rownames(head(ctrl14penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
ctrl14penVSctrl3penGSEADf <- ctrl14penVSctrl3penGSEADf[order(ctrl14penVSctrl3penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl14penVSctrl3penGSEA, geneSetID = rownames(head(ctrl14penVSctrl3penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA ctrl14penVSctrl7pen
### Top Biological Processes GO
```{r top40NES_ctrl14penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------ctrl14penVSctrl7pen------------------------------------
ctrl14penVSctrl7penGSEADf <- ctrl14penVSctrl7penGSEADf[order(ctrl14penVSctrl7penGSEADf$NES, decreasing = FALSE),]
ctrl14penVSctrl7penGSEADf50 <- rbind(head(ctrl14penVSctrl7penGSEADf, n = 25), tail(ctrl14penVSctrl7penGSEADf, n = 25))
ctrl14penVSctrl7penGSEADf50$Description <- factor(ctrl14penVSctrl7penGSEADf50$Description, levels = ctrl14penVSctrl7penGSEADf50$Description)

#### bubble plot for GOs
ggplot(ctrl14penVSctrl7penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(ctrl14penVSctrl7penGSEADf50))
#ypos <- ifelse(ctrl14penVSctrl7penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = ctrl14penVSctrl7penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(ctrl14penVSctrl7penGSEADf50$NES)), max(abs(ctrl14penVSctrl7penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_ctrl14penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
ctrl14penVSctrl7penGSEADf <- ctrl14penVSctrl7penGSEADf[order(ctrl14penVSctrl7penGSEADf$NES, decreasing = TRUE), ]
ctrl14penVSctrl7pen <- ctrl14penVSctrl7pen[order(ctrl14penVSctrl7pen$stat, decreasing = TRUE), ]
tail(ctrl14penVSctrl7pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(ctrl14penVSctrl7penGSEADf)), ID = ctrl14penVSctrl7penGSEADf$ID, Term = ctrl14penVSctrl7penGSEADf$Description, Genes = toupper(ctrl14penVSctrl7penGSEADf$core_enrichment), adj_pval = ctrl14penVSctrl7penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(ctrl14penVSctrl7pen)), logFC = ctrl14penVSctrl7pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(ctrl14penVSctrl7penGSEADf)[1:3,], tail(ctrl14penVSctrl7penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)

pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(ctrl14penVSctrl7pen, n = 500)[1:500,], tail(ctrl14penVSctrl7pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.02, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_ctrl14penVSctrl7pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(ctrl14penVSctrl7penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(ctrl14penVSctrl7penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- ctrl14penVSctrl7penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(ctrl14penVSctrl7penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_ctrl14penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_ctrl14penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, ctrl14penVSctrl7penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately
set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)

```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_ctrl14penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    ctrl14penVSctrl7penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_ctrl14penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}

GO1 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][10:14]
GO3 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][9:13]
GO4 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][20:24]

GO7 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][11:15]
GO9 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][30:34]
GO10 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][11:15]

GOI <- c(GO1, GO2, GO3, GO4, GO7, GO8, GO9, GO10)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


#pltAnno <- data.frame (row.names = GOI, 
#    "GO" = c(
#    rep(paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[1], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[2], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[3], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[4], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[1], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[2], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[3], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[4], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[4], sep = ":"), 5)
#    ))
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_ctrl14penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(ctrl14penVSctrl7penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]



GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[1], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[2], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[3], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[4], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[5], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(ctrl14penVSctrl7penGSEADf, n = 6)$Description[6], tail(ctrl14penVSctrl7penGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[1], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[2], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[3], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[4], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[5], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(ctrl14penVSctrl7penGSEADf, n = 6)$Description[6], head(ctrl14penVSctrl7penGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 60,
         cellwidth = 12,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_ctrl14penVSctrl7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
ctrl14penVSctrl7penGSEADf <- ctrl14penVSctrl7penGSEADf[order(ctrl14penVSctrl7penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl14penVSctrl7penGSEA, geneSetID = rownames(head(ctrl14penVSctrl7penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
ctrl14penVSctrl7penGSEADf <- ctrl14penVSctrl7penGSEADf[order(ctrl14penVSctrl7penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl14penVSctrl7penGSEA, geneSetID = rownames(head(ctrl14penVSctrl7penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
ctrl14penVSctrl7penGSEADf <- ctrl14penVSctrl7penGSEADf[order(ctrl14penVSctrl7penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(ctrl14penVSctrl7penGSEA, geneSetID = rownames(head(ctrl14penVSctrl7penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
ctrl14penVSctrl7penGSEADf <- ctrl14penVSctrl7penGSEADf[order(ctrl14penVSctrl7penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(ctrl14penVSctrl7penGSEA, geneSetID = rownames(head(ctrl14penVSctrl7penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA trt7coreVSctrl7core
### Top Biological Processes GO
```{r top40NES_trt7coreVSctrl7core, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------trt7coreVSctrl7core------------------------------------
trt7coreVSctrl7coreGSEADf <- trt7coreVSctrl7coreGSEADf[order(trt7coreVSctrl7coreGSEADf$NES, decreasing = FALSE),]
trt7coreVSctrl7coreGSEADf50 <- rbind(head(trt7coreVSctrl7coreGSEADf, n = 25), tail(trt7coreVSctrl7coreGSEADf, n = 25))
trt7coreVSctrl7coreGSEADf50$Description <- factor(trt7coreVSctrl7coreGSEADf50$Description, levels = trt7coreVSctrl7coreGSEADf50$Description)

#### bubble plot for GOs
ggplot(trt7coreVSctrl7coreGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(trt7coreVSctrl7coreGSEADf50))
#ypos <- ifelse(trt7coreVSctrl7coreGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = trt7coreVSctrl7coreGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(trt7coreVSctrl7coreGSEADf50$NES)), max(abs(trt7coreVSctrl7coreGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_trt7coreVSctrl7core, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
trt7coreVSctrl7coreGSEADf <- trt7coreVSctrl7coreGSEADf[order(trt7coreVSctrl7coreGSEADf$NES, decreasing = TRUE), ]
trt7coreVSctrl7core <- trt7coreVSctrl7core[order(trt7coreVSctrl7core$stat, decreasing = TRUE), ]
tail(trt7coreVSctrl7core)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(trt7coreVSctrl7coreGSEADf)), ID = trt7coreVSctrl7coreGSEADf$ID, Term = trt7coreVSctrl7coreGSEADf$Description, Genes = toupper(trt7coreVSctrl7coreGSEADf$core_enrichment), adj_pval = trt7coreVSctrl7coreGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(trt7coreVSctrl7core)), logFC = trt7coreVSctrl7core$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(trt7coreVSctrl7coreGSEADf)[1:3,], tail(trt7coreVSctrl7coreGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)

pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(trt7coreVSctrl7core, n = 500)[1:500,], tail(trt7coreVSctrl7core, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.02, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_trt7coreVSctrl7core, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt7coreVSctrl7coreGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt7coreVSctrl7coreGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt7coreVSctrl7coreGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt7coreVSctrl7coreGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_trt7coreVSctrl7core, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("Significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_trt7coreVSctrl7core, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, trt7coreVSctrl7coreGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately

set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)
```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_trt7coreVSctrl7core, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    trt7coreVSctrl7coreGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_trt7coreVSctrl7core, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO3 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[3], "/")[[1]][8:12]
GO4 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[4], "/")[[1]][26:30]
GO5 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[5], "/")[[1]][1:5]
GO6 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[6], "/")[[1]][15:19]

GO7 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO9 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[3], "/")[[1]][c(1:4, 20)]
GO10 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[4], "/")[[1]][2:6]
GO11 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[5], "/")[[1]][20:24]
GO12 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[6], "/")[[1]][15:19]

GOI <- c(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)

#pltAnno <- data.frame (row.names = c(GOI), 
#    "GO" = c(
#    rep(paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[1], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[2], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[3], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[4], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[5], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[5], sep = ":"), 5), 
#    rep(paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[6], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[6], sep = ":"), 5), 
#    rep(paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[1], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[2], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[3], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[4], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[4], sep = ":"), 5), 
#    rep(paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[5], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[5], sep = ":"), 5), 
#    rep(paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[6], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[6], sep = ":"), 5) 
#    ))
#
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))

p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_trt7coreVSctrl7core, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(trt7coreVSctrl7coreGSEADf, n = 6)$core_enrichment[6], "/")[[1]]



GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[1], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[2], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[3], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[4], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[5], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(trt7coreVSctrl7coreGSEADf, n = 6)$Description[6], tail(trt7coreVSctrl7coreGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[1], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[2], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[3], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[4], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[5], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(trt7coreVSctrl7coreGSEADf, n = 6)$Description[6], head(trt7coreVSctrl7coreGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 50,
         cellwidth = 10,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_trt7coreVSctrl7core, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
trt7coreVSctrl7coreGSEADf <- trt7coreVSctrl7coreGSEADf[order(trt7coreVSctrl7coreGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7coreVSctrl7coreGSEA, geneSetID = rownames(head(trt7coreVSctrl7coreGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
trt7coreVSctrl7coreGSEADf <- trt7coreVSctrl7coreGSEADf[order(trt7coreVSctrl7coreGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7coreVSctrl7coreGSEA, geneSetID = rownames(head(trt7coreVSctrl7coreGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
trt7coreVSctrl7coreGSEADf <- trt7coreVSctrl7coreGSEADf[order(trt7coreVSctrl7coreGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(trt7coreVSctrl7coreGSEA, geneSetID = rownames(head(trt7coreVSctrl7coreGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
trt7coreVSctrl7coreGSEADf <- trt7coreVSctrl7coreGSEADf[order(trt7coreVSctrl7coreGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7coreVSctrl7coreGSEA, geneSetID = rownames(head(trt7coreVSctrl7coreGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
<pre>



</pre>
## GSEA trt7coreVStrt7pen
### Top Biological Processes GO
```{r top40NES_trt7coreVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
#--------------trt7coreVStrt7pen------------------------------------
trt7coreVStrt7penGSEADf <- trt7coreVStrt7penGSEADf[order(trt7coreVStrt7penGSEADf$NES, decreasing = FALSE),]
trt7coreVStrt7penGSEADf50 <- rbind(head(trt7coreVStrt7penGSEADf, n = 25), tail(trt7coreVStrt7penGSEADf, n = 25))
trt7coreVStrt7penGSEADf50$Description <- factor(trt7coreVStrt7penGSEADf50$Description, levels = trt7coreVStrt7penGSEADf50$Description)

#### bubble plot for GOs
ggplot(trt7coreVStrt7penGSEADf50, aes(x = NES, y = Description, color = -log10(p.adjust), size = setSize)) + 
geom_point(stat = "identity") + theme_bw() + facet_wrap(~ Direction, scales = "free_x") +
scale_color_gradient(low = "deepskyblue3", high = "#990000")

### barplot for GOs

#xpos <- c(1:nrow(trt7coreVStrt7penGSEADf50))
#ypos <- ifelse(trt7coreVStrt7penGSEADf50$NES>0,-1.1,1.1)
#
#ggplot(data = trt7coreVStrt7penGSEADf50,
#    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
#    geom_bar(stat = "identity", width = 0.7) +
#    geom_text(aes(y = ypos, x = Description, label = Description)) +
#    scale_color_manual(values = c("deepskyblue3", "orangered4")) +
#    scale_fill_manual(values = c("deepskyblue3", "orangered4")) +
#    ylim(c(-max(abs(trt7coreVStrt7penGSEADf50$NES)), max(abs(trt7coreVStrt7penGSEADf50$NES)))) +
#    coord_flip() +
#    theme(
#        axis.title.y=element_blank(), 
#        axis.text.y = element_blank(), 
#        axis.line.y = element_blank(), 
#        axis.ticks.y = element_blank(), 
#        panel.background=element_blank(), 
#        panel.grid.minor = element_blank()) + 
#    guides(fill = FALSE, color = FALSE)
```
<pre>



</pre>
### Chord Diagram
<font size = "4">**This plot portaits the relation of the three biological processes with the highest, and the lowest normalized enrichment score, and the 500 genes with the highest and the lowest wald statistic (a statistics that combines having a high log2-fold change and a highly significant p-adjusted value)**</font>
```{r chordDiagram_trt7coreVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 18, fig.width = 18, fig.align = "center"}
## setting up matrix containing all genes and pathways (membermatrix)
trt7coreVStrt7penGSEADf <- trt7coreVStrt7penGSEADf[order(trt7coreVStrt7penGSEADf$NES, decreasing = TRUE), ]
trt7coreVStrt7pen <- trt7coreVStrt7pen[order(trt7coreVStrt7pen$stat, decreasing = TRUE), ]
tail(trt7coreVStrt7pen)

pltPathway <- data.frame()
pltPathway <- data.frame(Category = rep("BP", nrow(trt7coreVStrt7penGSEADf)), ID = trt7coreVStrt7penGSEADf$ID, Term = trt7coreVStrt7penGSEADf$Description, Genes = toupper(trt7coreVStrt7penGSEADf$core_enrichment), adj_pval = trt7coreVStrt7penGSEADf$p.adjust)
pltPathway$Genes <- gsub("/", ",", pltPathway$Genes)

pltGenes <- data.frame(ID = toupper(rownames(trt7coreVStrt7pen)), logFC = trt7coreVStrt7pen$log2FoldChange)

pltCircle <- circle_dat(pltPathway, pltGenes)

## defining a selection of pathways (top 3 up&down) and genes (top 500 wald up&down)

pltPathwaySelection <- data.frame()
pltPathwaySelection <- rbind(head(trt7coreVStrt7penGSEADf)[1:3,], tail(trt7coreVStrt7penGSEADf)[1:3,])
pltPathwaySelection <- data.frame(Category = rep("BP", nrow(pltPathwaySelection)), ID = pltPathwaySelection$ID, Term = pltPathwaySelection$Description, Genes = toupper(pltPathwaySelection$core_enrichment), adj_pval = pltPathwaySelection$p.adjust)
pltPathwaySelection$Genes <- gsub("/", ",", pltPathwaySelection$Genes)

pltGenesSelection <- data.frame()
pltGenesSelection <- rbind(head(trt7coreVStrt7pen, n = 500)[1:500,], tail(trt7coreVStrt7pen, n = 500)[1:500,])
pltGenesSelection <- data.frame(ID = toupper(rownames(pltGenesSelection)), logFC = pltGenesSelection$log2FoldChange)

## a large number of genes for particular pathways
#pltChord <- chord_dat(data = pltCircle, process = pltPathwaySelection$Term, genes = pltGenes)
#GOChord(pltChord, space =0.01, gene.order = 'logFC', gene.space = 0.25, gene.size = 3)

## plotting the chord diagram with the selected pathways and genes
pltChordGenes <- chord_dat(data = pltCircle, genes = pltGenesSelection, process = pltPathwaySelection$Term)
GOChord(pltChordGenes, space =0.02, gene.order = 'logFC', gene.space = 0.2, gene.size = 4, process.label = 8)
```
<pre>



</pre>
### GO Network
<font size = "4">**In these plots each dot represents one significant gene ontology pathway.**</font>
```{r, GoNetwork_trt7coreVStrt7pen, message = FALSE, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
geneSetCollection <- list()
for (i in rownames(trt7coreVStrt7penGSEA@result)) {
    geneSet <- GeneSet(unlist(strsplit(trt7coreVStrt7penGSEA@result[i, "core_enrichment"], split = "/"))[], setName = i)
    geneSetCollection[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object =    geneSetCollection)

## VISUALIZATION

# create an overlap network
gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)

# ADD GENE SET STATISTCS, just the padj of the GO pathways
geneSetsPadj <- trt7coreVStrt7penGSEA@result$p.adjust
names(geneSetsPadj) <- rownames(trt7coreVStrt7penGSEA@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(ig = gs_ovnet, genesetStat = -log10(geneSetsPadj)) +
ggtitle("A network of all the significant GO pathways") + 
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### GO clusters
```{r GOclusters_trt7coreVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneSetsPadj,
    alg = igraph::cluster_louvain,
    minSize = 15)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps, genesetStat = -log10(geneSetsPadj)) +
ggtitle("12 most significant GO pathway clusters, Louvain Algorithm") +
scale_fill_gradient(high = "#000099", low = "white", name = "-log10(p.adj)") + 
scale_radius(name= "# of genes in GO")
```
<pre>



</pre>
### Cluster Themes 
```{r wordclouds_trt7coreVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 4, fig.width = 4, fig.align = "center"}
##### creating description lists
descriptions <- ""
descriptionsList <- list()
for (cluster in grps){
    for (GO in cluster){
      descriptions <- paste(descriptions, trt7coreVStrt7penGSEA@result[GO, "Description"], sep = " ")
  }
  descriptionsList <- append(descriptionsList, descriptions)
  descriptions <- ""
}

#preparing data for plotting + additional parameters
termFreq <- TermDocumentMatrix(descriptionsList) 
termFreq <- as.data.frame(as.matrix(termFreq))
termFreq <- termFreq %>%
  mutate(angle = 90 * sample(c(0, 1), n(), replace = TRUE, prob = c(70, 30)))
clusterNames <- lessR::to ("cluster", from = 1, until = nrow(grps))
colnames(termFreq) <- c(clusterNames, "angle")
termFreq$expression <- rownames(termFreq)
myColor <- colorRampPalette(c("deepskyblue3", "#990000"))(nrow(termFreq))

#### I can not manage with a loop unfortunately
set.seed(1234)
pltData <- list()
for(i in c(1:length(descriptionsList))){
    p1 <- ggplot(termFreq, aes(label = expression, size = termFreq[,i], angle = angle, color = expression)) +
    geom_text_wordcloud(area_corr = TRUE, area_corr_power = 1.05, eccentricity = 0.4, grid_margin = 0.3) + 
    scale_size_area(max_size = 9) +
    theme_minimal() + 
    scale_color_manual(values = myColor) + 
    ggtitle(colnames(termFreq)[i])
    print(p1)
    pltData[[i]] <- p1  #### DOES NOT WORK
}

#grid.arrange(grobs=pltData)

```
<pre>



</pre> 
### GO hierarchy
```{r GoHierarchySignificance_trt7coreVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    trt7coreVStrt7penGSEA,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre> 
### Heatmap GO genes
```{r heatmapGOgenes_trt7coreVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}

GO1 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][1:5]
GO2 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][20:24]
GO3 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][10:14]
GO4 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][5:9]



GO7 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]][6:10]
GO8 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]][26:30]
GO9 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]][c(1:4, 22)]
GO10 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]][1:5]


GOI <- c(GO1, GO2, GO3, GO4, GO7, GO8, GO9, GO10)

duplicated(GOI)

pltCounts <- assay(vst[GOI])
pltCounts <- pltCounts - rowMeans(pltCounts)


#pltAnno <- data.frame (row.names = GOI, 
#    "GO" = c(
#    rep(paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[1], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[2], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[2], sep = ":"), 5),
#    rep(paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[3], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[3], sep = ":"), 5),
#    rep(paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[4], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[4], sep = ":"), 5),
#    rep(paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[1], head(trt7coreVStrt7penGSEADf, n = 6)$ID[1], sep = ":"), 5), 
#    rep(paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[2], head(trt7coreVStrt7penGSEADf, n = 6)$ID[2], sep = ":"), 5), 
#    rep(paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[3], head(trt7coreVStrt7penGSEADf, n = 6)$ID[3], sep = ":"), 5), 
#    rep(paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[4], head(trt7coreVStrt7penGSEADf, n = 6)$ID[4], sep = ":"), 5)
#    ))
#
#newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
#mycolors <- newCols(length(unique(pltAnno$GO)))
#names(mycolors) <- unique(pltAnno$GO)
#mycolors <- list(GO = mycolors)

paletteLength <- 50
heatColors <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
heatBreaks <- c(
    seq(max(abs(pltCounts))*-1, 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(abs(pltCounts)) / paletteLength, max(abs(pltCounts)), length.out = floor(paletteLength / 2)))



p1 <- pheatmap(pltCounts,
         color=heatColors,
         breaks = heatBreaks,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         gaps_row= c(5, 10, 15, 20, 25, 30, 35),
         gaps_col= c(5, 10, 15, 20, 25, 30, 35, 40),
         cellheight = 11,
         cellwidth = 11,
         border_color=NA,
         fontsize_row = 6,
         main="Genes grouped by GO",
         #annotation_row = pltAnno,
         #annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         silent = TRUE)


p1
dev.off()
```
<pre>



</pre>
### Normalized, scaled, total gene expression of GO pathways
```{r heatmapGOtotal_trt7coreVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 16, fig.width = 22, fig.align = "center"}
GO1 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO2 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO3 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO4 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO5 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO6 <- strsplit(tail(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]
GO7 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[1], "/")[[1]]
GO8 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[2], "/")[[1]]
GO9 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[3], "/")[[1]]
GO10 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[4], "/")[[1]]
GO11 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[5], "/")[[1]]
GO12 <- strsplit(head(trt7coreVStrt7penGSEADf, n = 6)$core_enrichment[6], "/")[[1]]



GOI_objects <- list(GO1, GO2, GO3, GO4, GO5, GO6, GO7, GO8, GO9, GO10, GO11, GO12)

GOI_names <- c("GO1", "GO2", "GO3", "GO4", "GO5", "GO6", "GO7", "GO8", "GO9", "GO10", "GO11", "GO12")

pltData <- data.frame(row.names = samples$fileName)

counter <- 1
for (GOI in GOI_objects) {
  GOI_counts <- assay(vst[GOI])  # Get counts for the current GOI
  GOI_counts <- GOI_counts - rowMeans(GOI_counts)
  GOI_counts <- colSums(GOI_counts)
  pltData[GOI_names[counter]] <- GOI_counts
  counter <- counter +1
}

pltData <- t(pltData)

pltAnno <- data.frame (row.names = GOI_names, 
    "GO" = c(
    paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[1], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[2], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[2], sep = ":"),
    paste(tail(trt7coreVStrt7penGSEADf, n = 6)$ID[3], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[3], sep = ":"),
    paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[4], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[4], sep = ":"),
    paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[5], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(tail(trt7coreVStrt7penGSEADf, n = 6)$Description[6], tail(trt7coreVStrt7penGSEADf, n = 6)$ID[6], sep = ":"), 
    paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[1], head(trt7coreVStrt7penGSEADf, n = 6)$ID[1], sep = ":"), 
    paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[2], head(trt7coreVStrt7penGSEADf, n = 6)$ID[2], sep = ":"), 
    paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[3], head(trt7coreVStrt7penGSEADf, n = 6)$ID[3], sep = ":"), 
    paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[4], head(trt7coreVStrt7penGSEADf, n = 6)$ID[4], sep = ":"), 
    paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[5], head(trt7coreVStrt7penGSEADf, n = 6)$ID[5], sep = ":"), 
    paste(head(trt7coreVStrt7penGSEADf, n = 6)$Description[6], head(trt7coreVStrt7penGSEADf, n = 6)$ID[6], sep = ":") 
    ))

newCols <- colorRampPalette(grDevices::rainbow(length(unique(pltAnno$GO))))
mycolors <- newCols(length(unique(pltAnno$GO)))
names(mycolors) <- unique(pltAnno$GO)
mycolors <- list(GO = mycolors)

p1 <- pheatmap(pltData,
         scale="row",
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         cellheight = 50,
         cellwidth = 10,
         border_color=NA,
         fontsize_row = 6,
         gaps_row= c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
         gaps_col= c(5, 5, 10, 10, 15, 15, 20, 20, 25, 25, 30, 30, 35, 35, 40, 40),
         main="Genes grouped by GO",
         annotation_row = pltAnno,
         annotation_colors = mycolors, 
         annotation_col = samples[, c("condition", "timeAfterStroke", "brainRegion"), drop = FALSE], 
         show_rownames = FALSE,
         silent = TRUE)

p1
dev.off()
```
<pre>



</pre> 
### Enrichment Scores 
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
<br></br>
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
<br></br>
**Definition of signal: The enrichment signal strength that combines the two previous
statistics: (Tag %)  (1  Gene %)  (N / (N  Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaplot_trt7coreVStrt7pen, message = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 16, fig.align = "center"}
## highest normalized enrichment score
trt7coreVStrt7penGSEADf <- trt7coreVStrt7penGSEADf[order(trt7coreVStrt7penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7coreVStrt7penGSEA, geneSetID = rownames(head(trt7coreVStrt7penGSEADf, n = 5)),title = "5 pathways with the highest absolute enrichment score", pvalue_table = FALSE)
## highest leading edge
trt7coreVStrt7penGSEADf <- trt7coreVStrt7penGSEADf[order(trt7coreVStrt7penGSEADf$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7coreVStrt7penGSEA, geneSetID = rownames(head(trt7coreVStrt7penGSEADf, n = 5)), title = "5 Pathways with the higest leading edge", pvalue_table = FALSE )
## highest tag 
trt7coreVStrt7penGSEADf <- trt7coreVStrt7penGSEADf[order(trt7coreVStrt7penGSEADf$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(trt7coreVStrt7penGSEA, geneSetID = rownames(head(trt7coreVStrt7penGSEADf, n = 5)), title = "5 Pathways with the higest tag", pvalue_table = FALSE )
## highest signal
trt7coreVStrt7penGSEADf <- trt7coreVStrt7penGSEADf[order(trt7coreVStrt7penGSEADf$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(trt7coreVStrt7penGSEA, geneSetID = rownames(head(trt7coreVStrt7penGSEADf, n = 5)), title = "5 Pathways with the higest signal", pvalue_table = FALSE )
```
                                                                                                                             