---
output: 
    html_document: 
        code_folding: hide
---
<style type="text/css">
.main-container {
  max-width: 1800px;
  margin-left: auto;
  margin-right: auto;
}
</style>
<pre>
</pre>
# Results: genes_ensembl_raw_salmon {.tabset}
<font size = "4">**Our sequencing data was pseudomapped with salmon to the ensembl transcriptome, and
then differentially expressed genes are modeled with DESeq2.**</font>
<br>
</br>
<font size = "4">**All results are generated by the contrast of control and treatment. 
Control was chosen as the base reference of this comparison.**</font>
<pre>
</pre>
## 1 DESeq2Model & Technical Visualization
### DESeq2 model 
```{r RcodeDESeq2Model, message = FALSE, warning = FALSE} 
# library Import, has to be in that order otherwise biomaRt does not work
library("DESeq2")
library("readr")
library("tximport")
library("edgeR")
library("httpgd")
library("tximport")
library("knitr")
library("ggplot2")
library("ggrepel")
library("tidyverse")
library("RColorBrewer")
library("pheatmap")
library("reshape2")
library("vsn")
library("ggvenn")
library("ggpubr")
library("ggplotify")
library("biomaRt")
library("clusterProfiler")
library("org.Mm.eg.db")
library("msigdbr")
library("vissE")
library("msigdb")
library("GSEABase")
library("igraph")
library("patchwork")
library("ggpattern")
# size of plots
knitr::opts_chunk$set(fig.width = 12, fig.height = 8)
#--------------------------------------DESeq2 Model for DGE analysis-------------------------------------------------
# mapping statistics star
ensembl_raw_star_statistics <- read.csv(
    "/projects/neuronet/seq_april/processing/mapping/ensembl/raw/statistics.tsv",
    sep = "\t", header = TRUE, row.names = "ID"
)
# sample table
salmon_samples <- read.table("/projects/neuronet/seq_april/samples_april.csv", sep = ",", header = TRUE, row.names = 1)
salmon_samples$condition <- as.factor(salmon_samples$condition)
# import of data
ensembl_tx2gene <- read_csv("/projects/neuronet/seq_april/processing/genome/ensembl/tx2gene_GRCm39.109.csv", col_names = FALSE)
ensembl_raw_salmon_files <- file.path("/projects/neuronet/seq_april/processing/mapping/ensembl/raw/salmon", rownames(salmon_samples), "quant.sf")
names(ensembl_raw_salmon_files) <- rownames(salmon_samples)
ensembl_raw_salmon_txi <- tximport(ensembl_raw_salmon_files, type = "salmon", tx2gene = ensembl_tx2gene)
# deseq2 model
ensembl_raw_salmon_dds <- DESeqDataSetFromTximport(ensembl_raw_salmon_txi, colData = salmon_samples, design = ~condition)
ensembl_raw_salmon_dds$condition <- relevel(ensembl_raw_salmon_dds$condition, ref = "ctrl")
ensembl_raw_salmon_dds_keep <- rowSums(counts(ensembl_raw_salmon_dds) >= 5) >= 3
ensembl_raw_salmon_dds <- ensembl_raw_salmon_dds[ensembl_raw_salmon_dds_keep, ]
ensembl_raw_salmon_dds <- DESeq(ensembl_raw_salmon_dds)
# results
ensembl_raw_salmon_DESeq_Results <- results(ensembl_raw_salmon_dds, alpha = 0.1, contrast = c("condition", "trt", "ctrl"))
ensembl_raw_salmon_DESeq_Results_SHAM <- results(ensembl_raw_salmon_dds, alpha = 0.1, contrast = c("condition", "sham", "ctrl"))
# selection specifics sets of results
ensembl_raw_salmon_candidates <- row.names(subset(ensembl_raw_salmon_DESeq_Results, padj < 0.1))
ensembl_raw_salmon_DESeq_ResSig <- subset(ensembl_raw_salmon_DESeq_Results, padj < 0.1)
up_down_top20_cand <- rbind(
    head(ensembl_raw_salmon_DESeq_ResSig[order(ensembl_raw_salmon_DESeq_ResSig$log2FoldChange, decreasing = TRUE), ], n = 10),
    tail(ensembl_raw_salmon_DESeq_ResSig[order(ensembl_raw_salmon_DESeq_ResSig$log2FoldChange, decreasing = TRUE), ], n = 10)
)
padj_top20_cand <- head(ensembl_raw_salmon_DESeq_ResSig[order(ensembl_raw_salmon_DESeq_ResSig$padj, decreasing = FALSE), ], n = 20)
ensembl_raw_salmon_DESeq_Results_NA <- ensembl_raw_salmon_DESeq_Results[!complete.cases(ensembl_raw_salmon_DESeq_Results), ]
ensembl_raw_salmon_DESeq_Results_NA_counts <- assay(ensembl_raw_salmon_dds)[rownames(ensembl_raw_salmon_dds) %in% rownames(ensembl_raw_salmon_DESeq_Results_NA), ]
# counts ordered
ensembl_raw_salmon_dds_counts <- assay(ensembl_raw_salmon_dds)
ensembl_raw_salmon_dds_counts_colSums <- as.data.frame(colSums(ensembl_raw_salmon_dds_counts))
# counts overall samples (inhouse)
ensembl_raw_salmon_dds_counts_inhouse <- assay(ensembl_raw_salmon_dds)[, !grepl("SRR*", colnames(ensembl_raw_salmon_dds))]
ensembl_raw_salmon_dds_counts_inhouse_rowSums <- as.data.frame(rowSums(ensembl_raw_salmon_dds_counts_inhouse))
names(ensembl_raw_salmon_dds_counts_inhouse_rowSums) <- "counts"
ensembl_raw_salmon_dds_counts_inhouse_rowSums <- ensembl_raw_salmon_dds_counts_inhouse_rowSums[order(ensembl_raw_salmon_dds_counts_inhouse_rowSums$counts, decreasing = TRUE), , drop = FALSE]
# normalizations and transformations
ensembl_raw_salmon_dds_vst <- vst(ensembl_raw_salmon_dds, blind = FALSE)
ensembl_raw_salmon_dds_rlog <- rlog(ensembl_raw_salmon_dds, blind = TRUE)
ensembl_raw_salmon_dds_normTransform <- normTransform(ensembl_raw_salmon_dds)
ensembl_raw_salmon_dds_lfcShrink <- lfcShrink(ensembl_raw_salmon_dds, coef = "condition_trt_vs_ctrl", type = "apeglm")
########### --------------------------------PATHWAY ENRICHMENT ANALYSIS---------------------------------------------------
# gene ID map, and additional metadata
# listEnsembl()
ensembl <- useEnsembl(biomart = "genes", version = "109")
datasets <- listDatasets(ensembl)
# datasets
ensembl <- useDataset(dataset = "mmusculus_gene_ensembl", mart = ensembl)
# listFilters(ensembl)
# listAttributes(ensembl)
# AnnotationDbi::keytypes(org.Mm.eg.db) ### KEYTYPES of my database !!!
idMap <- getBM(
    attributes = c("mgi_symbol", "ensembl_gene_id", "entrezgene_id", "chromosome_name", "start_position", "end_position"),
    filters = "mgi_symbol",
    values = as.character(rownames(ensembl_raw_salmon_DESeq_Results)),
    mart = ensembl
)
idMap$endMinusStart <- abs(idMap$end_position - idMap$start_position)
idMapAmbiguous <- idMap[idMap$mgi_symbol %in% unique(idMap[duplicated(idMap$mgi_symbol), ])$mgi_symbol, ]
ensembl_raw_salmon_DESeq_Results_DF_metadata <- merge(
    as.data.frame(ensembl_raw_salmon_DESeq_Results),
    idMap,
    by.x = "row.names",
    by.y = "mgi_symbol",
    all.x = TRUE,
    all.y = FALSE
)
ensembl_raw_salmon_DESeq_ResSig_DF_metadata <- merge(
    as.data.frame(ensembl_raw_salmon_DESeq_ResSig),
    idMap,
    by.x = "row.names",
    by.y = "mgi_symbol",
    all.x = TRUE,
    all.y = FALSE
)
rownames(ensembl_raw_salmon_DESeq_ResSig_DF_metadata) <- ensembl_raw_salmon_DESeq_ResSig_DF_metadata$Row.names
ensembl_raw_salmon_DESeq_ResSig_DF_metadata$Row.names <- NULL
```
<pre>



</pre>
### Statistical summary of the DESeq2 result
```{r summary}
summary(ensembl_raw_salmon_DESeq_Results)
as.data.frame(ensembl_raw_salmon_DESeq_ResSig)
```
<pre>



</pre>
### Reads assigned to genes
```{r readsAssigned}
ensembl_raw_salmon_readCounts <- as.data.frame(colSums(ensembl_raw_salmon_txi$counts))
ensembl_raw_salmon_readCounts$condition <- salmon_samples[rownames(ensembl_raw_salmon_readCounts), ]
names(ensembl_raw_salmon_readCounts)[names(ensembl_raw_salmon_readCounts) == "colSums(ensembl_raw_salmon_txi$counts)"] <- "counts"
ensembl_raw_salmon_readCounts_quantiles <- quantile(ensembl_raw_salmon_readCounts$counts, probs = c(0.05, 0.25, 0.75, 0.95))

ggplot(
    ensembl_raw_salmon_readCounts,
    aes(x = rownames(ensembl_raw_salmon_readCounts), y = counts),
    col = condition
) +
    geom_bar(
        aes(fill = condition),
        stat = "identity"
    ) +
    geom_hline(
        yintercept = as.numeric(ensembl_raw_salmon_readCounts_quantiles),
        color = "red"
    ) +
    geom_label_repel(
        data = data.frame(x = 0, y = as.numeric(ensembl_raw_salmon_readCounts_quantiles)),
        aes(x = x, y = y, label = names(ensembl_raw_salmon_readCounts_quantiles))
    ) +
    scale_y_continuous(labels = scales::comma) +
    coord_flip() +
    xlab("sample")
```
<pre>



</pre>
### Ratio top5/sample
Piechart showing the number of unmapped reads, various mapped reads and the fraction of the 5 most highly expressed genes for each sample. 
```{r MappingInfoSample}
for (i in colnames(ensembl_raw_salmon_dds_counts)) {
    total_reads <- (as.numeric(gsub(",", "", ensembl_raw_star_statistics[paste("raw/star/", i, sep = ""), "reads"])))
    total_mapped <- ensembl_raw_salmon_dds_counts_colSums[i, ]
    top_twenty <- as.data.frame(ensembl_raw_salmon_dds_counts[order(ensembl_raw_salmon_dds_counts[, i], decreasing = TRUE), i][1:5], header = TRUE)
    top_twenty_sum <- colSums(top_twenty)
    unmapped_reads <- total_reads - total_mapped
    remaining_mapped <- total_mapped - colSums(top_twenty)

    pltData <- data.frame(
        row.names = c("unmapped reads", "various", row.names(top_twenty)),
        values = mapply(c, unmapped_reads, remaining_mapped, top_twenty)
    )

    pltMillion <- ggplot(
        pltData,
        aes(x = "", y = values, fill = row.names(pltData))
    ) +
        geom_bar(stat = "identity", width = 1, color = "white") +
        coord_polar(theta = "y") +
        geom_label_repel(aes(label = paste(format(round(values / 1e6, 1), trim = TRUE), "M")), position = position_stack(vjust = 0.5)) +
        theme_void() +
        guides(fill = guide_legend(title = element_blank()))
    pltPercentage <- ggplot(
        pltData,
        aes(x = "", y = values, fill = row.names(pltData))
    ) +
        geom_bar(stat = "identity", width = 1, color = "white") +
        coord_polar(theta = "y") +
        geom_label_repel(aes(label = paste(format(round((values / colSums(pltData)) * 100), trim = TRUE), "%")), position = position_stack(vjust = 0.5)) +
        theme_void() +
        guides(fill = guide_legend(title = element_blank()))

    print(ggarrange(
        pltMillion,
        pltPercentage,
        labels = paste("Sample: ", i, sep = ""),
        common.legend = TRUE,
        legend = "bottom",
        ncol = 2,
        nrow = 1
    ))
}
```
<pre>



</pre>
### Ratio top5 across all samples
Piechart showing the number of unmapped reads, various mapped reads and the fraction of the 5 most highly expressed genes across all samples.
```{r MappingInfoTotal}
total_reads <- sum(as.numeric(gsub(",", "", ensembl_raw_star_statistics[!grepl("SRR*", row.names(ensembl_raw_star_statistics)), "reads"])))
total_mapped <- sum(ensembl_raw_salmon_dds_counts_inhouse_rowSums)
top_ten <- ensembl_raw_salmon_dds_counts_inhouse_rowSums[1:6, , drop = FALSE]
unmapped_reads <- total_reads - total_mapped
remaining_mapped <- total_mapped - sum(top_ten)
pltData <- data.frame(
    row.names = c("unmapped reads", "various", row.names(top_ten)),
    values = mapply(c, unmapped_reads, remaining_mapped, top_ten)
)
pltData <- pltData[order(pltData$values, decreasing = TRUE), , drop = FALSE]

pltMillion <- ggplot(
    pltData,
    aes(x = "", y = values, fill = row.names(pltData))
) +
    geom_bar(stat = "identity", width = 1, color = "white") +
    coord_polar(theta = "y") +
    geom_label_repel(aes(label = paste(format(round(values / 1e6, 1), trim = TRUE), "M")), position = position_stack(vjust = 0.5)) +
    theme_void() +
    guides(fill = guide_legend(title = element_blank())) +
    scale_fill_discrete(labels = pltData)
pltPercentage <- ggplot(
    pltData,
    aes(x = "", y = values, fill = row.names(pltData))
) +
    geom_bar(stat = "identity", width = 1, color = "white") +
    coord_polar(theta = "y") +
    geom_label_repel(aes(label = paste(format(round((values / sum(pltData$values)) * 100), trim = TRUE), "%")), position = position_stack(vjust = 0.5)) +
    theme_void() +
    guides(fill = guide_legend(title = element_blank()))

print(ggarrange(
    pltMillion,
    pltPercentage,
    labels = "Count fractions of all samples combined",
    common.legend = TRUE,
    legend = "bottom",
    legend.grob = get_legend(pltMillion),
    ncol = 2,
    nrow = 1
))
```
<pre>



</pre>
### Sparsity plot
A simple plot of the concentration of counts in a single sample 
over the sum of counts per gene. Not technically the same as "sparsity", 
but this plot is useful diagnostic for datasets which might not fit a 
negative binomial assumption: genes with many zeros and individual very 
large counts are difficult to model with the negative binomial 
distribution.
```{r plotSparsity}
plotSparsity(
    ensembl_raw_salmon_dds,
    col = ensembl_raw_salmon_dds$condition,
    ann = TRUE,
    cex = 0.4,
    pch = 4
)
legend(
    "topright",
    legend = levels(ensembl_raw_salmon_dds$condition),
    pch = 4,
    col = unique(ensembl_raw_salmon_dds$condition)
)
```
<pre>



</pre>
### Bland-Altman plot 
An MA-plot is a plot of log-intensity ratios (M-values) versus intensity averages (A-values). 
```{r plotMA}
DESeq2::plotMA(ensembl_raw_salmon_DESeq_Results,
    ylim = c(-5, 5),
    colNonSig = "black",
    colSig = "red"
)
legend("topright",
    legend = c("padj > 0.1", "padj < 0.1"),
    pch = 16,
    col = c("black", "red")
)
```
<pre>



</pre>
### Bland-Altman plot (LFCshrink)
LfcShrink adds shrunken log2-fold changes (LFC) and standard error to a results 
table from DESeq run without LFC shrinkage.
<br>
</br>
Specifying apeglm passes along DESeq2 MLE log2 fold 
changes and standard errors to the apeglm function in the apeglm 
package, and re-estimates posterior LFCs for the coefficient 
specified by coef.
```{r plotMAlfc}
DESeq2::plotMA(ensembl_raw_salmon_dds_lfcShrink,
    ylim = c(-5, 5),
    colNonSig = "black",
    colSig = "red"
)
legend(
    "topright",
    legend = c("padj > 0.1", "padj < 0.1"),
    pch = 16,
    col = c("black", "red")
)
```
<pre>



</pre>
### Mean vs sd (normTransform) 
```{r meanSdPlotNorm}
meanSdPlot(assay(ensembl_raw_salmon_dds_normTransform))
```
<pre>



</pre>
### Mean vs sd (vst)
```{r meanSdPlotVst}
meanSdPlot(assay(ensembl_raw_salmon_dds_vst))
```
<pre>



</pre>
### Mean vs sd (rlog)
```{r meanSdPlotRlog}
meanSdPlot(assay(ensembl_raw_salmon_dds_rlog))
```
<pre>



</pre>
### P-values histogram
The distribution of p-values across the ~20k hypothesis tested. 
```{r pValues}
ggplot(
    data = na.omit(as.data.frame(ensembl_raw_salmon_DESeq_Results)),
    aes(x = pvalue)
) +
    geom_histogram(bins = 150)
```
<pre>



</pre>
### Adjusted p-value histogram
The distribution of p-adjusted (Benjamini-Hochberg) values across the ~20k hypothesis tested. 
```{r padjValues}
ggplot(
    data = na.omit(as.data.frame(ensembl_raw_salmon_DESeq_Results)),
    aes(x = padj)
) +
    geom_histogram(bins = 150)
```
<pre>



</pre>
## DGE Visualization
### Principal component analysis (DESeq2)
Generic PCA plot of regularized log-transformed data. 
The rlog function transforms the count data to the 
log2 scale in a way which minimizes differences between 
samples for rows with small counts, and which normalizes with 
respect to library size.
```{r pcaDESeq2}
plotData <- plotPCA(ensembl_raw_salmon_dds_rlog, intgroup = "condition", returnData = TRUE)
ggplot(plotData, aes(x = PC1, y = PC2, color = condition)) +
geom_point(size = 4) +
scale_color_manual(values = c(sham = "black", trt = "#000099", ctrl = "#990000")) +
ylab("PC2: 5% variance") + xlab("PC1: 86% variance")
```
<pre>



</pre>
### Principal component analysis (R built-in)
```{r pcaR}
ensembl_raw_salmon_dds_rlog_PCA <- prcomp(t(assay(ensembl_raw_salmon_dds_rlog)), scale = TRUE)
var_explained <- data.frame("var_explained" = ensembl_raw_salmon_dds_rlog_PCA$sdev**2/sum(ensembl_raw_salmon_dds_rlog_PCA$sdev**2))
var_explained$PC <- factor(sprintf("PC%s", seq(1:13)), levels = sprintf("PC%s", seq(1:13)))
# variation explained by the respectice PC
 p1 <- ggplot(var_explained, aes(y = var_explained, x = PC)) + 
geom_bar(stat = "identity") +
geom_text(aes(label = round(var_explained, digits = 2)), vjust = -0.5) + 
ggtitle("Variance explained by the respective PC's")
# PCA
p2 <- ensembl_raw_salmon_dds_rlog_PCA$x %>%
    as.data.frame %>%
        ggplot(aes(x = PC1, y = PC2, color = salmon_samples$condition)) + 
        geom_point(size = 4) + guides(color = guide_legend("condtion")) +
        scale_color_manual(values = c("ctrl" = "#990000", "trt" = "#000099", "sham" = "grey35")) + 
        labs (
            x = paste ("PC1: ", round(var_explained[1,1]*100, 0), "%"), 
            y = paste("PC2: ", round(var_explained[2,1]*100, 0), "%")) +
        ggtitle("PCA")
# PCA with k-means clustering
set.seed(123)
ensembl_raw_salmon_dds_rlog_PCA_kmeans <- kmeans(as.data.frame(ensembl_raw_salmon_dds_rlog_PCA$x[,1:2]), centers = 4, iter.max = 10, nstart = 25)
p3 <- ensembl_raw_salmon_dds_rlog_PCA$x %>%
    as.data.frame %>%
        ggplot(aes(x = PC1, y = PC2, color = salmon_samples$condition, shape = factor(ensembl_raw_salmon_dds_rlog_PCA_kmeans$cluster))) + 
        geom_point(size = 4) + 
        guides(color = guide_legend("condtion"), shape = guide_legend("k-means cluster")) +
        scale_color_manual(values = c("ctrl" = "#990000", "trt" = "#000099", "sham" = "grey35")) +
        scale_shape_manual(values = c(15, 16, 17, 18)) +
        ggtitle("PCA with kmeans clustering") + 
        labs (x = paste ("PC1: ", round(var_explained[1,1]*100, 0), "%"), 
        y = paste("PC2: ", round(var_explained[2,1]*100, 0), "%"))
p1 + p2 + p3 + plot_layout(2,2)
```
<pre>



</pre>
### Heatmap of sample distance
Euclidean distances of the transformed, normalized samples, plotted in a heatmap. 
```{r heatmapSamples}
ensembl_raw_salmon_dist <- dist(t(assay(ensembl_raw_salmon_dds_vst)), method = "euclidean")
ensembl_raw_salmon_distMatrix <- as.matrix(ensembl_raw_salmon_dist)
rownames(ensembl_raw_salmon_distMatrix) <- paste(colnames(ensembl_raw_salmon_dds_vst), ensembl_raw_salmon_dds_vst$condition, sep = "-")
colnames(ensembl_raw_salmon_distMatrix) <- paste(colnames(ensembl_raw_salmon_dds_vst), ensembl_raw_salmon_dds_vst$condition, sep = "-")
paletteLength <- 255
myColor <- colorRampPalette(c("#000099", "white"))(paletteLength)
plt <- pheatmap(ensembl_raw_salmon_distMatrix,
    clustering_distance_rows = ensembl_raw_salmon_dist,
    clustering_distance_cols = ensembl_raw_salmon_dist,
    col = myColor, 
)
plt
```
<pre>



</pre>
### Volcano plot (trt vs ctrl)
Scatterplot showing statistical significance [-log10(padj)] versus magnitude of change [log2FoldChange].
```{r volcanoPlot}
ensembl_raw_salmon_DESeq_Results_volcano <- ensembl_raw_salmon_DESeq_Results
ensembl_raw_salmon_DESeq_Results_volcano$diffExpressed <- "NO"
ensembl_raw_salmon_DESeq_Results_volcano$diffExpressed[ensembl_raw_salmon_DESeq_Results_volcano$padj < 0.1 & ensembl_raw_salmon_DESeq_Results_volcano$log2FoldChange < 0] <- "Down"
ensembl_raw_salmon_DESeq_Results_volcano$diffExpressed[ensembl_raw_salmon_DESeq_Results_volcano$padj < 0.1 & ensembl_raw_salmon_DESeq_Results_volcano$log2FoldChange > 0] <- "Up"

ggplot(
    data = as.data.frame(ensembl_raw_salmon_DESeq_Results),
    aes(
        x = log2FoldChange,
        y = -log10(padj),
        col = ensembl_raw_salmon_DESeq_Results_volcano$diffExpressed,
        label = rownames(ensembl_raw_salmon_DESeq_Results)
    )
) +
    xlim(-7, 7) +
    geom_point(size = 2, na.rm = TRUE) +
    theme_minimal() +
    theme(legend.title = element_blank()) +
    geom_vline(xintercept = c(1, -1), col = "gray", linetype = "dashed") +
    geom_hline(yintercept = -log10(.1), col = "gray", linetype = "dashed") +
    scale_color_manual(
        values = c("#000099", "grey80", "#990000"),
        labels = c("sig DOWN", "no sig", "sig UP")
    ) +
    geom_text_repel(
        aes(
            x = log2FoldChange,
            y = -log10(padj),
            label = ifelse(padj < 0.1, as.character(rownames(ensembl_raw_salmon_DESeq_Results)), "")
        ),
        hjust = 0,
        vjust = -1.2,
        segment.color = NA
    )
```
<pre>



</pre>
### Volcano Plot (sham vs ctrl)
```{r volcanoPlotShamVsCtrl}
ensembl_raw_salmon_DESeq_Results_SHAM_volcano <- ensembl_raw_salmon_DESeq_Results_SHAM
ensembl_raw_salmon_DESeq_Results_SHAM_volcano$diffExpressed <- "NO"
ensembl_raw_salmon_DESeq_Results_SHAM_volcano$diffExpressed[ensembl_raw_salmon_DESeq_Results_SHAM_volcano$padj < 0.1 & ensembl_raw_salmon_DESeq_Results_SHAM_volcano$log2FoldChange < 0] <- "Down"
ensembl_raw_salmon_DESeq_Results_SHAM_volcano$diffExpressed[ensembl_raw_salmon_DESeq_Results_SHAM_volcano$padj < 0.1 & ensembl_raw_salmon_DESeq_Results_SHAM_volcano$log2FoldChange > 0] <- "Up"

ggplot(
    data = as.data.frame(ensembl_raw_salmon_DESeq_Results_SHAM),
    aes(
        x = log2FoldChange,
        y = -log10(padj),
        col = ensembl_raw_salmon_DESeq_Results_SHAM_volcano$diffExpressed,
        label = rownames(ensembl_raw_salmon_DESeq_Results_SHAM)
    )
) +
    geom_point(size = 2, na.rm = TRUE) +
    theme_minimal() +
    theme(legend.title = element_blank()) +
    geom_vline(xintercept = c(1, -1), col = "gray", linetype = "dashed") +
    geom_hline(yintercept = -log10(.1), col = "#646363", linetype = "dashed") +
    scale_color_manual(
        values = c("#000099", "grey", "#990000"),
        labels = c("sig Down", "no sig", "sig Up")
    )
```
<pre>



</pre>
### Heatmap padj<0.1 clustered
The normalized expression of our candidate genes across all samples plotted in a heatmap.
```{r heatmapClustered}
ensembl_raw_salmon_dds_vst_cand <- assay(ensembl_raw_salmon_dds_vst)[ensembl_raw_salmon_candidates, ]
ensembl_raw_salmon_dds_vst_cand <- ensembl_raw_salmon_dds_vst_cand - rowMeans(ensembl_raw_salmon_dds_vst_cand)
paletteLength <- 50
myColor <- colorRampPalette(c("#990000", "white", "#000099"))(paletteLength)
myBreaks <- c(
    seq(min(ensembl_raw_salmon_dds_vst_cand), 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(ensembl_raw_salmon_dds_vst_cand) / paletteLength, max(ensembl_raw_salmon_dds_vst_cand), length.out = floor(paletteLength / 2)))
pltNoSham <- function(){
    pltNoSham <- pheatmap(
    ensembl_raw_salmon_dds_vst_cand[, !grepl("sham", salmon_samples$condition)],
    color = myColor,
    breaks = myBreaks,
    annotation_col = salmon_samples,
    annotation_colors = list(condition = c(ctrl = "grey60", trt = "black")),
    show_colnames = FALSE,
    scale = "row", 
    silent = TRUE)
    rowOrder <- pltNoSham$tree_row$order
    return(pltNoSham)
    return(rowOrder)}
pltSham <- function(){
    pltSham <- pheatmap(
    ensembl_raw_salmon_dds_vst_cand[, grepl("sham", salmon_samples$condition)][rowOrder, ],
    color = myColor,
    breaks = myBreaks,
    annotation_col = salmon_samples,
    annotation_colors = list(condition = c(sham = "white")),
    show_colnames = FALSE,
    cluster_rows = FALSE, 
    silent = TRUE)
    return(pltSham) }
ggarrange(
    as.ggplot(pltNoSham()),
    as.ggplot(pltSham()),
    common.legend = TRUE,
    legend = "bottom",
    ncol = 2,
    nrow = 1,
    widths = c(2, 1))
```
<pre>



</pre>
### Heatmap NOT Clustered (ordered by LFC and condition)
```{r heatmapUnclustered}
ensembl_raw_salmon_dds_vst_cand_log2FCord <- assay(ensembl_raw_salmon_dds_vst)[row.names(ensembl_raw_salmon_DESeq_ResSig[order(ensembl_raw_salmon_DESeq_ResSig$log2FoldChange, decreasing = TRUE), ]), ]
ensembl_raw_salmon_dds_vst_cand_log2FCord <- as.data.frame(ensembl_raw_salmon_dds_vst_cand_log2FCord - rowMeans(ensembl_raw_salmon_dds_vst_cand_log2FCord))
ensembl_raw_salmon_dds_vst_cand_log2FCord <- ensembl_raw_salmon_dds_vst_cand_log2FCord %>% relocate(c("SRR6899123_1", "SRR6899124_1", "SRR6899125_1"), .before = "363")

paletteLength <- 50
myColor <- colorRampPalette(c("#000099", "white", "#990000"))(paletteLength)
myBreaks <- c(
    seq(min(ensembl_raw_salmon_dds_vst_cand), 0, length.out = ceiling(paletteLength / 2) + 1),
    seq(max(ensembl_raw_salmon_dds_vst_cand) / paletteLength, max(ensembl_raw_salmon_dds_vst_cand), length.out = floor(paletteLength / 2)))
plt <- pheatmap(
    ensembl_raw_salmon_dds_vst_cand_log2FCord,
    color = myColor,
    breaks = myBreaks,
    annotation_col = salmon_samples,
    annotation_colors = list(condition = c(ctrl = "grey60", trt = "black", sham = "white")),
    show_colnames = FALSE,
    cluster_rows = FALSE,
    cluster_cols = FALSE,
    scale = "row"
)
plt
```
<pre>



</pre>
### Barplot up/down
This plot summarizes the number of significantly up and down-regulated genes based on different padj and log2FoldChange thresholds (comparison control vs treatment, control is always the base level of the comparison).
```{r barplotUpDown}
filterUp <- function(df, log2fc = 1, p = 0.1) {
    nrow(df[df$log2FoldChange >= log2fc & !is.na(df$padj) & df$padj <= p, ])}
filterDown <- function(df, log2fc = 1, p = 0.1) {
    nrow(df[df$log2FoldChange <= -log2fc & !is.na(df$padj) & df$padj <= p, ])}
pVals <- c(0.001, 0.01, 0.05, 0.1)
fcVals <- c(0:(max(ensembl_raw_salmon_DESeq_ResSig$log2FoldChange) + 1))
summaryUpDown <- do.call(rbind, lapply(pVals, function(p) {
    do.call(rbind, lapply(fcVals, function(f) {
        up <- filterUp(ensembl_raw_salmon_DESeq_Results, f, p)
        down <- filterDown(ensembl_raw_salmon_DESeq_Results, f, p)
        return(data.frame(
            "log2FoldChange" = f, "padj" = p,
            "upRegulated" = up, "downRegulated" = down
        ))
    }))
}))
dataUpDown <- melt(summaryUpDown, id.vars = c("log2FoldChange", "padj"))

ggplot(
    dataUpDown,
    aes(x = log2FoldChange, y = value)) +
geom_bar(
    aes(fill = variable),
    stat = "identity",
    position = "dodge") +
facet_grid(~padj) +
scale_fill_manual(values = c("upRegulated" = "#990000", "downRegulated" = "#000099")) +
theme(legend.position = "bottom", legend.title = element_blank()) +
labs(
    title = "Number of differentially up/down regulated genes",
    subtitle = "based on different padj values and log2FoldChange cut-off values") +
ylab("gene count")
```
<pre>



</pre>
### Count plots top-features 
```{r 18_countPlots}
# creating a special dataframe with additional metadata
ensembl_raw_salmon_candidates_dds <- ensembl_raw_salmon_dds[ensembl_raw_salmon_candidates, ]
ensembl_raw_salmon_candidates_counts <- as.data.frame(t(counts(ensembl_raw_salmon_candidates_dds)))
ensembl_raw_salmon_candidates_counts <- merge(
    ensembl_raw_salmon_candidates_counts,
    salmon_samples,
    by.x = "row.names",
    by.y = "row.names",
    all.x = TRUE,
    all.y = TRUE
)
rownames(ensembl_raw_salmon_candidates_counts) <- ensembl_raw_salmon_candidates_counts$Row.names
ensembl_raw_salmon_candidates_counts$Row.names <- NULL
ensembl_raw_salmon_candidates_counts$sample <- rownames(ensembl_raw_salmon_candidates_counts)
ensembl_raw_salmon_candidates_counts <- melt(ensembl_raw_salmon_candidates_counts, id.vars = c("sample", "condition"))
ensembl_raw_salmon_candidates_padj <- data.frame(padj = ensembl_raw_salmon_DESeq_ResSig$padj, row.names = rownames(ensembl_raw_salmon_DESeq_ResSig))
ensembl_raw_salmon_candidates_VARIOUS <- merge(
    ensembl_raw_salmon_candidates_counts,
    ensembl_raw_salmon_candidates_padj,
    by.x = "variable",
    by.y = "row.names",
    all.x = TRUE,
    all.y = TRUE
)
## the actual plotting
set.seed(35)
ggplot(
    ensembl_raw_salmon_candidates_VARIOUS,
    aes(x = variable, y = log10(value), color = condition, size = -log10(padj))) +
geom_point() +
geom_jitter(height = NULL, width = 0.35) +
scale_color_manual(values = c("#990000", "grey35", "#000099")) +
ylab("log10(counts)") +
xlab("gene") +
scale_size_continuous(limits = c(0.5, 5.5), breaks = c(1, 3, 5), range = c(1, 5)) +
guides(size = guide_legend(override.aes = list(color = "grey")), 
    color = guide_legend(override.aes = list(size = 3)))
```
<pre>



</pre>
### Biological processes (GO) of our top-features level 2
```{r, GoBp2}
GoBp2_plot <- ensembl_raw_salmon_DESeq_GoBp2@result[which(ensembl_raw_salmon_DESeq_GoBp2@result$Count > 0), ]

for (i in rownames(ensembl_raw_salmon_DESeq_ResSig_DF_metadata)) {
    GoBp2_plot[, i] <- 0
    GoBp2_plot[grep(i, GoBp2_plot$geneID), i] <- 1
    GoBp2_plot[, i] <- factor(GoBp2_plot[, i], levels = c("0", "1"))
}

GoBp2_plotMelt <- GoBp2_plot[, rownames(ensembl_raw_salmon_DESeq_ResSig_DF_metadata)]
GoBp2_plotMelt$Description <- GoBp2_plot$Description
GoBp2_plotMelt <- melt(GoBp2_plotMelt, id.vars = "Description")

GoBp2_plotMelt$value <- factor(GoBp2_plotMelt$value, levels = c("0", "1"))

ggplot(
    data = GoBp2_plotMelt,
    aes(x = variable, y = Description)) +
geom_point(size = 10, colour = "black", alpha = I(ifelse(GoBp2_plotMelt$value == 0, 0, 1))) +
scale_x_discrete(guide = guide_axis(angle = 90))
```
<pre>


</pre> 
### Biological processes (GO) of our top-features level 3
```{r GoBp3}
GoBp3_plot <- ensembl_raw_salmon_DESeq_GoBp3@result[which(ensembl_raw_salmon_DESeq_GoBp3@result$Count > 0), ]

for (i in rownames(ensembl_raw_salmon_DESeq_ResSig_DF_metadata)) {
    GoBp3_plot[, i] <- 0
    GoBp3_plot[grep(i, GoBp3_plot$geneID), i] <- 1
    GoBp3_plot[, i] <- factor(GoBp3_plot[, i], levels = c("0", "1"))
}

GoBp3_plotMelt <- GoBp3_plot[, rownames(ensembl_raw_salmon_DESeq_ResSig_DF_metadata)]
GoBp3_plotMelt$Description <- GoBp3_plot$Description
GoBp3_plotMelt <- melt(GoBp3_plotMelt, id.vars = "Description")

GoBp3_plotMelt$value <- factor(GoBp3_plotMelt$value, levels = c("0", "1"))

ggplot(
    data = GoBp3_plotMelt,
    aes(x = variable, y = Description)) +
geom_point(size = 3, colour = "black", alpha = I(ifelse(GoBp3_plotMelt$value == 0, 0, 1))) +
scale_x_discrete(guide = guide_axis(angle = 90)) +
theme_minimal() +
theme(axis.text.y=element_text(size=5, face = "bold"))
```
<pre>



</pre>
## Pathway Enrichment Model & Visualization
### Code for Pathway enrichment
```{r RcodePathwayEnrichment, message = FALSE, warning = FALSE}
# GO Pathways of my significant genes in different levels
ensembl_raw_salmon_DESeq_GoBp1 <- groupGO(
    gene = as.character(ensembl_raw_salmon_DESeq_ResSig_DF_metadata$entrezgene_id),
    OrgDb = "org.Mm.eg.db",
    keyType = "ENTREZID",
    ont = "BP",
    level = 1,
    readable = TRUE
)
ensembl_raw_salmon_DESeq_GoBp2 <- groupGO(
    gene = as.character(ensembl_raw_salmon_DESeq_ResSig_DF_metadata$entrezgene_id),
    OrgDb = "org.Mm.eg.db",
    keyType = "ENTREZID",
    ont = "BP",
    level = 2,
    readable = TRUE
)
ensembl_raw_salmon_DESeq_GoBp3 <- groupGO(
    gene = as.character(ensembl_raw_salmon_DESeq_ResSig_DF_metadata$entrezgene_id),
    OrgDb = "org.Mm.eg.db",
    keyType = "ENTREZID",
    ont = "BP",
    level = 3,
    readable = TRUE
)
ensembl_raw_salmon_DESeq_GoBp4 <- groupGO(
    gene = as.character(ensembl_raw_salmon_DESeq_ResSig_DF_metadata$entrezgene_id),
    OrgDb = "org.Mm.eg.db",
    keyType = "ENTREZID",
    ont = "BP",
    level = 4,
    readable = TRUE
)
# GO Enrichment analysis
ensembl_raw_salmon_DESeq_GoBp <- enrichGO(rownames(ensembl_raw_salmon_DESeq_ResSig), keyType = "SYMBOL", OrgDb = "org.Mm.eg.db", ont = "BP", pvalueCutoff = 0.1, pAdjustMethod = "BH", universe = rownames(ensembl_raw_salmon_DESeq_Results))
ensembl_raw_salmon_DESeq_GoMf <- enrichGO(rownames(ensembl_raw_salmon_DESeq_ResSig), keyType = "SYMBOL", OrgDb = "org.Mm.eg.db", ont = "MF", pvalueCutoff = 0.1, pAdjustMethod = "BH", universe = rownames(ensembl_raw_salmon_DESeq_Results))
ensembl_raw_salmon_DESeq_GoCc <- enrichGO(rownames(ensembl_raw_salmon_DESeq_ResSig), keyType = "SYMBOL", OrgDb = "org.Mm.eg.db", ont = "CC", pvalueCutoff = 0.1, pAdjustMethod = "BH", universe = rownames(ensembl_raw_salmon_DESeq_Results))
# Molecular signature enrichment analysis
msigdbr_h_gene_sets <- msigdbr(species = "mouse", category = "H")
msigdbr_h <- clusterProfiler::enricher(
    gene = rownames(ensembl_raw_salmon_DESeq_ResSig_DF_metadata),
    universe = rownames(ensembl_raw_salmon_DESeq_Results),
    pAdjustMethod = "BH",
    TERM2GENE = msigdbr_h_gene_sets[, c("gs_name", "gene_symbol")]
)
# Gene set enrichment analysis GO terms
ensembl_raw_salmon_DESeq_WaldStats <- ensembl_raw_salmon_DESeq_Results$stat
names(ensembl_raw_salmon_DESeq_WaldStats) <- rownames(ensembl_raw_salmon_DESeq_Results)
ensembl_raw_salmon_DESeq_WaldStats <- ensembl_raw_salmon_DESeq_WaldStats[order(ensembl_raw_salmon_DESeq_WaldStats, decreasing = TRUE)]
ensembl_raw_salmon_gsea <- gseGO(
    geneList = ensembl_raw_salmon_DESeq_WaldStats,
    ont = "BP",
    OrgDb = "org.Mm.eg.db",
    keyType = "SYMBOL",
    minGSSize = 60,
    eps = 0,
    seed = T
)
ensembl_raw_salmon_gsea <- ensembl_raw_salmon_gsea[order(ensembl_raw_salmon_gsea$NES), asis = TRUE]
ensembl_raw_salmon_gsea_simplify <- clusterProfiler::simplify(ensembl_raw_salmon_gsea)
### GSEA selection
ensembl_raw_salmon_gsea_selection <- ensembl_raw_salmon_gsea[c(
    which(ensembl_raw_salmon_gsea$ID == "GO:0001525"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0050900"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0045766"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0001819"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0002252"),
    which(ensembl_raw_salmon_gsea$ID == "GO:1903555"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0071706"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0042110"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0032640"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0032680"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0002685"),
    which(ensembl_raw_salmon_gsea$ID == "GO:0007416")
), , asis = TRUE]
## GSEA results as dataframe
ensembl_raw_salmon_gseaDF <- as.data.frame(ensembl_raw_salmon_gsea)
ensembl_raw_salmon_gseaDF$leadTag <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ensembl_raw_salmon_gseaDF$leading_edge, ","), `[`, 1,
                FUN.VALUE = character(1)
            ), "tags=", ""
        ), "%", ""
    )
) / 100
ensembl_raw_salmon_gseaDF$leadGene <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ensembl_raw_salmon_gseaDF$leading_edge, ","), `[`, 2,
                FUN.VALUE = character(1)
            ), " list=", ""
        ), "%", ""
    )
) / 100
ensembl_raw_salmon_gseaDF$leadSignal <- as.integer(
    str_replace_all(
        str_replace_all(
            vapply(
                strsplit(ensembl_raw_salmon_gseaDF$leading_edge, ","), `[`, 3,
                FUN.VALUE = character(1)
            ), " signal=", ""
        ), "%", ""
    )
) / 100
ensembl_raw_salmon_gseaDF$leading_edge <- NULL
ensembl_raw_salmon_gseaDF <- ensembl_raw_salmon_gseaDF[order(ensembl_raw_salmon_gseaDF$leadTag, decreasing = TRUE), ]
ensembl_raw_salmon_gseaDF$Direction <- "DownRegulated"
ensembl_raw_salmon_gseaDF[which(ensembl_raw_salmon_gseaDF$enrichmentScore > 0), ]$Direction <- "UpRegulated"
# KEGG enrichment analysis STILL TO COME
# GSEA KEGG  STILL TO COME
# GSEA Molecular Signature STILL TO COME
```
<pre> 



</pre>
###Enrichment Score of the pathways with the highest absolute enrichment score
```{r gseaplotTop5NES}
ensembl_raw_salmon_gseaDF <- ensembl_raw_salmon_gseaDF[order(abs(ensembl_raw_salmon_gseaDF$NES), decreasing = TRUE), ]
enrichplot::gseaplot2(ensembl_raw_salmon_gsea, geneSetID = rownames(head(ensembl_raw_salmon_gseaDF)))
```
<pre>



</pre>
### Enrichment score of the 5 GO's with the higest leading edge
**Definition of leading edge: The percentage of gene tags before (for positive ES) or after (for negative ES) the
peak in the running enrichment score S. The larger the percentage, the more tags in the
gene set contribute to the final enrichment score.**
```{r gseaPlot5leadingEdge}
ensembl_raw_salmon_gseaDF <- ensembl_raw_salmon_gseaDF[order(ensembl_raw_salmon_gseaDF$leadTag, decreasing = TRUE), ]
enrichplot::gseaplot2(ensembl_raw_salmon_gsea, geneSetID = rownames(head(ensembl_raw_salmon_gseaDF)))
```
<pre>



</pre>
### Enrichment score of the 5 GO's with the highest Tag (ie. leading edge has high Wald stats)
**Defintion of tag: The percentage of genes in the gene list L before (for positive ES) or after (for
negative ES) the peak in the running enrichment score, thus it gives an indication of
where in the list the enrichment score is attained.**
```{r gseaPlot5NESLocation}
ensembl_raw_salmon_gseaDF <- ensembl_raw_salmon_gseaDF[order(ensembl_raw_salmon_gseaDF$leadGene, decreasing = FALSE), ]
enrichplot::gseaplot2(ensembl_raw_salmon_gsea, geneSetID = rownames(head(ensembl_raw_salmon_gseaDF)))
```
<pre>



</pre>
### Enrichment score of the 5 GO's with the highest signal
**The enrichment signal strength that combines the two previous
statistics: (Tag %) × (1 – Gene %) × (N / (N – Nh) , where n equals the number of genes
in the list and Nh is the number of genes in the gene set. The larger this quantity, the more
enriched the gene set is as a whole. If the gene set is entirely within the first Nh positions
in the list, then the signal strength is maximal or 1. If the gene set is spread throughout
the list, then the signal strength decreases toward 0.**
```{r gseaPlot5Signal}
ensembl_raw_salmon_gseaDF <- ensembl_raw_salmon_gseaDF[order(ensembl_raw_salmon_gseaDF$leadSignal, decreasing = TRUE), ]
enrichplot::gseaplot2(ensembl_raw_salmon_gsea, geneSetID = rownames(head(ensembl_raw_salmon_gseaDF)))
```
<pre>



</pre>
### Enrichment plot for a manual selection of Pathways
```{r gseaplotManualSelection}
enrichplot::gseaplot2(ensembl_raw_salmon_gsea, geneSetID = c("GO:0001525", "GO:0050900", "GO:0045766", "GO:0001819", "GO:0002252", "GO:1903555", "GO:0071706", "GO:0042110", "GO:0032640", "GO:0032680", "GO:0002685", "GO:0007416"))
```
<pre>



</pre>
### Dotplot of all Pathways, ordered and plotted by normalized enrichment score
```{r variousDotplots}
p1 <- enrichplot::dotplot(
    ensembl_raw_salmon_gsea, orderBy = "NES", x = "NES", showCategory = 800) + 
    theme(axis.text.y = element_blank()) +
    ggtitle("ordered NES of GO pathways")
p2 <- enrichplot::dotplot(
    ensembl_raw_salmon_gsea, x = "geneRatio", showCategory = 800) + 
    theme(axis.text.y = element_blank()) +
    ggtitle("ordered leading edge of Go pathways")
ensembl_raw_salmon_gseaDF <- ensembl_raw_salmon_gseaDF[order(ensembl_raw_salmon_gseaDF$NES, decreasing = TRUE), ]
ensembl_raw_salmon_gseaDF$ID <- factor(ensembl_raw_salmon_gseaDF$ID, levels = ensembl_raw_salmon_gseaDF$ID)
p3 <- ggplot(
    data = ensembl_raw_salmon_gseaDF,
    aes(y = ID, x = NES, color = Direction)) +
    geom_point(size = 2) +
    scale_color_manual(values = c("#000099", "#990000")) +
    theme(axis.text.y = element_blank()) +
    ggtitle("ordered NES of GO pathways")
ensembl_raw_salmon_gseaDF <- ensembl_raw_salmon_gseaDF[order(ensembl_raw_salmon_gseaDF$leadTag, decreasing = TRUE), ]
ensembl_raw_salmon_gseaDF$ID <- factor(ensembl_raw_salmon_gseaDF$ID, levels = ensembl_raw_salmon_gseaDF$ID)
p4 <- ggplot(
    data = ensembl_raw_salmon_gseaDF,
    aes(y = ID, x = leadTag, color = Direction)) +
    geom_point(size = 2) +
    scale_color_manual(values = c("#000099", "#990000")) +
    theme(axis.text.y = element_blank()) + 
    ggtitle("ordered leading tag of GO pathways")
p1 + p2 + p3 + p4 + plot_layout(2,2)
```
<pre>



</pre>
### The leadTags(support of the NES) for all GO terms plotted as barplot
```{r barplotLeadingEdge}
ggplot(
    data = data.frame(table(ensembl_raw_salmon_gseaDF$leadTag)),
    aes(y = Freq, x = Var1)) +
geom_bar(stat = "identity", color = "black", fill = "black") +
xlab("leadTag")
```
<pre>



</pre>
### The Normalized enrichment score for all GO terms as a histogram
```{r histogramNES}
ggplot(
    data = ensembl_raw_salmon_gseaDF,
    aes(x = NES, color = Direction, fill = Direction)
) +
    geom_histogram(binwidth = max(abs(ensembl_raw_salmon_gseaDF$NES)) / 80) +
    scale_color_manual(values = c("#000099", "#990000")) +
    scale_fill_manual(values = c("#000099", "#990000")) +
    xlab("Normalized enrichment score") +
    ylab("Count of GO Pathways") +
    xlim(-3, 3)
```
<pre>



</pre> 
### The 40 Gene Ontologies with the highest q-scores in a barplot 
```{r barplotTop40sig}
ensembl_raw_salmon_gseaDF_qValue40 <- ensembl_raw_salmon_gseaDF[order(ensembl_raw_salmon_gseaDF$qvalue), ]
ensembl_raw_salmon_gseaDF_qValue40 <- head(ensembl_raw_salmon_gseaDF_qValue40, n = 40)
ensembl_raw_salmon_gseaDF_qValue40 <- data.frame("Description" = ensembl_raw_salmon_gseaDF_qValue40$Description, "p.adjust" = ensembl_raw_salmon_gseaDF_qValue40$p.adjust, "qvalue" = ensembl_raw_salmon_gseaDF_qValue40$qvalue)
ensembl_raw_salmon_gseaDF_qValue40 <- melt(ensembl_raw_salmon_gseaDF_qValue40, id.vars = "Description")
ensembl_raw_salmon_gseaDF_qValue40 <- plyr::join(
    ensembl_raw_salmon_gseaDF_qValue40, 
    ensembl_raw_salmon_gseaDF[, c("Description", "Direction")], 
    by = "Description",
    type = "left", 
    match = "all")
# converting Description to factor with levels so ggplot doesn't reorder my plot
ensembl_raw_salmon_gseaDF_qValue40$Description <- factor(ensembl_raw_salmon_gseaDF_qValue40$Description, levels = unique(ensembl_raw_salmon_gseaDF_qValue40$Description), ordered = TRUE )
# actuall plotting
p1 <- ggplot(
    data = ensembl_raw_salmon_gseaDF_qValue40,
    aes(x = Description, y = -log10(value), pattern = variable, fill = Direction, color = Direction)) +
coord_flip() +
geom_bar_pattern(
    stat = "identity", 
    position = position_dodge(preserve = "single"), 
    color = "black", 
    pattern_fill = "black",
    pattern_angle = 45,
    pattern_density = 0.1,
    pattern_spacing = 0.025,
    pattern_key_scale_factor = 0.6) +
scale_pattern_manual(values = c(p.adjust = "stripe", qvalue = "none")) +
scale_color_manual(values = c("#000099", "#990000")) +
scale_fill_manual(values = c("#000099", "#990000")) +
labs(x = "-log 10", y = element_blank(), pattern = "padj/qvalue", color = "UpRegulated/DownRegulated") + 
guides(
    pattern = guide_legend(override.aes = list(fill = "white")),
    fill = guide_legend(override.aes = list(pattern = "none"))) +
scale_x_discrete(limits = rev(levels(ensembl_raw_salmon_gseaDF_qValue40$Description))) +
theme_minimal() + 
ggtitle ("padj and qvalue for the 40 most significant GO Pathways ordered by qvalue")
```
<pre>



</pre>
### The Normalized Enrichment Score of the top 20 up and top 20 downregulated pathways
```{r barplotTop40NES}
ensembl_raw_salmon_gseaDF <- ensembl_raw_salmon_gseaDF[order(ensembl_raw_salmon_gseaDF$NES), ]
ensembl_raw_salmon_gseaDF_top40 <- rbind(head(ensembl_raw_salmon_gseaDF, n = 20), tail(ensembl_raw_salmon_gseaDF, n = 20))
# converting Description to factor with levels so ggplot doesn't reorder my plot
ensembl_raw_salmon_gseaDF_top40$Description <- factor(ensembl_raw_salmon_gseaDF_top40$Description, levels = ensembl_raw_salmon_gseaDF_top40$Description)
# actuall plotting
ggplot(
    data = ensembl_raw_salmon_gseaDF_top40,
    aes(x = Description, y = NES, color = Direction, fill = Direction)) +
    coord_flip() +
    geom_bar(stat = "identity") +
    scale_color_manual(values = c("#000099", "#990000")) +
    scale_fill_manual(values = c("#000099", "#990000")) +
    theme_minimal()
```
<pre>



</pre>
### A small network plot of the genes mapping to a selection of GO terms
```{r networkGenesToGO, message = FALSE, warning = FALSE}
enrichplot::cnetplot(
    ensembl_raw_salmon_gsea_selection,
    showCategory = 12,
    color.params = list(foldChange = ensembl_raw_salmon_DESeq_WaldStats),
    node_label = "gene",
    color_category = "black"
) +
    guides(color = guide_legend(title = "Wald statistics"))
```
<pre> 



</pre>
### Heatmap plotting gene expression for a selection of GO terms
```{r heatmapGOselection, message = FALSE}
enrichplot::heatplot(
    ensembl_raw_salmon_gsea_selection,
    foldChange = ensembl_raw_salmon_DESeq_WaldStats,
    showCategory = 20
) +
    theme(axis.text.x = element_blank()) +
    scale_fill_continuous(name = "Wald statistics") +
    xlab("genes")
```
<pre>



</pre> 
### The top 200 lowest GO's in a Network graph clustered to GO categories
```{r networkGOToCategory}
x <- enrichplot::pairwise_termsim(ensembl_raw_salmon_gsea, method = "JC")
enrichplot::emapplot(
    x,
    showCategory = 200,
    repel = TRUE,
    color = "NES",
    cluster.params = list(cluster = TRUE, legend = TRUE),
    node_label = "none"
)
## NO COMPRENDE: why can I not show all the GO's ?????? it says out of bound
## how exactly does it categorize the GO terms??
```
<pre>



</pre> 
### GO plot showing the significant points in GO hierarchy
```{r GoHierarchySignificance, warning = FALSE}
## in goplot showCategory seems to relate to GO hierarchy
enrichplot::goplot(
    ensembl_raw_salmon_gsea,
    showCategory = 5,
    repel = TRUE
)
# NO COMPRENDE: showCategories does what
```
<pre>



</pre>
### Networking with vissE and igraph
```{r, vissE}
ensembl_raw_salmon_gsea_GSC <- list()
for (i in rownames(ensembl_raw_salmon_gsea@result)) {
    geneSet <- GeneSet(unlist(strsplit(ensembl_raw_salmon_gsea@result[i, "core_enrichment"], split = "/"))[], setName = i)
    ensembl_raw_salmon_gsea_GSC[[i]] <- geneSet
}
geneSetCollection <- GeneSetCollection(object = ensembl_raw_salmon_gsea_GSC)

## VISUALIZATION

gs_ovlap <- computeMsigOverlap(geneSetCollection, thresh = 0.25, measure = "jaccard")
# create an overlap network
gs_ovnet <- computeMsigNetwork(gs_ovlap, geneSetCollection)
# plot the network
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet)


# ADD GENE SET STATISTCS

geneset_stats <- ensembl_raw_salmon_gsea@result$qvalue
names(geneset_stats) <- rownames(ensembl_raw_salmon_gsea@result)

# plot the network and overlay gene-set statistics
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, genesetStat = -log10(geneset_stats))


## IDENTIFY CLUSTERS

# identify clusters - order based on cluster size and avg gene-set stats
grps <- findMsigClusters(gs_ovnet,
    genesetStat = geneset_stats,
    alg = cluster_louvain,
    minSize = 5
)
# cluster_louvain works better than cluster_walktrap
# plot the top 12 clusters
set.seed(36) # set seed for reproducible layout
plotMsigNetwork(gs_ovnet, markGroups = grps[1:6], genesetStat = -log10(geneset_stats))

## OUR CLUSTERS IN DETAIL

set.seed(36) # set seed for reproducible layout
plotMsigNetwork(
    gs_ovnet,
    markGroups = grps[1:6],
    genesetStat = geneset_stats,
    rmUnmarkedGroups = TRUE
)
```
<pre>



</pre>
## Session info
```{r sessionInfo}
sessionInfo()
```
